---
title: "Compile expression data for A. thaliana"
output: html_notebook
---

# Load needed packages
```{r include=FALSE}
#library(ewrefxn)
rm(list=ls())
gc()
# library(stringr)
# library(reshape)
# library(data.table)
# library(dplyr)
# library(tidyr)

notebookDirectory = getwd()
date = Sys.Date()
```

# Create list of SRA runs for DNA data, Arabidopsis thaliana
```{r}
dnaSra = read.csv("SraRunInfo_SRP056687.csv")
write.table(dnaSra$Run, "dnaSra.txt", quote = F, row.names = F, col.names = F, sep = "\t")
```

# Look for more published stuides in sustech database
```{r}
# Load sustech database files
sustech = read.csv("lib_base_Info_200501_table.csv", header = F)
sustechPMID = read.csv("sustechDatabaseWithPMID.csv")

# Add headers
names(sustech) = c("Sample", "SampleName", "Tissue", "Ecotype", "Genotype", "Treatment", "Project", "TotalReads", "UniquelyMapped", "ReleaseDate")

# Check that sample columns match
all(sustech$Sample %in% sustechPMID$Sample)
all(sustechPMID$Sample %in% sustech$Sample)

# merge PMID information with other metadata
sustech = merge(sustech, sustechPMID[,c("Sample", "PMID")], by = "Sample")

#Subset only data from non-mutant lines and where tissue type is known
#11871 SAMPLES
sustech = sustech[(sustech$Genotype == "wild type" |  sustech$Genotype == "--") & (sustech$Tissue != "--") & (sustech$PMID != "/"),]

# Download run numbers for sustech database
# for(file in sustech$Sample){
#   download.file(paste("http://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?save=efetch&db=sra&rettype=runinfo&term=", file, sep = ""), paste(file, "_runInfo.csv", sep = ""), quiet = T)
# }

# Concatenate run info
# sustechRuns = list.files(pattern = ".*runInfo.csv", recursive = T)
# sustechRuns = lapply(sustechRuns, read.csv)
# sustechRuns = do.call("rbind", sustechRuns)
# write.csv(sustechRuns, "sustechRuns.csv", quote = F, row.names = F)

# Load sustech run info
sustechRuns = read.csv("sustechRuns.csv", header = T)

#Merge run numbers with experiment names
names(sustech)[1] = c("Experiment")
sustech = merge(sustech, sustechRuns[,c("Experiment", "Run")], by = "Experiment", all.x = T)

# merge run numbers with sample names
names(sustech)[1] = c("SampleName")
names(sustech)[2] = c("LibraryName")
sustech = merge(sustech, sustechRuns[,c("SampleName", "Run")], by = "SampleName", all.x = T)

# Check which samples still don't have run ids
any(is.na(sustech$Run.x) & is.na(sustech$Run.y))
which(is.na(sustech$Run.x) & is.na(sustech$Run.y))

# Drop 61 samples which can't be found on the SRA
sustech = sustech[(!is.na(sustech$Run.x) | !is.na(sustech$Run.y)),]

# merge together Run.x and Run.y
any(!is.na(sustech$Run.x) & !is.na(sustech$Run.y))
sustech$Run = sustech$Run.x
sustech$Run[(is.na(sustech$Run.x))] = sustech$Run.y[(is.na(sustech$Run.x))]

any(is.na(sustech$Run)) # Check that there are no leftover NAs
sustech = sustech[,!(names(sustech) %in% c("Run.x", "Run.y"))] # remove run.x and run.y

# Check for duplicates
any(is.na(sustech$SampleName))
any(duplicated(sustech))

# remove duplicated rows
sustech = unique(sustech)
any(duplicated(sustech))

# remove duplicated runs, only one row is dropped
any(duplicated(sustech$Run))
sustech = sustech[!duplicated(sustech$Run),]

# Change Project to bioproject
names(sustech)[1] = "Experiment"
names(sustech)[7] = "BioProject"
```

# Check today's SRA for updated data
```{r}
# compare this curated data to the set of all A. thaliana data on the SRA
# Date SRA downloaded: 2022-05-25
# This only works on my local computer, not the HPCC
# download.file(paste("http://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?save=efetch&db=sra&rettype=runinfo&term=", '"', "Arabidopsis thaliana", '"', '%5BOrganism%5D+AND+"biomol+rna"%5BProperties%5D', sep = ""), paste("all_arabidopsis_thaliana_runs_", date, ".csv", sep = ""), quiet = TRUE)

# Subset out data that isn't in sustech database
allArabidopsisRunInfo = read.csv("all_arabidopsis_thaliana_runInfo_2022-05-25_websiteDownloadV2.csv")
allArabidopisSummary = read.csv("all_arabidopsis_thaliana_summary_2022-05-25_websiteDownloadV2.csv")

# merge run info with summary info, this will help with annotating later
names(allArabidopisSummary)[1] = "Experiment"
allArabidopsisRunSummary = merge(allArabidopsisRunInfo, allArabidopisSummary, by = "Experiment")

# Check if all sustech data is in SRA download
# It mostly isn't, which is curious...
# This seems to mainly be because such samples either are not illumina samples or are not public
all(sustech$Run %in% allArabidopsisRunSummary$Run)
length(which(!(sustech$Run %in% allArabidopsisRunSummary$Run))) # 407 samples
sustech[!(sustech$Run %in% allArabidopsisRunSummary$Run),]

# Check which SRA runs are not in sustech database
inSRANotSustech = allArabidopsisRunSummary[!(allArabidopsisRunSummary$Run %in% sustech$Run),]

# remove samples without bioproject, because that's what I need to pull pubmed ids
inSRANotSustech = inSRANotSustech[(inSRANotSustech$BioProject != ""),]

# Write list of BioProjects to see which are published
# CONVERT DOS ENDLINES TO UNIX ENDLINES IF YOU GET WEIRD ERRORS WHILE TRYING TO EFETCH PUBMED IDS!
write.table(unique(inSRANotSustech[,"BioProject"]), "bioprojectsToCheckForPMID.txt", quote = F, row.names = F, col.names = F)

# Load publications associated with bioprojects not found in sustech database
pmids = read.csv("PMIDforBioprojects.txt", header = F)
names(pmids) = c("BioProject", "PMID")

pmids = pmids[(pmids$PMID != " "),]

# merge pmids with bioprojects not found in sustech database
inSRANotSustechWithPMID = merge(inSRANotSustech, pmids, by = "BioProject")

# check for duplicates, want FALSE
any(duplicated(inSRANotSustechWithPMID$Run))

#concatenate sustech database with data not in sustech database
inSRANotSustechWithPMID$Ecotype = NA
inSRANotSustechWithPMID$Genotype = NA
inSRANotSustechWithPMID$Tissue = NA
inSRANotSustechWithPMID$Treatment = NA

sustech$Experiment.Title = NA
sustech$SampleName = NA

columnsToKeep = c("PMID", "BioProject", "Experiment", "Run", "LibraryName", "Experiment.Title", "SampleName", "Ecotype", "Genotype", "Tissue", "Treatment")

metaData = rbind(sustech[,columnsToKeep],inSRANotSustechWithPMID[,columnsToKeep])

#Check for duplicates, get final counts
any(duplicated(metaData$Run))
length(metaData$Run) # 46469 runs
length(unique(metaData$BioProject)) # 1631 studies

# Sort studies in order from smallest to largest, start by annotating largest studies first
sort(table(metaData$BioProject), decreasing = T)[1:20]
```



# Remove studies, runs that don't meet my criteria
```{r}
# All studies should:
# 1. Meet my original search criteria: RNA-sequencing experiment from A. thaliana
# 2. come from samples that are not transgenic, hybrid, mutants, or polyploids. In other words, I only want to look at natural transcriptome diversity
# 3. Have enough metadata that I can put them into a treatment and tissue category. This means that they should at a minimum report the growth temperature, day length, and tissue type of each sample

# Ontology for removing studies:
# can't decipher labels
metaData = metaData[!(metaData$BioProject == "PRJNA288754"),] # can't understand labels
metaData = metaData[!(metaData$BioProject == "PRJNA272859"),] # can't understand labels
metaData = metaData[!(metaData$BioProject == "PRJNA248603"),] # can't understand labels
metaData = metaData[!(metaData$BioProject == "PRJNA223275"),] # can't understand labels
metaData = metaData[!(metaData$BioProject == "PRJNA312199"),] # can't understand labels
metaData = metaData[!((metaData$BioProject == "PRJNA325158") & grepl("EV", metaData$LibraryName)),] # not sure what these are
metaData = metaData[!(metaData$BioProject == "PRJNA437291"),] # can't decipher labels
metaData = metaData[!(metaData$BioProject == "PRJNA526886"),] # can't decipher labels
metaData = metaData[!(metaData$BioProject == "PRJNA339870"),] # can't decipher labels
metaData = metaData[!(metaData$BioProject == "PRJNA320769"),] # can't decipher labels
metaData = metaData[!(metaData$BioProject == "PRJNA484670"),] # can't decipher labels
metaData = metaData[!(metaData$BioProject == "PRJNA254505"),] # can't decipher labels
metaData = metaData[!(metaData$BioProject == "PRJEB8886"),] # can't decipher labels
metaData = metaData[!(metaData$BioProject == "PRJNA526343"),] # don't know what PQ is
metaData = metaData[!(metaData$BioProject == "PRJNA526125"),]  # don't know what HS is
metaData = metaData[!(metaData$BioProject == "PRJNA254031"),] # can't decipher labels
metaData = metaData[!(metaData$BioProject == "PRJEB14117"),] # can't decipher labels
metaData = metaData[!(metaData$BioProject == "PRJNA395944"),] # can't decipher labels
metaData = metaData[!(metaData$BioProject == "PRJNA491484"),] # can't decipher labels
metaData = metaData[!(metaData$BioProject == "PRJNA318918"),] # can't decipher labels
metaData = metaData[!(metaData$BioProject == "PRJNA267853"),] # can't decipher labels
metaData = metaData[!(metaData$BioProject == "PRJNA497220"),] # can't decipher labels
metaData = metaData[!(metaData$BioProject == "PRJNA399521"),] # can't decipher labels
metaData = metaData[!(metaData$BioProject == "PRJNA520905"),] # can't decipher labels
metaData = metaData[!(metaData$BioProject == "PRJNA490087"),] # can't decipher labels
metaData = metaData[!(metaData$BioProject == "PRJNA533962"),] # can't decipher labels
metaData = metaData[!(metaData$BioProject == "PRJNA517148"),] # can't decipher labels
metaData = metaData[!(metaData$BioProject == "PRJNA691022"),] # can't decipher labels
metaData = metaData[!((metaData$BioProject == "PRJNA435505") & grepl("_re_", metaData$LibraryName)),] # not sure what these are
metaData = metaData[!(metaData$BioProject == "PRJNA336471"),] # can't decipher labels
metaData = metaData[!(metaData$BioProject == "PRJNA272563"),] # can't decipher labels, only one biosample on SRA
metaData = metaData[!(metaData$BioProject == "PRJNA309985"),] # can't decipher labels
metaData = metaData[!(metaData$BioProject == "PRJEB1440"),] # can't decipher labels
metaData = metaData[!(metaData$BioProject == "PRJEB21008"),] # can't decipher labels
metaData = metaData[!(metaData$BioProject == "PRJNA316322"),] # can't decipher labels, no growth temperature recorded
metaData = metaData[!(metaData$BioProject == "PRJNA354686"),] # can't decipher labels, mocks also labeled as being treated with JA
metaData = metaData[!(metaData$BioProject == "PRJNA486068"),] # can't decipher labels, two different daylengths reported (14 hr or 10 hr) but not linked to samples
metaData = metaData[!(metaData$BioProject == "PRJNA505705"),] # can't decipher labels, no daylength or temperature reported
metaData = metaData[!(metaData$BioProject == "PRJNA554414"),] # can't decipher labels, no daylength or temperature reported, just seeds buried in the ground
metaData = metaData[!(metaData$BioProject == "SRP012576"),] # can't decipher labels, no daylength or temperature reported
metaData = metaData[!(metaData$BioProject == "PRJNA289063"),] # can't decipher labels, no growth temperature reported
metaData = metaData[!(metaData$BioProject == "PRJNA301684"),] # can't decipher labels, no daylength or temperature reported
metaData = metaData[!(metaData$BioProject == "PRJNA602727"),] # can't decipher labels, no growth temperature reported
metaData = metaData[!(metaData$BioProject == "PRJNA152685"),] # can't decipher labels, no growth temperature reported
metaData = metaData[!(metaData$BioProject == "PRJNA780518"),] # can't decipher labels, no daylength or temperature reported
metaData = metaData[!(metaData$BioProject == "PRJNA156671"),] # can't decipher labels, no daylength or temperature reported
metaData = metaData[!(metaData$BioProject == "PRJNA230565"),] # can't decipher labels, no growth temperature reported
metaData = metaData[!(metaData$BioProject == "PRJNA493902"),] # can't decipher labels, don't know how Sustech database labeled samples, because info isn't in SRA annotations
metaData = metaData[!(metaData$BioProject == "PRJNA394215"),] # can't decipher labels, no daylength or temperature reported
metaData = metaData[!(metaData$BioProject == "PRJNA549137"),] # can't decipher labels, no daylength or temperature reported
metaData = metaData[!(metaData$BioProject == "PRJNA354373"),] # can't decipher labels, no daylength or temperature reported
metaData = metaData[!(metaData$BioProject == "PRJNA735204"),] # can't decipher labels, paper is 
metaData = metaData[!(metaData$BioProject == "PRJEB17995"),] # can't decipher labels, no growth temperature reported
metaData = metaData[!(metaData$BioProject == "PRJNA323731"),] # can't decipher labels, no daylength or temperature reported
metaData = metaData[!(metaData$BioProject == "PRJNA354456"),] # can't decipher labels, no daylength or temperature reported
metaData = metaData[!(metaData$BioProject == "PRJNA495850"),] # can't decipher labels, don't know how to label primed samples
metaData = metaData[!(metaData$BioProject == "PRJNA525779"),] # can't decipher labels, no daylength or temperature reported

### transgenic
metaData = metaData[!((metaData$BioProject == "PRJNA301684") & grepl("#", metaData$LibraryName)),]
metaData = metaData[!(metaData$BioProject == "PRJNA271499"),] # all transgenic, potentially
metaData = metaData[!((metaData$BioProject == "PRJNA416120") & grepl("35s", metaData$LibraryName)),]
metaData = metaData[!((metaData$BioProject == "PRJNA416120") & grepl("cbf", metaData$LibraryName)),]
metaData = metaData[!((metaData$BioProject == "PRJNA416120") & grepl("it", metaData$LibraryName)),]
metaData = metaData[!(metaData$BioProject == "PRJNA369824"),] # transgenic
metaData = metaData[!(metaData$BioProject == "PRJNA384359"),] # transgenic
metaData = metaData[!((metaData$BioProject == "PRJNA382852") & grepl("OX1-", metaData$LibraryName)),]
metaData = metaData[!((metaData$BioProject == "PRJNA382852") & grepl("GFP", metaData$LibraryName)),]
metaData = metaData[!((metaData$BioProject == "PRJNA382852") & grepl("sr45", metaData$LibraryName)),]
metaData = metaData[!((metaData$BioProject == "PRJNA380494") & grepl("GFP", metaData$LibraryName)),]
metaData = metaData[!((metaData$BioProject == "PRJNA325158") & grepl("CEB1", metaData$LibraryName)),]
metaData = metaData[!(metaData$BioProject == "PRJNA309713"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA418881"),] # transgenics
metaData = metaData[!((metaData$BioProject == "PRJNA419694") & grepl("OE", metaData$LibraryName)),] # transgenics 
metaData = metaData[!(metaData$BioProject == "PRJNA403945"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA354618"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA354104"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA354937"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA337962"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA267558"),] # transgenics
metaData = metaData[!((metaData$BioProject == "PRJNA292748") & grepl("abi|pub", metaData$LibraryName)),] # transgenics 
metaData = metaData[!(metaData$BioProject == "PRJNA579992"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA477936"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA355001"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA218027"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA575031"),] # maybe transgenics?
metaData = metaData[!(metaData$BioProject == "PRJNA529955"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA356835"),] # overexpression
metaData = metaData[!(metaData$BioProject == "PRJNA339001"),] # overexpression
metaData = metaData[!(metaData$BioProject == "PRJNA316811"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA316129"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA307295"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA319316"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA279505"),] # mutant
metaData = metaData[!(metaData$BioProject == "PRJNA316333"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA604023"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA517774"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA314824"),] # mutant
metaData = metaData[!(metaData$BioProject == "PRJNA435845"),] # mutant
metaData = metaData[!(metaData$BioProject == "PRJNA515610"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA530985"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA481647"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA173788"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA359514"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA498076"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA430725"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA388005"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA473829" & grepl("^coi", metaData$LibraryName)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA473829" & grepl("^npr", metaData$LibraryName)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA348676" & grepl("_deps_|_pad4_|_rr_|_ps_|_sid2_", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA703045"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA665236"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA318880"),] # transgenics and mutants
metaData = metaData[!(metaData$BioProject == "PRJNA323955"),] # transgenics, GFP marker lines
metaData = metaData[!(metaData$BioProject == "PRJNA639173"),] # transgenics, TF transfections?
metaData = metaData[!(metaData$BioProject == "PRJNA665689"),] # transgenics, marker lines to help do RNA-seq
metaData = metaData[!(metaData$BioProject == "PRJNA594026"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA529898"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA526775"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA341963"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA506408"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA429659"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA323943" & grepl("A|B", metaData$Experiment.Title)),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA525432" & grepl("pEG100", metaData$Experiment.Title)),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA595951"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA198192"),] # transgenics, single-cell RNA-seq markers
metaData = metaData[!(metaData$BioProject == "PRJNA300452"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA595605"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA234030"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA495335"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA742744"),] # transgenics, single-cell RNA-seq markers
metaData = metaData[!(metaData$BioProject == "PRJNA383997"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA596429"),] # transgenics, single-cell RNA-seq markers
metaData = metaData[!(metaData$BioProject == "PRJNA408295"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA79729"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA704079"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA677901"),] # transgenics, single-cell RNA-seq markers
metaData = metaData[!(metaData$BioProject == "PRJNA722928"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA354623"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA369329"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA594413"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA594539"),] # transgenics, single-cell RNA-seq markers
metaData = metaData[!(metaData$BioProject == "PRJNA433874"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA506007"),] # transgenics, single-cell RNA-seq markers
metaData = metaData[!(metaData$BioProject == "PRJNA315353"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA347353"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA352042"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA418787"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA589627"),] # transgenics

#### mutants ####
metaData = metaData[!(metaData$BioProject == "PRJNA486575" & grepl("elp2-5|met1-7", metaData$Experiment.Title)),]
metaData = metaData[!(metaData$BioProject == "PRJNA313379" & grepl("ein2-1|fls2|dde2|pad4-1|sid2-2", metaData$Experiment.Title)),]
metaData = metaData[!(metaData$BioProject == "PRJNA375085" & grepl("PhyD", metaData$LibraryName)),]
metaData = metaData[!(metaData$BioProject == "PRJNA422118" & !grepl("Col0", metaData$Experiment.Title)),] # T-DNA
metaData = metaData[!(metaData$BioProject == "PRJNA481720" & grepl("rbohd", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA666631" & !grepl("WT|Wtcif", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA508214" & grepl("dgat1-1", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA564095" & grepl("pp2a-b", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA564095" & grepl("pp2a-b", metaData$LibraryName)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA344452" & grepl("phf1|phr1|phl1", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA341504" & grepl("CS|VS|VC", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA566089"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA224705" & grepl("ein|jar|npr", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA272491" & grepl("ago2|HA-AGO2-DAD|HA-AGO1-DAH|HA-AGO10-DDH", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA344451" & grepl("phr1|phl1", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA592549" & grepl("siz", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA693252" & grepl("mbd56|mbd5|SLN|sln|mbd6|MBD6|MBD5|met1|fwa", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA524697" & grepl("H89C|NEET", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA692747" & grepl("clsy|nrpd|shh", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA641427" & grepl("wrky25/33", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA635635" & grepl("myb36", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA378644" & grepl("ss3|ss4|pgm", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA344453" & grepl("phr1|phl1", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA494862" & grepl("coi|dorn|npr|ein|Ox", metaData$SampleName)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA258547" & grepl("STM|REV|KAN|AS2", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA691326" & grepl("clsy|CLSY|nrpd|shh|nprd|nrp", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA171012"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA360931" & grepl("oe|bri|53t|54t|bes|atg|RNAi|6_", metaData$Experiment.Title)),] # mutants and transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA302786" & grepl("sfr|cob", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA383861"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA506000"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA551415" & grepl("myc", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA529796" & grepl("met|nrp|rdr|drm|mea", metaData$LibraryName)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA557616" & grepl("GUN", metaData$LibraryName)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA602947" & grepl("cif|sgn", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA611770" & grepl("bzip", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA476945" & grepl("camta", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA284739" & grepl("71", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA311764" & grepl("a56|mbd|thp|brca|EMS|sac|GFP|atsubtl", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA601788" & grepl("med|GFP|morc|ddc", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA527966" & grepl("MU", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA590705" & grepl("qem|svp|sail", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA595162" & grepl("myb", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA662710" & !grepl("C", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA368967" & grepl("tz|py|Mutant|mutant", metaData$SampleName)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA427202" & grepl("ZF108|suvh|DNAJ|dnaj", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA598201" & grepl("B", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA209363" & grepl("rrs", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA546251" & grepl("uvr|pif", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA626535" & grepl("bzip|ire", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA188075" & !grepl("WT", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA301162" & grepl("mutant", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA270873"),] # mutants? Single-cell sequencing
metaData = metaData[!(metaData$BioProject == "PRJNA310623" & grepl("i4g|K48M", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA310623" & grepl("4A-15", metaData$LibraryName)),] # overexpression
metaData = metaData[!(metaData$BioProject == "PRJNA702187" & grepl("atxr|ddm", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA245920"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA255977" & grepl("p4|p5", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA281364" & grepl("hml|ivd", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA305733" & grepl("aba|p5cs", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA315374" & grepl("O", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA339762" & grepl("w18|w40", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA363056" & grepl("bzip", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA391475" & grepl("c", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA525816" & grepl("pip", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA574854" & grepl("HY5|PHY", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA574855" & grepl("hy5|phy", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA609342" & grepl("81|82|83|hf|8WL", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA642797"),] # mutants and transgenics!
metaData = metaData[!(metaData$BioProject == "PRJNA658687" & grepl("atg", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA661316" & grepl("lyk|LYK", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA666544" & grepl("zin1", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA153297"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA261075"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA476778" & grepl("cca1|prr", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA406971" & grepl("tcp|TCP", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA446045" & grepl("bdrs", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA481452" & grepl("uvr|cry", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA543989"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA510535" & grepl("pif", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA513852" & grepl("pfd|lsm8", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA666436" & grepl("mutant", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA342309" & grepl("drm", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA565160" & grepl("^s189|^s144", metaData$LibraryName)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA672150" & grepl("ICT", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA716358" & grepl("xpf|uvr", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA531169" & grepl("phf|phr", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA258161" & grepl("lsm", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA261433" & grepl("clf28", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA274887" & !grepl("Col-0", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA294313" & grepl("Type-A", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA302003" & grepl("mapkkk5", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA322265" & grepl("rpn", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA345276" & grepl("hsp101", metaData$LibraryName)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA412889" & grepl("SIG", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA445392" & grepl("psp", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA589109" & grepl("agb", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA596803" & grepl("dxo", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA624317" & grepl("stop1", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA646744" & grepl("npr", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA667059" & grepl("hy5", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA681183" & grepl("_258", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA706054" & grepl("ino80", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA525452" & grepl("pkl", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA561790" & grepl("lrx", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA358395" & grepl("camta", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA474441" & grepl("della", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA624145" & grepl("urt", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA132271" & grepl("spl", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA267681" & grepl("gemin", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA378128" & grepl("ago", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA471243" & grepl("pyl", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA523174" & grepl("Mutant", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA593027" & grepl("stch", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA609261" & grepl("pifq", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA610184"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA643438" & grepl("abr", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA661303" & grepl("b123", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA704067" & grepl("floe", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA287644" & grepl("ESTAIG", metaData$Experiment.Title)),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA577116"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA309654"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA391284"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA544185"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA638809"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA648424"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA701753"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA145305"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA306014" & grepl("mutant", metaData$LibraryName)),] # mutants

#### hybrids ####
metaData = metaData[!(metaData$BioProject == "PRJNA368916"),] # RILs, MAGIC lines
metaData = metaData[!(metaData$BioProject == "PRJNA358059"),] # F2 population
metaData = metaData[!(metaData$BioProject == "PRJNA380541"),] # hybrids
metaData = metaData[!(metaData$BioProject == "PRJNA382583" & grepl("pool", metaData$Experiment.Title)),] # RILs
metaData = metaData[!(metaData$BioProject == "PRJNA302221"),] # RILs
metaData = metaData[!(metaData$BioProject == "PRJNA379803"),] # MAGIC lines
metaData = metaData[!(metaData$BioProject == "PRJNA544417" & grepl("HM|x|sml", metaData$Experiment.Title)),] # hybrids, and hybrid mimics
metaData = metaData[!(metaData$BioProject == "PRJNA529796" & grepl("Col0_x_Tsu|LerxCol|Ler_x_Col|Tsu_x_Col0|ColxLer|Col0_x_Ler", metaData$LibraryName)),] # hybrids
metaData = metaData[!(metaData$BioProject == "PRJNA562212"),] # hybrids
metaData = metaData[!(metaData$BioProject == "PRJNA263595" & grepl("Fcs|Fsc", metaData$LibraryName)),] # hybrids
metaData = metaData[!(metaData$BioProject == "PRJNA132535"),] # hybrids
metaData = metaData[!(metaData$BioProject == "PRJNA271089" & grepl("C24xLer|LerxC24|ColxLer|LerxCol|C24xCol|ColxC24", metaData$Experiment.Title)),] # hybrids
metaData = metaData[!(metaData$BioProject == "PRJNA476580"),] # hybrids
metaData = metaData[!(metaData$BioProject == "PRJNA387601" & grepl("x", metaData$LibraryName)),] # hybrids
metaData = metaData[!(metaData$BioProject == "PRJNA523801"),] # hybrids

# polyploid
metaData = metaData[!(metaData$BioProject == "PRJNA473317"),] # ploidy levels!
metaData = metaData[!(metaData$BioProject == "PRJNA141211"),] # ploidy levels!
metaData = metaData[!(metaData$BioProject == "PRJNA448456"),] # ploidy levels!
metaData = metaData[!(metaData$BioProject == "PRJNA358016"),] # ploidy levels!
metaData = metaData[!(metaData$BioProject == "PRJNA534240"),] # ploidy levels!

# aneuploid
metaData = metaData[!(metaData$BioProject == "PRJNA316697"),] # aneuploids

# wrong species
metaData = metaData[!grepl("Arabidopsis lyrata", metaData$LibraryName),] # remove data from other species
metaData = metaData[!(metaData$BioProject == "PRJNA284922"),] # includes other species

# cell cultures
metaData = metaData[!(metaData$BioProject == "PRJEB31496"),] # includes other species
metaData = metaData[!(metaData$BioProject == "PRJNA278518"),]

#### non-coding RNA-seq ####
metaData = metaData[!(metaData$BioProject == "PRJNA518089"),] # lncRNAseq
metaData = metaData[!(metaData$BioProject == "PRJNA494179" & grepl("ncRNA-seq", metaData$Experiment.Title)),]
metaData = metaData[!(metaData$BioProject == "PRJNA300285"),] # ncRNA-seq
metaData = metaData[!(metaData$BioProject == "PRJNA545832"),] # miRNA-seq
metaData = metaData[!(metaData$BioProject == "PRJNA554003"),] # miRNA-seq
metaData = metaData[!(metaData$BioProject == "PRJNA315805"),] # ncRNA-seq
metaData = metaData[!(metaData$BioProject == "PRJNA557616" & grepl("sRNA", metaData$LibraryName)),] # sRNA
metaData = metaData[!(metaData$BioProject == "PRJNA449355"),] # nanoPARE???
metaData = metaData[!(metaData$BioProject == "PRJNA389307"),] # ncRNA-seq
metaData = metaData[!(metaData$BioProject == "PRJNA695075"),] # ncRNA-seq
metaData = metaData[!(metaData$BioProject == "PRJNA186843" & grepl("ncRNA-Seq", metaData$Experiment.Title)),] # ncRNA-seq
metaData = metaData[!(metaData$BioProject == "PRJNA277424"),] # ncRNA-seq
metaData = metaData[!(metaData$BioProject == "PRJNA513320"),] # ncRNA-seq
metaData = metaData[!(metaData$BioProject == "PRJNA715469"),] # ncRNA-seq
metaData = metaData[!(metaData$BioProject == "PRJNA725493"),] # ncRNA-seq
metaData = metaData[!(metaData$BioProject == "PRJNA374333"),] # ncRNA-seq
metaData = metaData[!(metaData$BioProject == "PRJNA251351"),] # ncRNA-seq
metaData = metaData[!(metaData$BioProject == "PRJNA592037" & grepl("sRNA", metaData$LibraryName)),] # sRNA-seq
metaData = metaData[!(metaData$BioProject == "PRJNA717313"),] # ncRNA-seq
metaData = metaData[!(metaData$BioProject == "PRJNA736042"),] # ncRNA-seq
metaData = metaData[!(metaData$BioProject == "PRJNA433501" & grepl("GSM", metaData$Experiment.Title)),] # ncRNA-seq
metaData = metaData[!(metaData$BioProject == "PRJNA369749"),] # ncRNA-seq
metaData = metaData[!(metaData$BioProject == "PRJNA544565"),] # plaNET-seq
metaData = metaData[!(metaData$BioProject == "PRJNA196840"),] # plaNET-seq
metaData = metaData[!(metaData$BioProject == "PRJNA294703" & grepl("RIP-Seq", metaData$Experiment.Title)),]
metaData = metaData[!(metaData$BioProject == "PRJNA388398"),] # rRNA-seq???
metaData = metaData[!(metaData$BioProject == "PRJNA122227"),] # sRNA-seq
metaData = metaData[!(metaData$BioProject == "PRJNA680602"),] # NAD-seq
metaData = metaData[!(metaData$BioProject == "PRJNA145973"),] # sRNA-seq
metaData = metaData[!(metaData$BioProject == "PRJNA182746"),] # lncRNAseq

# grafting
metaData = metaData[!(metaData$BioProject == "PRJNA124841"),]

# no discernable treatment
metaData = metaData[!(metaData$BioProject == "PRJNA393568"),] # RNA polymerase, doesn't really fit my idea of treatment
metaData = metaData[!(metaData$BioProject == "PRJNA488814"),] # imbibed treatment, not sure what to do
metaData = metaData[!(metaData$BioProject == "PRJNA428974"),] # modifying RNA
metaData = metaData[!grepl("PRJNA483458", metaData$BioProject),] # Transcription factor infiltration
metaData = metaData[!(grepl("PRJNA232536", metaData$BioProject) & grepl("unassigned", metaData$Experiment.Title)),] # unassigned reads
metaData = metaData[!(grepl("PRJNA224705", metaData$BioProject) & grepl("unassigned", metaData$LibraryName)),] # unassigned reads
metaData = metaData[!(metaData$BioProject == "PRJNA693252" & grepl("GROseq", metaData$Experiment.Title)),] # odd sequencing approach
metaData = metaData[!(grepl("PRJNA209363", metaData$BioProject) & grepl("Unassigned", metaData$LibraryName)),] # unassigned reads
metaData = metaData[!(grepl("SRP000935", metaData$BioProject) & grepl("FC", metaData$LibraryName)),]

#### too few runs after removing runs that don't meet criteria (remove if # of good runs <= 8) ####
metaData = metaData[!(metaData$BioProject == "PRJNA360931"),] # hybrids, and hybrid mimics
metaData = metaData[!(metaData$BioProject == "PRJNA182723"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA560779"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA557616"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA736310"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA612079"),] # Ribo-seq
metaData = metaData[!(metaData$BioProject == "PRJNA427202"),] # can't decipher labels, didn't record growing temperature
metaData = metaData[!(metaData$BioProject == "PRJNA665187"),] # transgenic
metaData = metaData[!(metaData$BioProject == "PRJNA390966"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA377549"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA513324"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA688172"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA716051"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA388580"),] # hybrids
metaData = metaData[!(metaData$BioProject == "PRJNA255977"),] # can't decipher labels, didn't record growing temperature
metaData = metaData[!(metaData$BioProject == "PRJNA263595"),] # can't decipher labels, don't know which samples were infiltrated with pathogen or MgCl
metaData = metaData[!(metaData$BioProject == "PRJNA305733"),] # can't decipher labels, no growth temperature or day length
metaData = metaData[!(metaData$BioProject == "PRJNA554145"),] # can't decipher labels, no growth temperature
metaData = metaData[!(metaData$BioProject == "PRJNA761299"),] # ncRNA-seq
metaData = metaData[!(metaData$BioProject == "PRJNA663790"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA271927"),] # grafting
metaData = metaData[!(metaData$BioProject == "PRJNA560784"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA217719"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA421154"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA338792"),] # hybrids
metaData = metaData[!(metaData$BioProject == "PRJNA541235"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA549285"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA167620"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA230119"),] # hybrids
metaData = metaData[!(metaData$BioProject == "PRJNA371457"),] # hybrids
metaData = metaData[!(metaData$BioProject == "PRJNA591604"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA504621"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA720907"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA505255"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA421152"),] # hybrids
metaData = metaData[!(metaData$BioProject == "PRJNA716334"),] # hybrids
metaData = metaData[!(metaData$BioProject == "PRJNA529715"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA237334"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA244971"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA379206"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA388407"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA388948"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA388976"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA389787"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA394759"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA454801"),] # transgenics and hybrids
metaData = metaData[!(metaData$BioProject == "PRJNA630754"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA221223"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA156363"),] # hybrids
metaData = metaData[!(metaData$BioProject == "PRJNA295989"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA317322"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA336152"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA339455"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA428513"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA374742"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA176940"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA218542"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA284487"),] # wrong species
# transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA471232"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA591466"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA630258"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA433404"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA454379"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA566098"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA253731"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA321470"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA479446"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA298961"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA300223"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA304946"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA339285"),] # hybrids
metaData = metaData[!(metaData$BioProject == "PRJNA357201"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA358157"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA528094"),] # transgenics
metaData = metaData[!(metaData$BioProject == "PRJNA561509"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA262970"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA264544"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA657702"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA762245"),] # ncRNA-seq

# mutants
metaData = metaData[!(metaData$BioProject == "PRJNA702184"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA380303"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA380302"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA314197"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA314105"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA169134"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA300685"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA309285"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA318879"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA379910"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA415950"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA438240"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA527964"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA547818"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA574938"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA601526"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA639500"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA688924"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA702179"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA702180"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA222364"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA485633"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA745095"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA344394"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA389308"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA475117"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA481068"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA613599"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA684322"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA684323"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA596802"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA108797"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA190673"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA779686"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA151589"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA272719"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA318638"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA338072"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA354370"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA446046"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA473499"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA485018"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA491272"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA529391"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA531657"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA531896"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA560053"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA564817"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA595115"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA609667"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA667267"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA716333"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA718622"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA151239"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA326338"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA434313"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA549307"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA639271"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA215313"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA247731"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA283396"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA326409"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA361224"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA377889"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA507972"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA524935"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA566443"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA625566"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA641025"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA641644"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA642871"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA660429"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA702183"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA753821"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA816700"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA412461"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA418578"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA436416"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA494398"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA600990"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA604567"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA638126"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA672885"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA702185"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA704076"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA716808"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA790958"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA302602"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA314484"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA429781"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA486520"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA529202"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA153225"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA245833"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA245874"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA437500"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA493255"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA573713"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA167967"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA170237"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA234023"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA259643"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA268383"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA276444"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA277840"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA280957"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA281004"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA289110"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA293514"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA314522"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA314692"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA315850"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA320883"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA321477"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA331045"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA336058"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA339647"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA341576"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA342005"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA347360"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA357502"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA362650"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA369328"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA379315"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA384596"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA384645"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA384979"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA388207"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA389285"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA390146"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA395456"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA398488"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA401700"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA408168"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA414070"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA421364"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA427312"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA427587"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA429247"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA436256"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA439011"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA449911"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA451423"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA472158"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA481007"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA482933"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA487051"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA490100"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA505275"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA509341"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA511129"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA523248"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA525997"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA526669"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA541229"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA542085"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA544694"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA550085"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA556865"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA559405"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA562295"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA564278"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA589533"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA593852"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA597366"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA601525"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA609973"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA614936"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA629545"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA630230"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA646453"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA647214"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA649527"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA659346"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA666927"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA668243"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA675765"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA678965"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA692444"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA702186"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA744409"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA744433"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA744560"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA176941"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA227808"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA248481"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA254520"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA354407"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA162669"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA221230"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA294094"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA304022"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA317179"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA326392"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA399864"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA432917"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA472997"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA480268"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA486344"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA498071"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA579920"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA176484"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA288172"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA297107"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA310312"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA317722"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA353122"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA354385"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA354408"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA389675"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA417457"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA437964"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA446047"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA448672"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA476942"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA479861"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA497374"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA511473"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA529456"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA548351"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA549690"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA562289"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA592211"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA612487"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA616378"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA632034"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA632509"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA639732"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA722978"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA734575"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA735326"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA735758"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA780875"),] # mutants
metaData = metaData[!(metaData$BioProject == "PRJNA788534"),] # mutants

# hybrids
metaData = metaData[!(metaData$BioProject == "PRJNA275436"),] # hybrids
metaData = metaData[!(metaData$BioProject == "PRJNA183948"),] # hybrids
metaData = metaData[!(metaData$BioProject == "PRJEB35324"),] # hybrids
metaData = metaData[!(metaData$BioProject == "PRJNA244365"),] # hybrids
metaData = metaData[!(metaData$BioProject == "PRJNA223529"),] # hybrids
metaData = metaData[!(metaData$BioProject == "PRJNA264397"),] # hybrids
metaData = metaData[!(metaData$BioProject == "PRJNA362953"),] # hybrids
metaData = metaData[!(metaData$BioProject == "PRJNA186735"),] # hybrids
metaData = metaData[!(metaData$BioProject == "PRJNA143331"),] # hybrids
metaData = metaData[!(metaData$BioProject == "PRJNA529796"),] # hybrids

# not focused on mRNA
metaData = metaData[!(metaData$BioProject == "PRJNA730562"),] # RIP-Seq
metaData = metaData[!(metaData$BioProject == "PRJNA607603"),] # RIP-Seq
metaData = metaData[!(metaData$BioProject == "PRJNA272491"),] # RIP-Seq
metaData = metaData[!(metaData$BioProject == "PRJNA527732"),] # ATAC-Seq, CHIP-Seq
metaData = metaData[!(metaData$BioProject == "PRJNA156359"),] # smRNA-seq
metaData = metaData[!(metaData$BioProject == "PRJNA422491"),] # RIP-Seq
metaData = metaData[!(metaData$BioProject == "PRJNA649267"),] # smNucRNA-Seq

# batch remove runs based on key words
metaData = metaData[(!grepl("GUS|YFP|GFP|35S", metaData$Experiment.Title)),] # transgenics

metaData = metaData[!(grepl(" x |_x_", metaData$Experiment.Title)),] # hybrids

# metaData[(grepl("; ncRNA-Seq|sRNA|; miRNA-Seq|; OTHER| small RNA|RIP-seq", metaData$Experiment.Title)),]
metaData = metaData[!(grepl("; ncRNA-Seq|sRNA|; miRNA-Seq|; OTHER| small RNA|RIP-seq", metaData$Experiment.Title)),] # not mRNA-seq

# Remove studies that are too small
smallStudies = names(table(metaData$BioProject)[table(metaData$BioProject) <= 8])
# metaData[(metaData$BioProject %in% smallStudies),]
metaData = metaData[!(metaData$BioProject %in% smallStudies),]

# summarize remaining data
length(metaData$Run) # 45162 runs
length(unique(metaData$BioProject)) # 1557 BioProjects
length(unique(metaData$PMID)) # 1713 study groups

# Sort studies in order from smallest to largest
sort(table(metaData$BioProject), decreasing = T)[290:324] 

# screen for studies that obviously won't work
# screenStudies = names(sort(table(metaData$BioProject)[table(metaData$BioProject) <= 26], decreasing = T))
# for(project in screenStudies[561:564]){
#   print(metaData[(metaData$BioProject == project),])
# }
```

# Assign Treatment factor levels
## Use Plant Experimental Conditions Ontology: https://browser.planteome.org/amigo/term/PECO:0007359#display-lineage-tab
```{r}
# Create column to store treatment ontologies
metaData$TreatmentOntology = NA

# Interesting studies to remember for later:
# ***Look at zero-gravity studies, probably involved constant light too
# ***Look at studies larger than PRJNA314076 to look for long day length samples***
# ***Annotate tissues for PRJNA314076***
# PRJNA475542 is a potentially useful study for accounting for time of day variation in sampling
# PRJNA315516 some water recovery samples are mislabeld in sustech database, 
# PRJNA478998 one PMID is wrong...
# PRJNA691022 has pH treatment, but no labels on samples without looking at SRA. I removed it for now
# PRJNA575019 interesting local vs systemic treatments
# PRJNA562704 has many runs per experiment when it shouldn't, possibly
# PRJEB32665 has mixed tissue samples
# PRJNA665689 not sure if samples are from transgenics or not
# PRJNA622644 and PRJNA609704 would be a good studies to double-check findings, while accounting for batch effects
# PRJNA496426, embryo development timeline
# PRJNA310623, conflicting labels for control samples, some labeled as "dark" treatment on sra
# PRJNA626535, not sure if this recorded temperature and light data
# PRJNA481452, UV0 and blue0 may refer to exclusion of light
# PRJNA551121, has a sound treatment, consider including it
# PRJNA716358, associated article is not publically available
# PRJNA314076 may be a great study to get a more rigorous calculation of tissue-specificity from

#### control exposure ####
# 1. temperature is around 22-23 C
# 2. data comes from growth chambers unless specified otherwise
# 3. data comes from long day conditions unless specified otherwise
# Hot temperatures defined as 27 C and above
# Cold temperatures are 17 C and below
# Freezing temperatures are 4 C and below
# short day defined as strictly less than 12 hrs of light
# I got these definitions from this paper: https://www.pnas.org/doi/10.1073/pnas.1814015116
# 0h time points are assumed to occur just after treatment
# recovery samples are placed into category associated with the recovery state
# PEG was counted as drought treatment
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA319904")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA590939")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA660263")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA657867" & grepl("NoBact_Mock",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA680241")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA525303" & grepl("CTRL_wo_ETOH",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA348676" & grepl("mock",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB19974" & grepl("Mock|--",metaData$Treatment))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB15161" & grepl("_C",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA300543" & grepl("Uncut",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA666631" & grepl("WT",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA475542")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA187928")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA382583")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA272491" & grepl("Mock",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA657868" & grepl("NoBact.*_Mock",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA314076")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA341504")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA434094" & grepl("Lab LD",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA245231" & grepl("control|mock",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA494179" & grepl("silique|leaf|inflorescence",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA508214")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA640389")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA478998" & grepl("ctrl", metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA544417")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA344545" & grepl("22", metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA275437")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA575019" & grepl("CT", metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA588176" & grepl("Control", metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA320878" & grepl("white light", metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB32665" & grepl("pollen",metaData$Tissue))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA529796")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA602947" & grepl("no_peptide",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA476945" & grepl("control",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA231088")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA284739")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA601788" & grepl("col_rep",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA705878" & grepl("mock",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB31312" & grepl("h2o",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA508267" & grepl("18|19|20|21|22|23|24|25|26",metaData$Treatment))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA598201" & grepl("CM",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA609704" & grepl("Control",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA626535" & grepl("WTG",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA496426")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA154687")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA186843")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA188075" & grepl("Long days",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA301162")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA310623" & grepl("control",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA310623" & grepl("control",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB8427" & grepl("_C",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA702187")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA279529" & grepl("warm",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA323943")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA339762" & grepl("Mock",metaData$Treatment))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA363056" & grepl("0h",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA391475" & grepl("m",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA411947" & grepl("control",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA421355")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA445300" & grepl("Mock",metaData$Treatment))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA525816" & grepl("Mock",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA574854" & grepl("_21_",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA574855" & grepl("_21_",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA592037" & grepl("control|Ctrl",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA639497" & grepl("PS0|S0|PH0",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA658687" & grepl("light",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA661316" & grepl("mock MS",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA476778" & grepl("22C",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA406971")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA566413" & grepl("ZT",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA433501" & grepl("col_RNA|Poly",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA268115")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA271089")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA369750"  & grepl("freshly harvested|dry seed",metaData$LibraryName))] = "PECO:0001062 control exposure" # not sure about labeling dry seed as control
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA513852" & grepl("_20_",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "SRP000935" & grepl("Control|control",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA294703")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA315570" & grepl("22c",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA342309")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA497883" & grepl("control",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA565160" & grepl("h0",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA703036" & grepl(" 23| 25",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB25745")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA261430" & grepl("ctrl",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA261433")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA274887" & grepl("ctrl",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA294313" & grepl("control",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA302003" & grepl("mock",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA345276" & grepl("Control|h",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA445392")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA472051" & grepl("Mock",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA596803" & grepl("_NT_",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA639499" & grepl("_0",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA646744" & grepl("Mock",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA272425" & grepl("mock",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA525452" & grepl("0h",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA561790" & grepl("_0h",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB24412" & grepl("23C\\-rep|23C\\-expA\\-rep",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA483528" & grepl("WW",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA526082" & grepl("NHS",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA543995" & grepl("standard light",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA635147" & grepl("H2O|DRY_OH",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA717244" & grepl("At_",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA474441" & grepl("MOCK",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA488799" & grepl("0 Control",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA494693" & grepl("^Aerated|Second Aerated",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA511671" & grepl("WW",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA624145")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA285810" & grepl("Mock",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA378128" & grepl("control|WT$|untreated",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA387601")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA471243" & grepl("control",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA523174" & grepl("Control",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA593027" & grepl("0h",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA601157" & grepl("Only Plant-N",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA603477" & grepl("_0_",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA607298" & grepl("Mock_Uncut",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA608657" & grepl("WT_shoot",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA659897" & grepl("no.bacteria",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA661303" & grepl("Mock",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA704067" & grepl("Dry|Imbibed",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA714877" & grepl("C_",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA287644" & grepl("\\.0\\.",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA392864" & grepl("PDA",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA492842" & grepl("TSA",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA493021" & grepl("time0",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA504886")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA301696" & grepl("NC",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA473032" & grepl("cont",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA481217" & grepl("ck",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA505965" & grepl("C|Zn3|NP3",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA521623" & grepl(" M",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA722743" & grepl("Water|Non-treated",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA297270" & grepl("Mock",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA451508")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA507243")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA535479" & grepl("NV|T10",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA151357")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA416120" & grepl("warm",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA416120" & grepl("warm",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA478084")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJDB6957" & grepl("-F",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB28610" & grepl("light-23",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA210438" & grepl("_0_",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA280870")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA295091" & grepl("c-",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA321410" & grepl("Control",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA325245" & grepl("23C_Mock",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA339032")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA380494" & grepl("minus",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA380922" & grepl("Control",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA407271" & grepl("21",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA419538" & grepl("untreated",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA436778")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA477336" & grepl("days",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA480911" & grepl("control",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA555093" & grepl("CT",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA558807")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA576144" & grepl("C",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA592062" & grepl("H2O \\+ H2O",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA663969" & grepl("C",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA232621")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB33240")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA213635" & grepl("Control|Tissue",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA306014")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA353660" & grepl("BG11|water at 0h",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA512644")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB24752")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA325158")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA419694" & grepl("C",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA430092")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA450539" & grepl("Mock",metaData$LibraryName))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA485134" & grepl("Control",metaData$Experiment.Title))] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA557609")] = "PECO:0001062 control exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA489360" & grepl("Mock",metaData$LibraryName))] = "PECO:0001062 control exposure"

### ABIOTIC ################################################################
# 1-N-naphthylphthalamic acid
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA489360" & grepl("NPA",metaData$LibraryName))] = "PECO:0007189 chemical exposure, 1-N-naphthylphthalamic acid"

# anchorene
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA489360" & grepl("Anchorene",metaData$LibraryName))] = "PECO:0007189 chemical exposure, Anchorene"

# antibiotics
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA450539" & grepl("Carbenicillin",metaData$LibraryName))] = "PECO:0007041 antibiotic exposure, Carbenicillin"

# boric acid
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA663969" & grepl("B",metaData$LibraryName))] = "PECO:0007540 boric acid exposure"

# super fifty
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA592062" & grepl("SF \\+ H2O",metaData$LibraryName))] = "PECO:0007189 chemical exposure, super fifty"

# diacetyl
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA576144" & grepl("Dione",metaData$LibraryName))] = "PECO:0007189 chemical exposure, diacetyl"

# auxin
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA553702" & grepl("NAA",metaData$LibraryName))] = "PECO:0007074 auxin exposure"

# palmitoyl-CoA
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA380922" & grepl("C16:0",metaData$LibraryName))] = "PECO:0007189 chemical exposure, palmitoyl-CoA"

# C18:0-CoA
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA380922" & grepl("C18:0",metaData$LibraryName))] = "PECO:0007189 chemical exposure, C18:0-CoA"

# oleoyl-CoA
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA380922" & grepl("C18:1",metaData$LibraryName))] = "PECO:0007189 chemical exposure, oleoyl-CoA"

# radiation
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA380494" & grepl("plus",metaData$LibraryName))] = "PECO:0001039 radiation mutagen exposure"

# benzothiadiazole
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA325245" & grepl("23C_BTH",metaData$LibraryName))] = "PECO:0001036 chemical stress exposure, benzothiadiazole"

# oxidative stress, CysNO
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA321410" & grepl("CysNO",metaData$LibraryName))] = "PECO:0001037 oxidative stress exposure, CysNO"

# BHTC
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA297270" & grepl("BHTC",metaData$LibraryName))] = "PECO:0007189 chemical exposure, (2-(5-bromo-2-hydroxy-phenyl)-thiazolidine-4-carboxylic acid)"

# hydrogen peroxide
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA639496" & grepl(" H2O2 ",metaData$Experiment.Title))] = "PECO:0007415 hydrogen peroxide exposure"

# rose bengal
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA639495" & grepl("RB R",metaData$Experiment.Title))] = "PECO:0007189 chemical exposure, rose bengal"

# GR244DO
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA521623" & grepl(" G",metaData$Experiment.Title))] = "PECO:0007189 chemical exposure, GR244DO"

# nanoparticles
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA505965" & grepl("NP-",metaData$LibraryName))] = "PECO:0007189 chemical exposure, nanoparticles"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA722743" & grepl("SWNT",metaData$Experiment.Title))] = "PECO:0007189 chemical exposure, nanoparticles"

# zinc
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA505965" & grepl("Zn-",metaData$LibraryName))] = "PECO:0007309 zinc nutrient exposure"

# serotonin
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA481217" & grepl("S",metaData$LibraryName))] = "PECO:0007189 chemical exposure, serotonin"

# melatonin
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA481217" & grepl("M",metaData$LibraryName))] = "PECO:0007189 chemical exposure, melatonin"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA553702" & grepl("MEL",metaData$LibraryName))] = "PECO:0007189 chemical exposure, melatonin"

# touch
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA473032" & grepl("touch",metaData$LibraryName))] = "PECO:0007316 physical plant exposure, touch"

# potassium chloride
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA563066" & grepl("KCl",metaData$LibraryName))] = "PECO:0001065 potassium chloride exposure"

# VOC
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA392864" & grepl("RS",metaData$LibraryName))] = "PECO:0007189 chemical exposure, fungal volatile organic compounds"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA492842" & grepl("EC",metaData$LibraryName))] = "PECO:0007189 chemical exposure, bacteria volatile organic compounds"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA576144" & grepl("5T_",metaData$LibraryName))] = "PECO:0007189 chemical exposure, bacteria volatile organic compounds"

# estrogen
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA287644" & grepl("60|120",metaData$LibraryName))] = "PECO:0007189 chemical exposure, estrogen"

# KMBA
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA608657" & grepl("KMBA",metaData$Experiment.Title))] = "PECO:0007189 chemical exposure, 2-keto-4-methylthiobutyric acid"

# L-kynurenine
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA607298" & grepl("Lkyn_Uncut",metaData$Experiment.Title))] = "PECO:0007189 chemical exposure, L-kynurenine"

# jasmonic acid
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA523174" & grepl("JA",metaData$Experiment.Title))] = "PECO:0007080 jasmonic acid exposure"

# magnesium exposure
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA395664" & grepl("MgCl2",metaData$LibraryName))] = "PECO:0007288 magnesium nutrient exposure"

# oligogalacturonides
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA285810" & grepl("Trimers|OG",metaData$LibraryName))] = "PECO:0007189 chemical exposure, oligogalacturonides"

# limited carbon
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA494693" & grepl("Non|Micropore",metaData$LibraryName))] = "PECO:0001044 limited carbon exposure"

# paclobutrazol
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA474441" & grepl("PAC$",metaData$LibraryName))] = "PECO:0007189 chemical exposure, paclobutrazol"

# ethylene
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA635147" & grepl("C2H4",metaData$Experiment.Title))] = "PECO:0007068 ethylene exposure"

# UV light exposure
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA272425" & grepl("UV",metaData$LibraryName))] = "PECO:0007222 UV light exposure"

# phosphate buffered saline (PBS)
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA667430" & grepl("PBS",metaData$Experiment.Title))] = "PECO:0007189 chemical exposure, phosphate buffered saline"

# tenoxicam 
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA646744" & grepl("Col\\-0_TNX",metaData$Experiment.Title))] = "PECO:0007189 chemical exposure, tenoxicam"

# SHAM
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA639498")] = "PECO:0001037 oxidative stress exposure, salicylhdroxamic acid"

# aluminum
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA624317" & grepl("Al",metaData$Experiment.Title))] = "PECO:0007306 aluminum nutrient exposure"

# N-acylethanolamines
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA589109" & grepl("NAE",metaData$Experiment.Title))] = "PECO:0007189 chemical exposure, N-acylethanolamines"

# chitin
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA302003" & grepl("chitin",metaData$LibraryName))] = "chitin exposure"

# ozone
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA261430" & grepl("O3",metaData$Experiment.Title))] = "PECO:0001021 ozone exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA274887" & grepl("ozone",metaData$Experiment.Title))] = "PECO:0001021 ozone exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA480911" & grepl("ozone",metaData$Experiment.Title))] = "PECO:0001021 ozone exposure"

# UV-B exposure
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA716358" & grepl(" 3K | 30K ",metaData$Experiment.Title))] = "PECO:0007001 UV-B light exposure"

# indole 3 carbinol
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA672150" & grepl("_200_|_400_",metaData$Experiment.Title))] = "PECO:0001036 chemical stress exposure, Indole-3-carbinol"

# stratification exposure
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA369750"  & grepl("of stratification",metaData$LibraryName))] = "PECO:0001035 stratification exposure"

# cordycepin
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA433501" & grepl("Degradation",metaData$LibraryName))] = "PECO:0001036 chemical stress exposure, cordycepin"

# iron exposure
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA315374")] = "PECO:0007242 iron nutrient exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA624317" & grepl("Fe",metaData$Experiment.Title))] = "PECO:0007242 iron nutrient exposure"

# standing water
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA310623" & grepl("submergence",metaData$Experiment.Title))] = "PECO:0007282 standing water exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA310623" & grepl("submergence",metaData$LibraryName))] = "PECO:0007282 standing water exposure"

# mannitol
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA609704" & grepl(" Mannitol_1| Mannitol_2| Mannitol_3",metaData$Experiment.Title))] = "PECO:0001038 osmotic stress exposure, mannitol"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA471243" & grepl("osmotic",metaData$LibraryName))] = "PECO:0001038 osmotic stress exposure, mannitol"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA661303" & grepl("Man",metaData$Experiment.Title))] = "PECO:0001038 osmotic stress exposure, mannitol"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA213635" & grepl("Drought",metaData$LibraryName))] = "PECO:0001038 osmotic stress exposure, mannitol"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA419694" & grepl("M",metaData$LibraryName))] = "PECO:0001038 osmotic stress exposure, mannitol"


# thiamin exposure
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA368967" & grepl("30nM|10uM|Th2|ler",metaData$LibraryName))] = "PECO:0007241 plant nutrient exposure, thiamin exposure"

# limited thiamin
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA368967" & grepl("no",metaData$LibraryName))] = "PECO:0007241 plant nutrient exposure, limited thiamin"

# sulfur nutrient exposure
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA662710" & grepl("\\+",metaData$Experiment.Title))] = "PECO:0007295 sulfur nutrient exposure"

# limited sulfur
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA662710" & grepl("C-",metaData$Experiment.Title))] = "PECO:0001052 limited sulfur exposure"

# DMBQ
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA527966" & grepl("_D_",metaData$Experiment.Title))] = "PECO:0001003 organic chemical exposure, DMBQ"

# light
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA601788" & grepl("light",metaData$Experiment.Title))] = "PECO:0007196 light exposure" # not sure how to classify these
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA610701" & grepl("White Light",metaData$Experiment.Title))] = "PECO:0007196 light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA446045" & grepl("LIGHT",metaData$LibraryName))] = "PECO:0007196 light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA549286" & grepl("15min|30min|1h|3h|6h|12h",metaData$LibraryName))] = "PECO:0007196 light exposure"

# salt
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA476945" & grepl("NaCl",metaData$LibraryName))] = "PECO:0007048 sodium chloride exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA609704" & grepl("Salt_1|Salt_2|Salt_3",metaData$Experiment.Title))] = "PECO:0007048 sodium chloride exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA661316" & grepl("mock NaCl",metaData$Experiment.Title))] = "PECO:0007048 sodium chloride exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA513852" & grepl("NaCl",metaData$Experiment.Title))] = "PECO:0007048 sodium chloride exposure"
metaData$TreatmentOntology[(metaData$BioProject == "SRP000935" & grepl("Salt|salt",metaData$LibraryName))] = "PECO:0007048 sodium chloride exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA561790" & grepl("_6h|_24h|_72h",metaData$Experiment.Title))] = "PECO:0007048 sodium chloride exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA704067" & grepl("Salt",metaData$Experiment.Title))] = "PECO:0007048 sodium chloride exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA295091" & grepl("s-",metaData$LibraryName))] = "PECO:0007048 sodium chloride exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA213635" & grepl("Salt",metaData$LibraryName))] = "PECO:0007048 sodium chloride exposure"

# DMSO
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA611770" & grepl("DMSO",metaData$Experiment.Title))] = "PECO:0007189 chemical exposure, DMSO"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA527966" & grepl("_M_",metaData$Experiment.Title))] = "PECO:0007189 chemical exposure, DMSO"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA394661" & grepl("DMSO",metaData$LibraryName))] = "PECO:0007189 chemical exposure, DMSO"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA672150" & grepl("NT",metaData$Experiment.Title))] = "PECO:0007189 chemical exposure, DMSO"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA589109" & grepl("DMSO",metaData$Experiment.Title))] = "PECO:0007189 chemical exposure, DMSO"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA434180" & grepl("DMSO",metaData$LibraryName))] = "PECO:0007189 chemical exposure, DMSO"

# azetidine-2-carboxylate
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA391475" & grepl("_a",metaData$LibraryName))] = "PECO:0001036 chemical stress exposure, azetidine-2-carboxylate"

# dithiothreitol
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA391475" & grepl("_d",metaData$LibraryName))] = "PECO:0001036 chemical stress exposure, dithiothreitol"

# tunicamycin
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA611770" & grepl("TM",metaData$Experiment.Title))] = "PECO:0001036 chemical stress exposure, tunicamycin"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA391475" & grepl("_t",metaData$LibraryName))] = "PECO:0001036 chemical stress exposure, tunicamycin"

# blue light exposure
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA610701" & grepl("Blue Light",metaData$Experiment.Title))] = "PECO:0007218 blue light exposure"

# red light exposure
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA256037" & grepl("w", metaData$LibraryName))] = "PECO:0007207 red light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA610701" & grepl("Red Light",metaData$Experiment.Title))] = "PECO:0007207 red light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA750922" & grepl("Red light",metaData$Experiment.Title))] = "PECO:0007207 red light exposure"

# far red exposure
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA320878" & grepl("supplemental", metaData$LibraryName))] = "PECO:0007203 far red light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA256037" & grepl("s", metaData$LibraryName))] = "PECO:0007203 far red light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA610701" & grepl("Far-red Light",metaData$Experiment.Title))] = "PECO:0007203 far red light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA512107")] = "PECO:0007203 far red light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJDB6957" & grepl("\\+F",metaData$LibraryName))] = "PECO:0007203 far red light exposure"

# magnesium deficiency
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA588176" & grepl("-Mg", metaData$Experiment.Title))] = "PECO:0001049 limited magnesium exposure"

# heat
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA575019" & grepl("Local HS |Local HS_|Systemic HS_",metaData$Experiment.Title))] = "PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA508267" & grepl("28|30",metaData$Treatment))] = "PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA609704" & grepl(" Heat_",metaData$Experiment.Title))] = "PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA363056" & grepl("3h",metaData$LibraryName))] = "PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA574854" & grepl("_27_",metaData$Experiment.Title))] = "PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA574855" & grepl("_27_",metaData$Experiment.Title))] = "PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA476778" & grepl("37C",metaData$LibraryName))] = "PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA324514" & grepl("Heat",metaData$LibraryName))] = "PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA446045" & grepl("HEAT",metaData$LibraryName))] = "PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA510535" & grepl("27",metaData$Experiment.Title))] = "PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "SRP000935" & grepl("Heat|heat",metaData$LibraryName))] = "PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA315570" & grepl("27c|37c",metaData$LibraryName))] = "PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA497883" & grepl("heat",metaData$LibraryName))] = "PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA703036" & grepl(" 27",metaData$Experiment.Title))] = "PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA345276" & grepl("HS",metaData$LibraryName))] = "PECO:0007319 warm/hot air temperature exposure" # plates were submerged in water bath
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB24412" & grepl("27C\\-rep|27C\\-expA\\-rep",metaData$LibraryName))] = "PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA526082" & grepl("^HS",metaData$LibraryName))] = "PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA488799" & grepl("0 Heat",metaData$LibraryName))] = "PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA601157" & grepl("Only Plant-44|Only Plant-37",metaData$Experiment.Title))] = "PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA295091" & grepl(" h-",metaData$LibraryName))] = "PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA325245" & grepl("30C_Mock",metaData$LibraryName))] = "PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA407271" & grepl("27",metaData$LibraryName))] = "PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA555093" & grepl(" HS",metaData$LibraryName))] = "PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA213635" & grepl("Heat",metaData$LibraryName))] = "PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA485134" & grepl("Warming|Heat",metaData$Experiment.Title))] = "PECO:0007319 warm/hot air temperature exposure"

# dexamethasone
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA258547" )] = "PECO:0001003 organic chemical exposure, dexamethasone"

# 1-aminocyclopropane-1-carboxylic acid (ACC)
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA478998" & grepl("ACC", metaData$LibraryName))] = "PECO:0007165 growth hormone exposure, ACC"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA608657" & grepl("ACC", metaData$Experiment.Title))] = "PECO:0007165 growth hormone exposure, ACC"

# indole acetic acid (IAA)
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA478998" & grepl("IAA", metaData$LibraryName))] = "PECO:0007070 indole acetic acid exposure"

# drought
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB15161" & grepl("_S",metaData$LibraryName))] = "PECO:0007404 drought environment exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA315516" & grepl("Drought main treatment;",metaData$LibraryName))] = "PECO:0007404 drought environment exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA494179" & grepl("drought",metaData$Experiment.Title))] = "PECO:0007404 drought environment exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB8427" & grepl("_S",metaData$LibraryName))] = "PECO:0007404 drought environment exposure"
metaData$TreatmentOntology[(metaData$BioProject == "SRP000935" & grepl("Drought|drought",metaData$LibraryName))] = "PECO:0007404 drought environment exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA639499" & grepl("_1|_2|_3",metaData$Experiment.Title))] = "PECO:0007404 drought environment exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA483528" & grepl("WD",metaData$LibraryName))] = "PECO:0007404 drought environment exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA511671" & grepl("DT",metaData$LibraryName))] = "PECO:0007404 drought environment exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA301696" & grepl("Dehy",metaData$LibraryName))] = "PECO:0007404 drought environment exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA592062" & grepl("H2O \\+ Drought",metaData$LibraryName))] = "PECO:0007404 drought environment exposure"

# coronatine
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA245231" & grepl("coronatine",metaData$LibraryName))] = "PECO:0007080 jasmonic acid exposure, coronatine"

# greenhouse
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA693252")] = "PECO:0007248 greenhouse study"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA692747")] = "PECO:0007248 greenhouse study"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA691326")] = "PECO:0007248 greenhouse study"

# diphenyleneiodonium
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA481718" & grepl("DPI_Control",metaData$LibraryName))] = "PECO:0007200 short day length exposure,  PECO:0001003 organic chemical exposure, diphenyleneiodonium"

# continuous light
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB31094" & grepl("^C",metaData$LibraryName))] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA524697")] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA302786")] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB32665" & !grepl("pollen",metaData$Tissue))] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA311764" & grepl("flower|cotyledon|\\.0Gy",metaData$LibraryName))] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA622644" & grepl(" Control ", metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA546251" & grepl("WL", metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA188075" & grepl("LL",metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA513856" & grepl("0h|21C mock",metaData$LibraryName))] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA377574" & grepl("Ground Control",metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA377717" & grepl("Ground Control",metaData$LibraryName))] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA486575" & grepl("Ground control",metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA375085" & grepl("Light grown, Ground Control",metaData$LibraryName))] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA525432" & grepl("control",metaData$LibraryName))] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA609342" & grepl("WL",metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA566413" & grepl("CT",metaData$LibraryName))] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA369750"  & grepl("countinuous light",metaData$LibraryName))] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA666436")] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA354615" & grepl("mock|Mock", metaData$LibraryName))] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA435737" & grepl("GL", metaData$LibraryName))] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA685167" & grepl("PplusFeplus",metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB32446")] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA258161")] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA322265" & grepl("No Treatment",metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA475464")] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA667059" & grepl(" 22 ",metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA488799" & grepl("HH Control",metaData$LibraryName))] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA267681" & grepl("control",metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA422982" & grepl("DAC",metaData$LibraryName))] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA684572" & grepl("control",metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA286779" & grepl("D",metaData$LibraryName))] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA297861")] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA415015" & grepl("Control",metaData$LibraryName))] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA259680" & grepl("Contr",metaData$LibraryName))] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA271595")] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA487166" & grepl("21C mock",metaData$LibraryName))] = "PECO:0007162 continuous light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA750922" & grepl("White light",metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure"

# exogenous CIF2 treatment
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA666631" & grepl("Wtcif",metaData$Experiment.Title))] = "PECO:0001003 organic chemical exposure, CIF2"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA602947" & grepl("CIF2",metaData$LibraryName))] = "PECO:0001003 organic chemical exposure, CIF2"

# short day
# 8 hr days usually, but also less than 12 hours of light included
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA232536" & grepl("H2O",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA481720" & grepl("Control",metaData$LibraryName))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA315516" & grepl("Control",metaData$LibraryName))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA315516" & grepl("Mock for B.cinerea pre-treatment;",metaData$LibraryName))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA315516" & grepl("Mock for B.cinerea main treatment;",metaData$LibraryName))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA564095")] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA224705" & grepl("water",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA481718" & grepl("H2O_Control",metaData$LibraryName))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA419306" & grepl("Intact",metaData$LibraryName))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA635635" & grepl("NB",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA321559" & grepl("Col-0 0h rep|Bur-0 0h rep",metaData$LibraryName))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA378644" & grepl("col.*22c|col_22",metaData$LibraryName))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA494862" & grepl("mock",metaData$SampleName))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB33339" & grepl("CON",metaData$LibraryName))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA476754" & grepl("_M_",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure" # 10 hrs of light
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA412447" & grepl("CT",metaData$LibraryName))] = "PECO:0007200 short day length exposure" # 10 hrs of light
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB26953" & grepl("mock",metaData$LibraryName))] = "PECO:0007200 short day length exposure" # 8.5 hrs of light
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB34520" & grepl("mock|UNT",metaData$LibraryName))] = "PECO:0007200 short day length exposure" # 10 hrs of light
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA813448" & grepl("MOCK",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure" # 9 hrs of light
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA549850" & grepl("H2O",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA481719" & grepl("_H2O_",metaData$LibraryName))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA590705")] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA209363" & grepl("NT",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA281364" & grepl("L3",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA666544" & grepl("Mg",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA449619" & grepl("mock",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJDB5141" & grepl("Ctrl",metaData$LibraryName))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB31314")] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA271131" & grepl("AC",metaData$LibraryName))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA338071" & grepl("25C",metaData$LibraryName))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA542115" & grepl("Col.*Control",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA706054" & grepl("_21",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB31803" & grepl("ctrl",metaData$LibraryName))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA319370" & grepl("mock",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB27944" & grepl("_c_",metaData$LibraryName))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA354369" & grepl("Mock",metaData$LibraryName))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA809747" & grepl("CK_22|CK_25",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA384110" & grepl("22C",metaData$LibraryName))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB31311" & grepl("-0min",metaData$LibraryName))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB31337" & grepl("h2o",metaData$LibraryName))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB6701" & grepl("_C_",metaData$LibraryName))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA477336" & grepl("0day",metaData$LibraryName))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA681187" & grepl("Light_20C",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB23627" & grepl("Mock",metaData$Treatment))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA274954" & grepl("23",metaData$LibraryName))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJDB7233" & grepl("control",metaData$LibraryName))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA622692" & grepl("control",metaData$LibraryName))] = "PECO:0007200 short day length exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA224133" & grepl("mock$",metaData$LibraryName))] = "PECO:0007200 short day length exposure"

# natural long day
# 16 hrs
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA434094" & grepl("natural LD",metaData$LibraryName))] = "PECO:0007256 field study, PECO:0007199 long day length exposure"

# high light
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA481720" & grepl("Highlight",metaData$LibraryName))] = "PECO:0007075 high light intensity exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA575019" & grepl("Local HL |Local HL_|Systemic HL_",metaData$Experiment.Title))] = "PECO:0007075 high light intensity exposure"
metaData$TreatmentOntology[(metaData$BioProject == "SRP000935" & grepl("Highlight|highlight",metaData$LibraryName))] = "PECO:0007075 high light intensity exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA543995" & grepl("high light",metaData$Experiment.Title))] = "PECO:0007075 high light intensity exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA555093" & grepl("HL-",metaData$LibraryName))] = "PECO:0007075 high light intensity exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA259680" & grepl("HL",metaData$LibraryName))] = "PECO:0007075 high light intensity exposure"

# nitrogen
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA422118" & grepl("10mM",metaData$Experiment.Title))] = "PECO:0007284 nitrogen macronutrient exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA382136" & grepl("N",metaData$LibraryName))] = "PECO:0007284 nitrogen macronutrient exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA638728" & grepl("hn",metaData$Experiment.Title))] = "PECO:0007284 nitrogen macronutrient exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA681183" & grepl("_2KN",metaData$Experiment.Title))] = "PECO:0007284 nitrogen macronutrient exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA563066" & grepl("KNO3",metaData$LibraryName))] = "PECO:0007284 nitrogen macronutrient exposure"

# limited nitrogen
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA422118" & grepl("1mM",metaData$Experiment.Title))] = "PECO:0001050 limited nitrogen exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA382136" & grepl("C",metaData$LibraryName))] = "PECO:0001050 limited nitrogen exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA382136" & grepl("S0|R0",metaData$LibraryName))] = "PECO:0001050 limited nitrogen exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA638728" & grepl("ln",metaData$Experiment.Title))] = "PECO:0001050 limited nitrogen exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA681183" & grepl("02KN",metaData$Experiment.Title))] = "PECO:0001050 limited nitrogen exposure"

# mechanical damage, wounding
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA300543" & !grepl("Uncut",metaData$LibraryName))] = "PECO:0007373 mechanical damage exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA324514" & grepl("Wounding",metaData$LibraryName))] = "PECO:0007373 mechanical damage exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA672150" & grepl("_w_",metaData$Experiment.Title))] = "PECO:0007373 mechanical damage exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA607298" & grepl("Mock_Cut",metaData$Experiment.Title))] = "PECO:0007373 mechanical damage exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA493021" & grepl("10min|30min|1h|2h|4h|8h|12h",metaData$LibraryName))] = "PECO:0007373 mechanical damage exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA489360" & grepl("RAM",metaData$LibraryName))] = "PECO:0007373 mechanical damage exposure"

# dark
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA375085" & grepl("Dark grown, Ground Control",metaData$LibraryName))] = "PECO:0007270 continuous dark (no light) exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA551415" & grepl("JA_0hour",metaData$LibraryName))] = "PECO:0007270 continuous dark (no light) exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA601788" & grepl("dark",metaData$Experiment.Title))] = "PECO:0007270 continuous dark (no light) exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA281364" & grepl("0h|D",metaData$Experiment.Title))] = "PECO:0007270 continuous dark (no light) exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA658687" & grepl("dark",metaData$Experiment.Title))] = "PECO:0007270 continuous dark (no light) exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA610701" & grepl("Dark",metaData$Experiment.Title))] = "PECO:0007270 continuous dark (no light) exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA446045" & grepl("DARK",metaData$LibraryName))] = "PECO:0007270 continuous dark (no light) exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA549286" & grepl("Dark",metaData$LibraryName))] = "PECO:0007270 continuous dark (no light) exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA716358" & grepl(" 0K ",metaData$Experiment.Title))] = "PECO:0007270 continuous dark (no light) exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA543995" & grepl("dark",metaData$Experiment.Title))] = "PECO:0007270 continuous dark (no light) exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA609261")] = "PECO:0007270 continuous dark (no light) exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA735204" & grepl("22C",metaData$Experiment.Title))] = "PECO:0007270 continuous dark (no light) exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB28610" & grepl("dark-23",metaData$LibraryName))] = "PECO:0007270 continuous dark (no light) exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA750922" & grepl("Darkness \\(D",metaData$Experiment.Title))] = "PECO:0007270 continuous dark (no light) exposure"

# limited phosphate
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA344452" & grepl("LowP_noBact", metaData$Experiment.Title))] = "PECO:0001046 limited phosphate exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA344451" & grepl("LowP_noBact", metaData$Experiment.Title))] = "PECO:0001046 limited phosphate exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA344453" & grepl("LowP_noTreatment", metaData$Experiment.Title))] = "PECO:0001046 limited phosphate exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA478998" & grepl("lowPi", metaData$LibraryName))] = "PECO:0001046 limited phosphate exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA435505" & grepl("minus", metaData$LibraryName))] = "PECO:0001046 limited phosphate exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA624317" & grepl("\\-Pi",metaData$Experiment.Title))] = "PECO:0001046 limited phosphate exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA470732" & grepl("limited",metaData$LibraryName))] = "PECO:0001046 limited phosphate exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA542492" & grepl("5Pi_R",metaData$LibraryName))] = "PECO:0001046 limited phosphate exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA308080" & grepl("Pi\\-",metaData$LibraryName))] = "PECO:0001046 limited phosphate exposure"

# phosphate exposure
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA397098" & grepl("No Bacteria",metaData$LibraryName))] = "PECO:0007397 phosphorus nutrient exposure" # germination in presence or absence of phosphate ignored
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA344452" & grepl("HighP_noBact", metaData$Experiment.Title))] = "PECO:0007397 phosphorus nutrient exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA344451" & grepl("HighP_noBact", metaData$Experiment.Title))] = "PECO:0007397 phosphorus nutrient exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA344453" & grepl("HighP_noTreatment", metaData$Experiment.Title))] = "PECO:0007397 phosphorus nutrient exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA478998" & grepl("HighPi", metaData$LibraryName))] = "PECO:0007397 phosphorus nutrient exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA435505" & grepl("plus", metaData$LibraryName))] = "PECO:0007397 phosphorus nutrient exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA624317" & grepl("\\+Pi",metaData$Experiment.Title))] = "PECO:0007397 phosphorus nutrient exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA470732" & grepl("replete|resupplied",metaData$LibraryName))] = "PECO:0007397 phosphorus nutrient exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA542492" & grepl("^Pi_R",metaData$LibraryName))] = "PECO:0007397 phosphorus nutrient exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA308080" & grepl("Pi\\+",metaData$LibraryName))] = "PECO:0007397 phosphorus nutrient exposure"

# t-chalcone
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA525303" & grepl("CHAL",metaData$LibraryName))] = "PECO:0001037 oxidative stress exposure, t-chalcone"

# ethanol exposure
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA525303" & grepl("ETOH_1H|ETOH_3H|ETOH_6H|ETOH_12H|ETOH_24H|w_ETOH",metaData$LibraryName))] = "PECO:0007189 chemical exposure, ethanol"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA553702" & grepl("EtOH",metaData$LibraryName))] = "PECO:0007189 chemical exposure, ethanol"

# Cold air temperature
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA237342")] = "PECO:0007332 cold air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB19974" & grepl("Cold",metaData$Treatment))] = "PECO:0007332 cold air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA494179" & grepl("cold",metaData$Experiment.Title))] = "PECO:0007332 cold air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA344545" & grepl("15",metaData$LibraryName))] = "PECO:0007332 cold air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA508267" & grepl("10|12|14|16",metaData$Treatment))] = "PECO:0007332 cold air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA279529" & grepl("cold",metaData$LibraryName))] = "PECO:0007332 cold air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA411947" & grepl("cold",metaData$LibraryName))] = "PECO:0007332 cold air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA445300" & grepl("cold",metaData$Treatment))] = "PECO:0007332 cold air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA592037" & grepl("Cold",metaData$LibraryName))] = "PECO:0007332 cold air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA476778" & grepl("10C",metaData$LibraryName))] = "PECO:0007332 cold air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA324514" & grepl("Cold",metaData$LibraryName))] = "PECO:0007332 cold air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA446045" & grepl("COLD",metaData$LibraryName))] = "PECO:0007332 cold air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA510535" & grepl("17",metaData$Experiment.Title))] = "PECO:0007332 cold air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA513852" & grepl("_4_",metaData$Experiment.Title))] = "PECO:0007332 cold air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "SRP000935" & grepl("Cold|cold",metaData$LibraryName))] = "PECO:0007332 cold air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA315570" & grepl("17c_0min_rep|17c_15min_rep|17c_1hr_rep|17c_4hr_rep",metaData$LibraryName))] = "PECO:0007332 cold air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA525452" & grepl("3h|24h",metaData$LibraryName))] = "PECO:0007332 cold air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB24412" & grepl("16C\\-rep|16C\\-expA\\-rep",metaData$LibraryName))] = "PECO:0007332 cold air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA593027" & grepl("4h",metaData$LibraryName))] = "PECO:0007332 cold air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA416120" & grepl("cold",metaData$Experiment.Title))] = "PECO:0007332 cold air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA416120" & grepl("cold",metaData$LibraryName))] = "PECO:0007332 cold air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB28610" & grepl("light-16",metaData$LibraryName))] = "PECO:0007332 cold air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA407271" & grepl("15",metaData$LibraryName))] = "PECO:0007332 cold air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA213635" & grepl("Cold",metaData$LibraryName))] = "PECO:0007332 cold air temperature exposure"

# space flight
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA626535" & grepl("F",metaData$Experiment.Title))] = "PECO:0007315 zero gravity exposure"

# flagellin 22 exposure
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA657867" & grepl("NoBact_flg22_",metaData$Experiment.Title))] = "flagellin 22 exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA313379" & grepl("Col-0",metaData$Experiment.Title))] = "flagellin 22 exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA657868" & grepl("NoBact.*_flg22_",metaData$Experiment.Title))] = "flagellin 22 exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA339762" & grepl("Hlg22",metaData$Treatment))] = "flagellin 22 exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA603477" & grepl("_10_|_30_|_60_",metaData$Experiment.Title))] = "flagellin 22 exposure"

# methyl jasmonate
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA478998" & grepl("meJA",metaData$LibraryName))] = "PECO:0007407 methyl jasmonate exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA378128" & grepl("MeJA$",metaData$LibraryName))] = "PECO:0007407 methyl jasmonate exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA210438" & grepl("_10_",metaData$LibraryName))] = "PECO:0007407 methyl jasmonate exposure"

# salicylic acid
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA646744" & grepl("SA_biorep",metaData$Experiment.Title))] = "PECO:0007187 salicylic acid exposure"

# absissic acid (ABA)
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA494179" & grepl("ABA",metaData$Experiment.Title))] = "PECO:0007105 abscisic acid exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA478998" & grepl("ABA",metaData$LibraryName))] = "PECO:0007105 abscisic acid exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA394661" & grepl("ABA",metaData$LibraryName))] = "PECO:0007105 abscisic acid exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA472051" & grepl("ABA",metaData$LibraryName))] = "PECO:0007105 abscisic acid exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA596803" & grepl("_ABA_",metaData$Experiment.Title))] = "PECO:0007105 abscisic acid exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA434180" & grepl("ABA",metaData$LibraryName))] = "PECO:0007105 abscisic acid exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA471243" & grepl("ABA",metaData$LibraryName))] = "PECO:0007105 abscisic acid exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA661303" & grepl("ABA",metaData$Experiment.Title))] = "PECO:0007105 abscisic acid exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA301696" & grepl("ABA",metaData$LibraryName))] = "PECO:0007105 abscisic acid exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA419694" & grepl("A",metaData$LibraryName))] = "PECO:0007105 abscisic acid exposure"

# AM1 fluorine derivatives
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA394661" & grepl("AMF",metaData$LibraryName))] = "PECO:0007189 chemical exposure, AMF"

### BIOTIC #################################################################
# chlorella
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA353660" & grepl("Chlorella at 0h",metaData$LibraryName))] = "PECO:0007005 Viridiplantae plant exposure, Chlorella fusca"

# spider mite egg extract
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA714877" & grepl("E_",metaData$Experiment.Title))] = "PECO:0007357 biotic plant exposure, spider mite egg extract exposure"

# Nematoda exposure
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA294313" & grepl("infected",metaData$LibraryName))] = "PECO:0007337 Nematoda exposure"

# Turnip mosaic virus
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA272491" & grepl("TuMV",metaData$Experiment.Title))] = "PECO:0007109 ssRNA positive-strand virus with no DNA stage exposure, Turnip mosaic virus"

# Pseudomonas syringae plant exposure
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA348676" & !grepl("mock",metaData$Experiment.Title))] = "PECO:0001034 Pseudomonas syringae plant exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA395664" & grepl("Rpm",metaData$LibraryName))] = "PECO:0001034 Pseudomonas syringae plant exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA353660" & grepl("water and DC3000",metaData$LibraryName))] = "PECO:0001034 Pseudomonas syringae plant exposure"

# Pseudomonas plant exposure
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA474441" & grepl("PAO$",metaData$LibraryName))] = "PECO:0007144 Pseudomonas plant exposure, Pseudomonas aeruginosa"

# insect attacks
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA488315")] = "PECO:0007333 insect plant exposure, PECO:0007256 field study"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA315516" & grepl("Mock Hor B.cinerea pre-treatment, Pieris main treatment",metaData$LibraryName))] = "PECO:0007114 Lepidoptera exposure, Pieris rapae"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA315516" & grepl("Mock for B.cinerea pre-treatment, Pieris main treatment,",metaData$LibraryName))] = "PECO:0007114 Lepidoptera exposure, Pieris rapae"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA315516" & grepl("P.rapae main treatment;",metaData$LibraryName))] = "PECO:0007114 Lepidoptera exposure, Pieris rapae"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA565160" & grepl("h1|h4",metaData$LibraryName))] = "PECO:0007114 Lepidoptera exposure, Trichoplusia ni"

# bacterial plant exposure
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA657867" & grepl("MF.*_Mock_",metaData$Experiment.Title))] = "PECO:0007246 bacterial plant exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA657867" & grepl("cl.*_Mock_",metaData$Experiment.Title))] = "PECO:0007246 bacterial plant exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA657867" & grepl("e.coli.*_Mock_",metaData$Experiment.Title))] = "PECO:0007246 bacterial plant exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA657867" & grepl("SynCom.*_Mock_",metaData$Experiment.Title))] = "PECO:0007246 bacterial plant exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA657868" & grepl("SynCom.*_Mock_",metaData$Experiment.Title))] = "PECO:0007246 bacterial plant exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA598201" & grepl("CE",metaData$Experiment.Title))] = "PECO:0007246 bacterial plant exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA661316" & grepl("SA187 MS",metaData$Experiment.Title))] = "PECO:0007246 bacterial plant exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA667430" & grepl("V. vulnificus",metaData$Experiment.Title))] = "PECO:0007246 bacterial plant exposure, Vibrio vulnificus"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA601157" & grepl("Plant\\+SA  187-N",metaData$Experiment.Title))] = "PECO:0007246 bacterial plant exposure, Enterobacter"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA608657" & grepl("SA187",metaData$Experiment.Title))] = "PECO:0007246 bacterial plant exposure, Enterobacter"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA659897" & grepl("SynCom",metaData$Experiment.Title))] = "PECO:0007246 bacterial plant exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA419538" & grepl(" treated",metaData$LibraryName))] = "PECO:0007246 bacterial plant exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA261035")] = "PECO:0007246 bacterial plant exposure, Vibrio vulnificus"

# Fungus
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA562704" & grepl("m1390",metaData$Experiment.Title))] = "PECO:0007233 fungal plant exposure, Hyaloperonospora arabidopsidis"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA705878" & grepl("inf",metaData$Experiment.Title))] = "PECO:0007233 fungal plant exposure, Fusarium oxysporum"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA525816" & grepl("Ec",metaData$LibraryName))] = "PECO:0007233 fungal plant exposure, Erysiphe cichoracearum"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA717244" & grepl("At\\-",metaData$Experiment.Title))] = "PECO:0007233 fungal plant exposure" # multiple fungal species here!

# metaData$TreatmentOntology[(metaData$BioProject == "PRJNA665689" & grepl("Verticillium",metaData$Experiment.Title))] = "PECO:0007233 fungal plant exposure, Verticillium longisporum"
# metaData$TreatmentOntology[(metaData$BioProject == "PRJNA665689" & grepl("Piriformospora",metaData$Experiment.Title))] = "PECO:0007233 fungal plant exposure, Serendipita indica"
# metaData$TreatmentOntology[(metaData$BioProject == "PRJNA665689" & grepl("Phytoph",metaData$Experiment.Title))] = "PECO:0007233 fungal plant exposure, Phytophtora parasitica"

### MULTIPLE TREATMENTS ####################################################
# short day + salicylic acid
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA224133" & grepl("SA",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007187 salicylic acid exposure"

# chlorella + pseudomonas syringae
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA353660" & grepl("Chlorella and DC3000",metaData$LibraryName))] = "PECO:0001034 Pseudomonas syringae plant exposure, PECO:0007005 Viridiplantae plant exposure, Chlorella fusca"

# short day + pipecolic acid
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB23627" & grepl("pipecolic acid",metaData$Treatment))] = "PECO:0007200 short day length exposure, PECO:0007189 chemical exposure, pipecolic acid"

# short day + heat + shade
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA681187" & grepl("Shade_28C",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, PECO:0007319 warm/hot air temperature exposure, PECO:0007078 light quantity exposure, shade"

# short day + shade
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA681187" & grepl("Shade_20C",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, PECO:0007078 light quantity exposure, shade"

# drought + super fifty
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA592062" & grepl("SF \\+ Drought",metaData$LibraryName))] = "PECO:0007404 drought environment exposure, PECO:0007189 chemical exposure, super fifty"

# continous light + cytokinin
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA415015" & grepl("BA",metaData$LibraryName))] = "PECO:0007162 continuous light exposure, PECO:0007066 cytokinin exposure, 6-benzyladenine"

# heat + benzothiadiazole
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA325245" & grepl("30C_BTH",metaData$LibraryName))] = "PECO:0007319 warm/hot air temperature exposure, PECO:0001036 chemical stress exposure, benzothiadiazole"

# continuous light + cytokinin
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA286779" & grepl("BA",metaData$LibraryName))] = "PECO:0007162 continuous light exposure, PECO:0007066 cytokinin exposure, 6-benzylaminopurine"

# dark + cold
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB28610" & grepl("dark-16",metaData$LibraryName))] = "PECO:0007270 continuous dark (no light) exposure, PECO:0007332 cold air temperature exposure"

# dark + hot + shoot inducing medium
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA735204" & grepl("27C.*SIM",metaData$Experiment.Title))] = "PECO:0007270 continuous dark (no light) exposure, PECO:0007319 warm/hot air temperature exposure, PECO:0007147 plant growth medium exposure, shoot-inducing media"

# dark + cold + shoot inducing medium
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA735204" & grepl("17C.*SIM",metaData$Experiment.Title))] = "PECO:0007270 continuous dark (no light) exposure, PECO:0007332 cold air temperature exposure, PECO:0007147 plant growth medium exposure, shoot-inducing media"

# dark + hot + callus inducing medium
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA735204" & grepl("27C.*CIM",metaData$Experiment.Title))] = "PECO:0007270 continuous dark (no light) exposure, PECO:0007319 warm/hot air temperature exposure, PECO:0007147 plant growth medium exposure, callus-inducing media"

# dark + cold + callus inducing medium
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA735204" & grepl("17C.*CIM",metaData$Experiment.Title))] = "PECO:0007270 continuous dark (no light) exposure, PECO:0007332 cold air temperature exposure, PECO:0007147 plant growth medium exposure, callus-inducing media"

# rose bengal + H2O2
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA639495" & grepl("RBH",metaData$Experiment.Title))] = "PECO:0007415 hydrogen peroxide exposure, PECO:0007189 chemical exposure, rose bengal"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA639496" & grepl("RBH",metaData$Experiment.Title))] = "PECO:0007415 hydrogen peroxide exposure, PECO:0007189 chemical exposure, rose bengal"

# continuous light + ABA + glucose
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA684572" & grepl("_ABA\\+glucose_",metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, PECO:0007105 abscisic acid exposure, PECO:0001013 glucose exposure"

# continuous light + glucose
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA684572" & grepl("_glucose_",metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, PECO:0001013 glucose exposure"

# continuous light + ABA
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA684572" & grepl("_ABA_",metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, PECO:0007105 abscisic acid exposure"

# short day + hot + fungus
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA809747" & grepl("Si_28",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, PECO:0007319 warm/hot air temperature exposure, PECO:0007233 fungal plant exposure, Serendipita indica"

# mechanical damage + L-kynurenine
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA607298" & grepl("Lkyn_Cut",metaData$Experiment.Title))] = "PECO:0007373 mechanical damage exposure, PECO:0007189 chemical exposure, L-kynurenine"

# limited phosphate + cis-zeatin
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA542492" & grepl("5Pi_cZ",metaData$LibraryName))] = "PECO:0001046 limited phosphate exposure, PECO:0007189 chemical exposure, cis-zeatin"

# limited phosphate + trans-zeatin
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA542492" & grepl("5Pi_tZ",metaData$LibraryName))] = "PECO:0001046 limited phosphate exposure, PECO:0007189 chemical exposure, trans-zeatin"

# phosphate + cis-zeatin
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA542492" & grepl("^Pi_cZ",metaData$LibraryName))] = "PECO:0007397 phosphorus nutrient exposure, PECO:0007189 chemical exposure, cis-zeatin"

# phosphate + trans-zeatin
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA542492" & grepl("^Pi_tZ",metaData$LibraryName))] = "PECO:0007397 phosphorus nutrient exposure, PECO:0007189 chemical exposure, trans-zeatin"

# continuous light + sucrose
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA422982" & grepl("0",metaData$LibraryName))] = "PECO:0007162 continuous light exposure, PECO:0001015 sucrose exposure"

# short day + benzothiadiazole
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA354369" & grepl("BTH",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0001036 chemical stress exposure, benzothiadiazole"

# short day + MeJA
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA354369" & grepl("MeJA",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007407 methyl jasmonate exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA224133" & grepl("MeJA",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007407 methyl jasmonate exposure"

# short day + aphid
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB27944" & grepl("hpi",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007333 insect plant exposure, aphid"

# short day + estradiol
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA319370" & grepl("estradiol",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, PECO:0007189 chemical exposure, estradiol"

# short day + copper
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA132271" & grepl("Copper-sufficient",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007261 copper nutrient exposure"

# gibberlic acid + bacteria
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA474441" & grepl("PAOG$",metaData$LibraryName))] = "PECO:0007069 gibberellic acid exposure, PECO:0007144 Pseudomonas plant exposure, Pseudomonas aeruginosa"

# gibberlic acid + paclobutrazol
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA474441" & grepl("PACG$",metaData$LibraryName))] = "PECO:0007069 gibberellic acid exposure, PECO:0007189 chemical exposure, paclobutrazol"

# salicylic acid + tenoxicam 
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA646744" & grepl("SA_TNX",metaData$Experiment.Title))] = "PECO:0007187 salicylic acid exposure, PECO:0007189 chemical exposure, tenoxicam"

# short day + heatin
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA542115" & grepl("Heatin.*Control",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, PECO:0007189 chemical exposure, Heatin"

# short day + hot air temp + heatin
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA542115" & grepl("Heatin.*hightemp",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, PECO:0007319 warm/hot air temperature exposure, PECO:0007189 chemical exposure, Heatin"

# continuous light + red light exposure
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA412889")] = "PECO:0007162 continuous light exposure, PECO:0007207 red light exposure"

# continuous light + MG132 proteasome inhibitor
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA322265" & grepl("hours",metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, MG132 exposure"

# short day + elevated carbon dioxide exposure + limited magnesium exposure
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA271131" & grepl("EL",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0001056 elevated carbon dioxide exposure, PECO:0001049 limited magnesium exposure"

# short day + elevated carbon dioxide exposure + magnesium exposure
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA271131" & grepl("EH",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0001056 elevated carbon dioxide exposure, PECO:0007288 magnesium nutrient exposure"

# short day + elevated carbon dioxide exposure
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA271131" & grepl("EC",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0001056 elevated carbon dioxide exposure"

# short day + magnesium exposure
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA271131" & grepl("AH",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007288 magnesium nutrient exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA643438" & grepl("Mg",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, PECO:0007288 magnesium nutrient exposure"

# continuous light + limited phosphate
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA685167" & grepl("PminusFeplus",metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, PECO:0001046 limited phosphate exposure"

# continuous light + limited iron
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA685167" & grepl("PplusFeminus",metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, PECO:0001047 limited iron exposure"

# continuous light + limited iron + limited phosphate
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA685167" & grepl("PminusFeminus",metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, PECO:0001047 limited iron exposure, PECO:0001046 limited phosphate exposure"

# short day + limited molybdenum
metaData$TreatmentOntology[(metaData$BioProject == "PRJDB5141" & grepl("Mo_",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0001048 limited molybdenum exposure"

# short day + limited manganese
metaData$TreatmentOntology[(metaData$BioProject == "PRJDB5141" & grepl("Mn_",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0001053 limited manganese exposure"

# short day + limited copper
metaData$TreatmentOntology[(metaData$BioProject == "PRJDB5141" & grepl("Cu_",metaData$LibraryName))] = "PECO:0007200 short day length exposure, limited copper exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA132271" & grepl("Copper-deficient",metaData$LibraryName))] = "PECO:0007200 short day length exposure, limited copper exposure"

# short day + limited zinc
metaData$TreatmentOntology[(metaData$BioProject == "PRJDB5141" & grepl("Zn_",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0001041 limited zinc exposure"

# short day + limited iron
metaData$TreatmentOntology[(metaData$BioProject == "PRJDB5141" & grepl("Fe_",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0001047 limited iron exposure"

# short day + limited magnesium
metaData$TreatmentOntology[(metaData$BioProject == "PRJDB5141" & grepl("Mg_",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0001049 limited magnesium exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA271131" & grepl("AL",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0001049 limited magnesium exposure"

# short day + limited boron
metaData$TreatmentOntology[(metaData$BioProject == "PRJDB5141" & grepl("B_",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0001042 limited boron exposure"

# short day + limited sulfur
metaData$TreatmentOntology[(metaData$BioProject == "PRJDB5141" & grepl("S_",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0001052 limited sulfur exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA622692" & grepl("Low-S",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0001052 limited sulfur exposure"

# short day + limited calcium
metaData$TreatmentOntology[(metaData$BioProject == "PRJDB5141" & grepl("Ca_",metaData$LibraryName))] = "PECO:0007200 short day length exposure, limited calcium exposure"

# short day + limited potassium
metaData$TreatmentOntology[(metaData$BioProject == "PRJDB5141" & grepl("K_",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0001051 limited potassium exposure"

# short day + limited nitrogen
metaData$TreatmentOntology[(metaData$BioProject == "PRJDB5141" & grepl("N_",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0001050 limited nitrogen exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJDB7233" & grepl("-N",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0001050 limited nitrogen exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA622692" & grepl("Low-N",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0001050 limited nitrogen exposure"

# continuous light + n-acetylethanolamine
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA554610" & grepl("NAE", metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, PECO:0007189 chemical exposure, N-acylethanolamines"

# continuous light + DMSO
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA554610" & grepl("DMSO", metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, PECO:0007189 chemical exposure, DMSO"

# continuous light + IAA
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA354615" & grepl("IAA", metaData$LibraryName))] = "PECO:0007162 continuous light exposure, PECO:0007070 indole acetic acid exposure"

# dark + ethanol
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA319318"  & grepl("EtOH",metaData$LibraryName))] = "PECO:0007270 continuous dark (no light) exposure, PECO:0007189 chemical exposure, ethanol"

# dark + ABA
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA319318"  & grepl("ABA",metaData$LibraryName))] = "PECO:0007270 continuous dark (no light) exposure, PECO:0007105 abscisic acid exposure"

# bacterial exposure + heat + benzothiazadole
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA551121"  & grepl("_B",metaData$Experiment.Title))] = "PECO:0007319 warm/hot air temperature exposure, PECO:0007246 bacterial plant exposure, Ralstonia solanacearum, PECO:0001036 chemical stress exposure, benzothiadiazole"

# bacterial exposure + heat
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA551121"  & grepl("_C",metaData$Experiment.Title))] = "PECO:0007319 warm/hot air temperature exposure, PECO:0007246 bacterial plant exposure, Ralstonia solanacearum"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA601157" & grepl("Plant\\+SA-187.*44",metaData$Experiment.Title))] = "PECO:0007319 warm/hot air temperature exposure, PECO:0007246 bacterial plant exposure, Enterobacter"

# bacterial exposure + sound wave exposure + heat
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA551121"  & grepl("_10",metaData$Experiment.Title))] = "PECO:0007319 warm/hot air temperature exposure, PECO:0007246 bacterial plant exposure, Ralstonia solanacearum, sound wave exposure"

# field + UV
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA481452" & grepl("UV0",metaData$Experiment.Title))] = "PECO:0007256 field study, PECO:0007222 UV light exposure"

# field + blue
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA481452" & grepl("blue",metaData$Experiment.Title))] = "PECO:0007256 field study, PECO:0007218 blue light exposure"

# field + UVA
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA481452" & grepl("UVA",metaData$Experiment.Title))] = "PECO:0007256 field study, PECO:0007002 UV-A light exposure"

# field + UVA + UVB
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA481452" & grepl("UV-A-B",metaData$Experiment.Title))] = "PECO:0007256 field study, PECO:0007002 UV-A light exposure, PECO:0007001 UV-B light exposure"

# salt + bacteria
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA661316" & grepl("SA187 NaCl",metaData$Experiment.Title))] = "PECO:0007246 bacterial plant exposure, PECO:0007048 sodium chloride exposure"

# drought + SHAM
# PEG is counted as drought
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA639497" & grepl("PS1|PS2|PS3",metaData$Experiment.Title))] = "PECO:0007404 drought environment exposure, PECO:0001037 oxidative stress exposure, salicylhdroxamic acid"

# continuous light + far red light + UVB
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA609342" & grepl("FRUV;", metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, PECO:0007203 far red light exposure, PECO:0007001 UV-B light exposure"

# continuous light + far red light
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA609342" & grepl("FR;", metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, PECO:0007203 far red light exposure"

# continuous light + drought
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA525432" & grepl("stress",metaData$LibraryName))] = "PECO:0007162 continuous light exposure, PECO:0007404 drought environment exposure"

# continuous light + zero gravity
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA375085" & grepl("Light grown, Spaceflight",metaData$LibraryName))] = "PECO:0007162 continuous light exposure, PECO:0007315 zero gravity exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA486575" & grepl("Flight",metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, PECO:0007315 zero gravity exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA377717" & grepl("Spaceflight",metaData$LibraryName))] = "PECO:0007162 continuous light exposure, PECO:0007315 zero gravity exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA377574" & grepl("Spaceflight",metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, PECO:0007315 zero gravity exposure"

# continuous light + cold + GA
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA513856" & grepl("4C GA",metaData$LibraryName))] = "PECO:0007162 continuous light exposure, PECO:0007332 cold air temperature exposure, PECO:0007069 gibberellic acid exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA487166" & grepl("4C GA",metaData$LibraryName))] = "PECO:0007162 continuous light exposure, PECO:0007332 cold air temperature exposure, PECO:0007069 gibberellic acid exposure"

# continuous light + GA
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA513856" & grepl("21C GA",metaData$LibraryName))] = "PECO:0007162 continuous light exposure, PECO:0007069 gibberellic acid exposure"

# continuous light + cold
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA513856" & grepl("4C mock",metaData$LibraryName))] = "PECO:0007162 continuous light exposure, PECO:0007332 cold air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA267681" & grepl("1h|24h",metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, PECO:0007332 cold air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA487166" & grepl("4C mock",metaData$LibraryName))] = "PECO:0007162 continuous light exposure, PECO:0007332 cold air temperature exposure"

# continuous light + UV-B
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA546251" & grepl("6h", metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, PECO:0007001 UV-B light exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA609342" & grepl("UVB", metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, PECO:0007001 UV-B light exposure"

# salt + mannitol + heat
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA609704" & grepl("Salt_Mannitol_Heat",metaData$Experiment.Title))] = "PECO:0007048 sodium chloride exposure, PECO:0007319 warm/hot air temperature exposure, PECO:0001038 osmotic stress exposure, mannitol"

# mannitol + heat
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA609704" & grepl(" Mannitol_Heat_1| Mannitol_Heat_2| Mannitol_Heat_3",metaData$Experiment.Title))] = "PECO:0007319 warm/hot air temperature exposure, PECO:0001038 osmotic stress exposure, mannitol"

# salt + mannitol
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA609704" & grepl("Salt_Mannitol_1|Salt_Mannitol_2|Salt_Mannitol_3",metaData$Experiment.Title))] = "PECO:0007048 sodium chloride exposure, PECO:0001038 osmotic stress exposure, mannitol"

# salt + heat
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA609704" & grepl("Salt_Heat_1|Salt_Heat_2|Salt_Heat_3",metaData$Experiment.Title))] = "PECO:0007048 sodium chloride exposure, PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA295091" & grepl("s\\+h-",metaData$LibraryName))] = "PECO:0007048 sodium chloride exposure, PECO:0007319 warm/hot air temperature exposure"

# continuous light + salt + paraquat + highlight + heat
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA622644" & grepl(" salt\\+paraquat\\+highlight\\+heat_stress ", metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, PECO:0007048 sodium chloride exposure, PECO:0007075 high light intensity exposure, PECO:0007319 warm/hot air temperature exposure, PECO:0001037 oxidative stress exposure, paraquat"

# continuous light + paraquat + highlight + heat
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA622644" & grepl(" paraquat\\+highlight\\+heat_stress ", metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, PECO:0007075 high light intensity exposure, PECO:0007319 warm/hot air temperature exposure, PECO:0001037 oxidative stress exposure, paraquat"

# continuous light + salt + paraquat + heat
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA622644" & grepl(" salt\\+paraquat\\+heat_stress ", metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, PECO:0007048 sodium chloride exposure, PECO:0007319 warm/hot air temperature exposure, PECO:0001037 oxidative stress exposure, paraquat"

# continuous light + highlight + heat + salt
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA622644" & grepl(" salt\\+highlight\\+heat_stress ", metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, PECO:0007075 high light intensity exposure, PECO:0007048 sodium chloride exposure, PECO:0007319 warm/hot air temperature exposure"

# continuous light + highlight + salt + paraquat
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA622644" & grepl(" salt\\+paraquat\\+highlight ", metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, PECO:0007075 high light intensity exposure, PECO:0007048 sodium chloride exposure, PECO:0001037 oxidative stress exposure, paraquat"

# continuous light + paraquat + heat
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA622644" & grepl(" paraquat\\+heat_stress ", metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, PECO:0007319 warm/hot air temperature exposure, PECO:0001037 oxidative stress exposure, paraquat"

# continuous light + salt + paraquat
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA622644" & grepl(" salt\\+paraquat ", metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, PECO:0007048 sodium chloride exposure, PECO:0001037 oxidative stress exposure, paraquat"

# continuous light + salt + heat
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA622644" & grepl(" salt\\+heat_stress ", metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, PECO:0007048 sodium chloride exposure, PECO:0007319 warm/hot air temperature exposure"

# continuous light + highlight + paraquat
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA622644" & grepl(" paraquat\\+highlight ", metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, PECO:0007075 high light intensity exposure, PECO:0001037 oxidative stress exposure, paraquat"

# continuous light + highlight + salt
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA622644" & grepl(" salt\\+highlight ", metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, PECO:0007075 high light intensity exposure, PECO:0007048 sodium chloride exposure"

# continuous light + highlight + heat
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA622644" & grepl(" highlight\\+heat_stress ", metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, PECO:0007075 high light intensity exposure, PECO:0007319 warm/hot air temperature exposure"

# continuous light + highlight
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA622644" & grepl(" highlight ", metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, PECO:0007075 high light intensity exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA435737" & grepl("HL|Recovery", metaData$LibraryName))] = "PECO:0007162 continuous light exposure, PECO:0007075 high light intensity exposure"

# continuous light + paraquat
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA622644" & grepl(" paraquat ", metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, PECO:0001037 oxidative stress exposure, paraquat"

# continuous light + heat
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA622644" & grepl(" heat_stress ", metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA667059" & grepl(" 28 ",metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA488799" & grepl("HH Heat",metaData$LibraryName))] = "PECO:0007162 continuous light exposure, PECO:0007319 warm/hot air temperature exposure"

# continous light + DMSO
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA615658" & grepl("WT", metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, PECO:0007189 chemical exposure, DMSO"

# continuous light + brassinosteroid
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA615658" & grepl("BL", metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, PECO:0007409 brassinosteroid exposure, brassinolide"

# short day + hydrogen peroxide
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA481719" & grepl("_H2O2_",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007415 hydrogen peroxide exposure"

# continuous light + jasmonic acid
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA351920")] = "PECO:0007162 continuous light exposure, PECO:0007080 jasmonic acid exposure"

# continuous light + radiation mutagen exposure
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA311764" & grepl("100Gy",metaData$LibraryName))] = "PECO:0007162 continuous light exposure, PECO:0001039 radiation mutagen exposure"

# natural long day + Hyaloperonospora arabidopsidis
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA562704" & grepl("m1389",metaData$Experiment.Title))] = "PECO:0001012 hexose exposure, d-tagatose, PECO:0007233 fungal plant exposure, Hyaloperonospora arabidopsidis"

# dark + jasmonate
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA551415" & grepl("JA_0_25hour|JA_0_5hour|JA_0\\.5hour|JA_1hour|JA_2hour|JA_4hour|JA_8hour|JA_12hour|JA_24hour",metaData$LibraryName))] = "PECO:0007270 continuous dark (no light) exposure, PECO:0007080 jasmonic acid exposure"

# high light + heat
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA575019" & grepl("Local HL\\+HS |Local HL\\+HS_|Systemic HL\\+HS_|Systemic HL&HS",metaData$Experiment.Title))] = "PECO:0007075 high light intensity exposure, PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA555093" & grepl("HL\\+HS",metaData$LibraryName))] = "PECO:0007075 high light intensity exposure, PECO:0007319 warm/hot air temperature exposure"

# short day + high light
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA481718" & grepl("H2O_Highlight",metaData$LibraryName))] = "PECO:0007200 short day length exposure,  PECO:0007075 high light intensity exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA595162")] = "PECO:0007200 short day length exposure,  PECO:0007075 high light intensity exposure"

# short day + salt
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB33339" & grepl("SS2|SS4",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007048 sodium chloride exposure"

# short day + ATP
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA494862" & grepl("ATP",metaData$SampleName))] = "PECO:0007200 short day length exposure, PECO:0007189 chemical exposure, ATP"

# phosphate + flg22
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA344453" & grepl("HighP_flg22", metaData$Experiment.Title))] = "PECO:0007397 phosphorus nutrient exposure, flagellin 22 exposure"

# phosphate + methyl jasmonate
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA344453" & grepl("HighP_MeJA", metaData$Experiment.Title))] = "PECO:0007397 phosphorus nutrient exposure, PECO:0007407 methyl jasmonate exposure"

# limited phosphate + methyl jasmonate
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA344453" & grepl("LowP_MeJA", metaData$Experiment.Title))] = "PECO:0001046 limited phosphate exposure, PECO:0007407 methyl jasmonate exposure"

# limited phosphate + flg22
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA344453" & grepl("LowP_flg22", metaData$Experiment.Title))] = "PECO:0001046 limited phosphate exposure, flagellin 22 exposure"

# short day + DCMU + heat
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA378644" & grepl("dcmu_30um_27",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007319 warm/hot air temperature exposure, PECO:0007189 chemical exposure, DCMU"

# short day + dbmib + heat
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA378644" & grepl("dbmib_50um_27",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007319 warm/hot air temperature exposure, PECO:0007189 chemical exposure, DBMIB"

# short day + dbmib
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA378644" & grepl("dbmib_50um_22",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007189 chemical exposure, DBMIB"

# short day + 3-(3,4-dichlorophenyl)-1,1-dimethylurea
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA378644" & grepl("dcmu_30um_22",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007189 chemical exposure, DCMU"

# short day + ethanol + hot air temp
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA378644" & grepl("etoh_27.*",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007319 warm/hot air temperature exposure, PECO:0007189 chemical exposure, ethanol"

# short day + ethanol
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA378644" & grepl("etoh_22.*",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007189 chemical exposure, ethanol"

# short day + hot air temp
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA378644" & grepl("col.*27c|col_27",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB33339" & grepl("HS",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA542115" & grepl("Col.*hightemp",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA706054" & grepl("_27",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA809747" & grepl("CK_28",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA384110" & grepl("27C",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA328771" & grepl("27",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA681187" & grepl("Light_28C",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, PECO:0007319 warm/hot air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA274954" & grepl("27",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007319 warm/hot air temperature exposure"

# short day + cold air temp
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA378644" & grepl("col_17",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007332 cold air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB33339" & grepl("CS",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007332 cold air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA338071" & grepl("16C\\-16C",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007332 cold air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA535479" & grepl("_V",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, PECO:0007332 cold air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB6701" & grepl("_T_",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007332 cold air temperature exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA328771" & grepl("16",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007332 cold air temperature exposure"

# short day + limited phosphate
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA287627" & grepl("minusP.*mock",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0001046 limited phosphate exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB27007" & grepl("noP_MgCl",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0001046 limited phosphate exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJDB5141" & grepl("P_",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0001046 limited phosphate exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA531169" & grepl("_low_",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, PECO:0001046 limited phosphate exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA531168" & grepl("NoBacteria_50Pi",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, PECO:0001046 limited phosphate exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJDB7233" & grepl("-P",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0001046 limited phosphate exposure"

# short day + phosphate exposure
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA287627" & grepl("plusP.*mock",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007397 phosphorus nutrient exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB27007" & grepl("^P_MgCl",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007397 phosphorus nutrient exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA531169" & grepl("_low\\+|_medium_|high",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, PECO:0007397 phosphorus nutrient exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA531168" & grepl("NoBacteria_1000Pi",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, PECO:0007397 phosphorus nutrient exposure"

# continuous light + salt + paraquat
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA622644" & grepl(" salt\\+paraquat ", metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, PECO:0007048 sodium chloride exposure, PECO:0001037 oxidative stress exposure, paraquat"

# continuous light + salt
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA641427")] = "PECO:0007162 continuous light exposure, PECO:0007048 sodium chloride exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA622644" & grepl("Salt", metaData$Experiment.Title))] = "PECO:0007162 continuous light exposure, PECO:0007048 sodium chloride exposure"

# dark + in vitro growth medium exposure
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA592549" & grepl("CIM",metaData$Experiment.Title))] = "PECO:0007270 continuous dark (no light) exposure, PECO:0007262 in vitro solid growth medium exposure, callus inducing media"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA592549" & grepl("SIM",metaData$Experiment.Title))] = "PECO:0007270 continuous dark (no light) exposure, PECO:0007262 in vitro solid growth medium exposure, shoot inducing media"

# short day + diphenyleneiodonium + high light
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA481718" & grepl("DPI_Highlight",metaData$LibraryName))] = "PECO:0007200 short day length exposure,  PECO:0007075 high light intensity exposure, PECO:0001003 organic chemical exposure, diphenyleneiodonium"

# far red + fluctuating temp
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA434094" & grepl("LD\\+FR\\+TEMP",metaData$LibraryName))] = "PECO:0007203 far red light exposure, fluctuating air temperature"

# intermittent light, fluctuating light
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB31094" & grepl("^S",metaData$LibraryName))] = "PECO:0007128 intermittent light exposure"

# insects + bacteria
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA315516" & grepl("P.rapae main treatment, B.cinerea pre-treatment;",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007114 Lepidoptera exposure, Pieris rapae, PECO:0007233 fungal plant exposure, Botrytis cinerea"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA315516" & grepl("B.cinerea main treatment, P.rapae pre-treatment;",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007114 Lepidoptera exposure, Pieris rapae, PECO:0007233 fungal plant exposure, Botrytis cinerea"

# drought + bacteria
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA315516" & grepl("Drought main treatment, B.cinerea pre-treatment;",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007404 drought environment exposure, PECO:0007233 fungal plant exposure, Botrytis cinerea"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA315516" & grepl("B.cinerea main treatment, drought pre-treatment;",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007404 drought environment exposure, PECO:0007233 fungal plant exposure, Botrytis cinerea"

# drought + insects
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA315516" & grepl("Drought main treatment, P.rapae pre-treatment",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007404 drought environment exposure, PECO:0007114 Lepidoptera exposure, Pieris rapae"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA315516" & grepl("P.rapae main treatment, drought pre-treatment;",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007404 drought environment exposure, PECO:0007114 Lepidoptera exposure, Pieris rapae"

# short day + mechanical damage
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA419306" & grepl("Separated|Grafted",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007373 mechanical damage exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB31311" & grepl("2.5min|5min|10min",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007373 mechanical damage exposure"

# dark + mechanical damage
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA592549" & grepl("AfterCut",metaData$Experiment.Title))] = "PECO:0007270 continuous dark (no light) exposure, PECO:0007373 mechanical damage exposure"

# bacteria + limited phosphate
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA344452" & grepl("LowP_SynCom", metaData$Experiment.Title))] = "PECO:0001046 limited phosphate exposure, PECO:0007246 bacterial plant exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA344451" & grepl("LowP_SynCom", metaData$Experiment.Title))] = "PECO:0001046 limited phosphate exposure, PECO:0007246 bacterial plant exposure"

# bacteria + phosphate exposure
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA397098" & !grepl("No Bacteria",metaData$LibraryName))] = "PECO:0007246 bacterial plant exposure, PECO:0007397 phosphorus nutrient exposure" # germination in presence or absence of phosphate ignored
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA344452" & grepl("HighP_SynCom", metaData$Experiment.Title))] = "PECO:0007246 bacterial plant exposure, PECO:0007397 phosphorus nutrient exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA344451" & grepl("HighP_SynCom", metaData$Experiment.Title))] = "PECO:0007246 bacterial plant exposure, PECO:0007397 phosphorus nutrient exposure"

# dark + space 
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA375085" & grepl("Dark grown, Spaceflight",metaData$LibraryName))] = "PECO:0007270 continuous dark (no light) exposure, PECO:0007315 zero gravity exposure"

# short day + bacterial
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA635635" & grepl("SynCom",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, PECO:0007246 bacterial plant exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA321559" & grepl("HopAM1|EV",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007246 bacterial plant exposure, Pseudomonas syringae"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA412447" & grepl("BAC",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007246 bacterial plant exposure, Pseudomonas simiae" # 10 hrs of light
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB34520" & grepl("KV|A2|A4",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007246 bacterial plant exposure, Pseudomonas fluorescens" # 10 hrs of light
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA813448" & grepl("avrRpm",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, PECO:0007246 bacterial plant exposure, Pseudomonas syringae"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA209363" & grepl("C321A|PopP2",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, PECO:0007246 bacterial plant exposure, Pseudomonas fluorescens" # 10 hrs of light
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA666544" & grepl("DC|hrcC",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, PECO:0007246 bacterial plant exposure, Pseudmonas syringae"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA358395" & grepl("Rps",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007246 bacterial plant exposure, Pseudomonas syringae"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA358395" & grepl("Rps|Rpm",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, PECO:0007246 bacterial plant exposure, Pseudomonas syringae"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA643438" & grepl("DC|hrcC",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, PECO:0007246 bacterial plant exposure, Pseudomonas syringae"

# short day + nlp20
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA549850" & grepl("nlp20",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, nlp20 exposure"

# short day + nitrogen
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA477359" & grepl("\\-NoInf",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, PECO:0007284 nitrogen macronutrient exposure"

# short day + flagellin 22
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA224705" & grepl("flg22",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, flagellin 22 exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA476754" & grepl("_T_",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, flagellin 22 exposure" # 10 hrs of light
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA412447" & grepl("FLG",metaData$LibraryName))] = "PECO:0007200 short day length exposure, flagellin 22 exposure" # 10 hrs of light
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA549850" & grepl("flg22",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, flagellin 22 exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB31312" & grepl("flg22",metaData$LibraryName))] = "PECO:0007200 short day length exposure, flagellin 22 exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB31313")] = "PECO:0007200 short day length exposure, flagellin 22 exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA449619" & grepl("flg22", metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, flagellin 22 exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA358395" & grepl("Flg22",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, flagellin 22 exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA358395" & grepl("Flg22",metaData$LibraryName))] = "PECO:0007200 short day length exposure, flagellin 22 exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB31337" & grepl("flg22",metaData$LibraryName))] = "PECO:0007200 short day length exposure, flagellin 22 exposure"

# short day + pep1
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA449619" & grepl("Pep1", metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, Pep1 exposure"

# short day + chitin
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA412447" & grepl("CHITIN",metaData$LibraryName))] = "PECO:0007200 short day length exposure, chitin exposure" # 10 hrs of light
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA549850" & grepl("C6",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, chitin exposure"

# bacterial exposure + flagellin 22
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA657867" & grepl("MF.*_flg22_",metaData$Experiment.Title))] = "PECO:0007246 bacterial plant exposure, flagellin 22 exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA657867" & grepl("cl.*_flg22_",metaData$Experiment.Title))] = "PECO:0007246 bacterial plant exposure, flagellin 22 exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA657867" & grepl("e.coli.*_flg22_",metaData$Experiment.Title))] = "PECO:0007246 bacterial plant exposure, flagellin 22 exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA657867" & grepl("SynCom.*_flg22_",metaData$Experiment.Title))] = "PECO:0007246 bacterial plant exposure, flagellin 22 exposure"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA657868" & grepl("SynCom.*_flg22_",metaData$Experiment.Title))] = "PECO:0007246 bacterial plant exposure, flagellin 22 exposure"

# short day + fungus
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA315516" & grepl("^B.cinerea main treatment;",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007233 fungal plant exposure, Botrytis cinerea"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB26953" & grepl("hpa",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007233 fungal plant exposure, Hyaloperonospora arabidopsidis"
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB31803" & grepl("rep._sq",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007233 fungal plant exposure, Albugo laibachii"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA809747" & grepl("Si_22|Si_25",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, PECO:0007233 fungal plant exposure, Serendipita indica"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA625345" & grepl("M. albugensis Peeled",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, PECO:0007233 fungal plant exposure, Moesziomyces albugensis"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA625345" & grepl("M. albugensis \\+ A. laibachii Peeled",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, PECO:0007233 fungal plant exposure, Moesziomyces albugensis, Albugo laibachii"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA662919")] = "PECO:0007200 short day length exposure, PECO:0007233 fungal plant exposure, Fusarium oxysporum"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA232536" & !grepl("H2O",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, PECO:0007233 fungal plant exposure, Hyaloperonospora arabidopsidis"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA473829" & grepl("^col",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007233 fungal plant exposure, Botrytis cinerea"

# short day + bacteria + fungus
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA625345" & grepl("SynCom",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, PECO:0007246 bacterial plant exposure, PECO:0007233 fungal plant exposure"

# short day + nitrogen + fungus
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA477359" & grepl("\\-Inf",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, PECO:0007284 nitrogen macronutrient exposure, PECO:0007233 fungal plant exposure, Botrytis cinerea"

# short day + phosphate + fungus
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA287627" & grepl("plusP.*Ct",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007397 phosphorus nutrient exposure, PECO:0007233 fungal plant exposure, Colletotrichum tofieldiae"

# short day + limited phosphate + fungus
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA287627" & grepl("minusP.*Ct",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0001046 limited phosphate exposure, PECO:0007233 fungal plant exposure, Colletotrichum tofieldiae"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA287627" & grepl("minusP.*Ci",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0001046 limited phosphate exposure, PECO:0007233 fungal plant exposure, Colletotrichum incanum"

# short day + limited phosphate + bacteria
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB27007" & grepl("noP_350",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0001046 limited phosphate exposure, PECO:0007246 bacterial plant exposure, Rhizobium"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA531168" & grepl("SynCom_50Pi",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, PECO:0001046 limited phosphate exposure, PECO:0007246 bacterial plant exposure"

# short day + phosphate + bacteria
metaData$TreatmentOntology[(metaData$BioProject == "PRJEB27007" & grepl("^P_350",metaData$LibraryName))] = "PECO:0007200 short day length exposure, PECO:0007397 phosphorus nutrient exposure, PECO:0007246 bacterial plant exposure, Rhizobium"
metaData$TreatmentOntology[(metaData$BioProject == "PRJNA531168" & grepl("SynCom_1000Pi",metaData$Experiment.Title))] = "PECO:0007200 short day length exposure, PECO:0007397 phosphorus nutrient exposure, PECO:0007246 bacterial plant exposure"


```

# Assign Tissue factor levels
# Use Plant Ontology: https://browser.planteome.org/amigo/term/PO:0025131#display-lineage-tab
```{r}
metaData$TissueOntology = NA

### LEAF #######################################################################
metaData$TissueOntology[(metaData$BioProject == "PRJNA319904")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA224133")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA473829")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA486575" & grepl("Leaves",metaData$Experiment.Title))] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA237342")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA313379")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA488315")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA348676")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJEB19974")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJEB15161")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA481720")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA315516")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA187928")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA564095")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJEB31094")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA314076" & grepl("Leaf",metaData$LibraryName))] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA382583")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA224705")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA638728")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA481718")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA692747" & grepl("Lv",metaData$Experiment.Title))] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA494179" & grepl("leaf",metaData$Experiment.Title))] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA321559")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA544417")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA691326" & grepl("Lv|Rs",metaData$Experiment.Title))] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA344545" & grepl("^R",metaData$LibraryName))] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJEB26953")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJEB34520")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA575019")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA813448")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA588176" & grepl("leaves",metaData$Experiment.Title))] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA562704")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJEB32665" & grepl("leaf|leaves|petiole",metaData$Tissue))] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA284739" & grepl("leaf WT",metaData$LibraryName))] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJEB31312")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJEB31313")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA595162")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA368967" & grepl("rosette|ler",metaData$LibraryName))] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA598201")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA609704")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA186843")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJEB8427")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA279529")] = "PO:0025034 leaf" # conflicting labels, labeled as seedling on sra, but leaf in publication
metaData$TissueOntology[(metaData$BioProject == "PRJNA281364")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA315374" & grepl("CL",metaData$Experiment.Title))] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA323943")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA363056" & grepl("RL",metaData$LibraryName))] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA411947" & grepl("Leaf",metaData$LibraryName))] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA421355" & grepl("leaf",metaData$LibraryName))] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA445300")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA525816")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA666544")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA324514")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA477359")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA481452")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA394661")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA209363")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA294703" & grepl("leaves",metaData$Tissue))] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA342309" & grepl("leaf",metaData$LibraryName))] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA565160")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJEB31314")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA261430")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA274887")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA338071")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA445392")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA472051")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA667430")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA358395")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA483528")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJEB31803")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA511671")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA319370")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJEB27944")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA387601")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA395664")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA422982")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA643438")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA714877")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA493021")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA504886" & grepl("leaves",metaData$LibraryName))] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA722743")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJEB31311")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJEB31337")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA280870")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA295091")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA321410")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA325245")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA339032")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA407271" & grepl("Leaf",metaData$LibraryName))] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA477336")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA480911")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA553702")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA555093")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA592062")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA625345")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJEB23627")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA232621" & grepl("^L",metaData$LibraryName))] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJEB33240")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA261035")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA353660")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA512644" & grepl("Leaf|Rosette",metaData$Experiment.Title))] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJDB7233")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA325158" & grepl("leaf",metaData$LibraryName))] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA485134")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA557609")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA622692")] = "PO:0025034 leaf"
metaData$TissueOntology[(metaData$BioProject == "PRJNA377574" & grepl("Leaf",metaData$Experiment.Title))] = "PO:0025034 leaf"

### ROOT #######################################################################
metaData$TissueOntology[(metaData$BioProject == "PRJNA657867")] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA377717")] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA486575" & grepl("Roots",metaData$Experiment.Title))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA525303" & grepl("Rt",metaData$LibraryName))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA300543" & grepl("root$",metaData$Tissue))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA422118")] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA314076" & grepl("Root|root",metaData$LibraryName))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA382136" & grepl("R",metaData$LibraryName))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA657868")] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA692747" & grepl("Rs",metaData$Experiment.Title))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA635635")] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA287627")] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA640389")] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA478998" & grepl("_R",metaData$LibraryName))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA412447")] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA588176" & grepl("Root",metaData$Experiment.Title))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJEB32665" & grepl("root",metaData$Tissue))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA602947")] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA435505" & grepl("root",metaData$LibraryName))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA231088" & grepl("root",metaData$LibraryName))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA705878")] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJEB27007")] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA662710" & grepl("R;",metaData$Experiment.Title))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA368967" & grepl("root",metaData$LibraryName))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA315374" & grepl("CR",metaData$Experiment.Title))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA574854" & grepl("_R",metaData$Experiment.Title))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA574855" & grepl("_R",metaData$Experiment.Title))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA551121")] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA666436")] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA294703" & grepl("root",metaData$Tissue))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA449619")] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA497883")] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA672150")] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJDB5141")] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA531169")] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA261433" & grepl("roots",metaData$LibraryName))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA271131" & grepl("R",metaData$LibraryName))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA294313")] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA475464")] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA624317")] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA639498" & grepl("\\-R\\-",metaData$Experiment.Title))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA639499" & grepl("AR|BR|CR",metaData$Experiment.Title))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA667059" & grepl("root",metaData$Experiment.Title))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA717244")] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA132271" & grepl("root",metaData$LibraryName))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA470732" & grepl("root",metaData$LibraryName))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA608657" & grepl("root",metaData$Experiment.Title))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA308080" & grepl("Root",metaData$LibraryName))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA392864" & grepl("_R",metaData$LibraryName))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA492842" & grepl("_R",metaData$LibraryName))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA639495")] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA639496")] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA297270" & grepl("root",metaData$LibraryName))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJEB6701" & grepl("Rt",metaData$LibraryName))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA210438" & grepl("root",metaData$LibraryName))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA286779" & grepl("Root",metaData$LibraryName))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA380494")] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA407271" & grepl("Root",metaData$LibraryName))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA415015" & grepl("Root",metaData$LibraryName))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA419538")] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA663969" & grepl("R",metaData$LibraryName))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA662919")] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA271595")] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA325158" & grepl("root",metaData$LibraryName))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA450539")] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA542492")] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA639497" & grepl("-R-",metaData$Experiment.Title))] = "PO:0009005 root"
metaData$TissueOntology[(metaData$BioProject == "PRJNA377574" & grepl("Root",metaData$Experiment.Title))] = "PO:0009005 root"

### WHOLE PLANT ###############################################################
metaData$TissueOntology[(metaData$BioProject == "PRJNA397098")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA475542")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA344452")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA341504")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA344451")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA434094")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA524697")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA245231")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA494179" & grepl("ABA|drought|cold",metaData$Experiment.Title))] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA641427")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA378644")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA344453")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA478998" & !grepl("_R|_S",metaData$LibraryName))] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA494862")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA258547")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA476754")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA302786")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA551415")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA611770")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA476945")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA549850")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA231088" & grepl("seedling",metaData$LibraryName))] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA311764" & grepl("seedling",metaData$LibraryName))] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA601788")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA481719")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA527966")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA615658")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA622644")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA368967" & grepl("Th2",metaData$LibraryName))] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA508267")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA626535")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA154687")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA188075")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA310623")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA339762")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA391475")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA525432")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA609342")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA658687")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA661316")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA476778")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA566413")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA433501")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA271089")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA446045")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA510535")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA513852")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "SRP000935")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA315570")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA554610")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA435737")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJEB32446")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA258161")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA302003")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA322265")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA345276")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA412889")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA531168")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA542115")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA589109")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA646744")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA706054")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA272425")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA525452")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA561790")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJEB24412")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA526082")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA488799")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA267681")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA285810")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA354369")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA378128")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA434180")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA471243")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA523174")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA546251")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA593027")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA601157")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA603477")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA659897")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA661303")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA287644")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA384110")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA504886" & grepl("RNA_rep",metaData$LibraryName))] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA563066")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA301696")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA481217")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA505965")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA521623")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA451508")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA535479")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA416120")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA478084")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJEB28610")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA380922")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA407271" & grepl("Seedling",metaData$LibraryName))] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA576144")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA681187")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA213635" & grepl("seedlings",metaData$Tissue))] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA274954")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA750922")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA319318")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA351920")] = "PO:0000003 whole plant"
metaData$TissueOntology[(metaData$BioProject == "PRJNA610701")] = "PO:0000003 whole plant"

# meiocyte
metaData$TissueOntology[(metaData$BioProject == "PRJNA342309" & grepl("meiocyte", metaData$LibraryName))] = "PO:0006204 sporocyte"

# ground tissue
metaData$TissueOntology[(metaData$BioProject == "PRJEB33339")] = "PO:0025059 portion of ground tissue"

# cotyledon
metaData$TissueOntology[(metaData$BioProject == "PRJNA320878" & grepl("cotyledon", metaData$LibraryName))] = "PO:0020030 cotyledon"
metaData$TissueOntology[(metaData$BioProject == "PRJEB32665" & grepl("cotyledon",metaData$Tissue))] = "PO:0020030 cotyledon"
metaData$TissueOntology[(metaData$BioProject == "PRJNA256037" & grepl("Cot",metaData$LibraryName))] = "PO:0020030 cotyledon"
metaData$TissueOntology[(metaData$BioProject == "PRJNA311764" & grepl("cotyledon",metaData$LibraryName))] = "PO:0020030 cotyledon"
metaData$TissueOntology[(metaData$BioProject == "PRJNA702187")] = "PO:0020030 cotyledon"
metaData$TissueOntology[(metaData$BioProject == "PRJNA411947" & grepl("Cotyledon",metaData$LibraryName))] = "PO:0020030 cotyledon"
metaData$TissueOntology[(metaData$BioProject == "PRJNA421355" & grepl("cotyledon",metaData$LibraryName))] = "PO:0020030 cotyledon"
metaData$TissueOntology[(metaData$BioProject == "PRJNA609261" & grepl("cotyledon",metaData$Experiment.Title))] = "PO:0020030 cotyledon"
metaData$TissueOntology[(metaData$BioProject == "PRJDB6957")] = "PO:0020030 cotyledon"
metaData$TissueOntology[(metaData$BioProject == "PRJNA314076" & grepl("Seedling Cotyledon",metaData$LibraryName))] = "PO:0020030 cotyledon"

# cotyledon + shoot apical meristem
metaData$TissueOntology[(metaData$BioProject == "PRJNA549286" & grepl("Cot",metaData$LibraryName))] = "PO:0020030 cotyledon, PO:0020148 shoot apical meristem"

#### hypocotyl ###################################################################################################
metaData$TissueOntology[(metaData$BioProject == "PRJNA592549")] = "PO:0020100 hypocotyl"
metaData$TissueOntology[(metaData$BioProject == "PRJNA419306")] = "PO:0020100 hypocotyl"
metaData$TissueOntology[(metaData$BioProject == "PRJNA320878" & grepl("hypocotyl", metaData$LibraryName))] = "PO:0020100 hypocotyl"
metaData$TissueOntology[(metaData$BioProject == "PRJEB32665" & grepl("hypocotyl",metaData$Tissue))] = "PO:0020100 hypocotyl"
metaData$TissueOntology[(metaData$BioProject == "PRJNA256037" & grepl("Hyp",metaData$LibraryName))] = "PO:0020100 hypocotyl"
metaData$TissueOntology[(metaData$BioProject == "PRJNA411947" & grepl("Hypocotyl",metaData$LibraryName))] = "PO:0020100 hypocotyl"
metaData$TissueOntology[(metaData$BioProject == "PRJNA549286" & grepl("Hypo",metaData$LibraryName))] = "PO:0020100 hypocotyl"
metaData$TissueOntology[(metaData$BioProject == "PRJNA716358")] = "PO:0020100 hypocotyl"
metaData$TissueOntology[(metaData$BioProject == "PRJNA667059" & grepl("hyp",metaData$Experiment.Title))] = "PO:0020100 hypocotyl"
metaData$TissueOntology[(metaData$BioProject == "PRJNA609261" & grepl("hypocotyl",metaData$Experiment.Title))] = "PO:0020100 hypocotyl"

# callus
metaData$TissueOntology[(metaData$BioProject == "PRJEB32665" & grepl("callus",metaData$Tissue))] = "PO:0005052 callus"

# cauline 
metaData$TissueOntology[(metaData$BioProject == "PRJNA272491" & grepl("Cauline",metaData$Experiment.Title))] = "PO:0000013 cauline leaf"

# fruit, silique
metaData$TissueOntology[(metaData$BioProject == "PRJNA314076" & grepl("Silique|silique",metaData$LibraryName))] = "PO:0009001 fruit"
metaData$TissueOntology[(metaData$BioProject == "PRJNA494179" & grepl("silique",metaData$Experiment.Title))] = "PO:0009001 fruit"
metaData$TissueOntology[(metaData$BioProject == "PRJEB32665" & grepl("silique|fruit",metaData$Tissue))] = "PO:0009001 fruit"
metaData$TissueOntology[(metaData$BioProject == "PRJEB25745")] = "PO:0009001 fruit"
metaData$TissueOntology[(metaData$BioProject == "PRJNA261433" & grepl("siliques",metaData$LibraryName))] = "PO:0009001 fruit"
metaData$TissueOntology[(metaData$BioProject == "PRJNA512644" & grepl("Silique",metaData$Experiment.Title))] = "PO:0009001 fruit"

# ovule
metaData$TissueOntology[(metaData$BioProject == "PRJNA692747" & grepl("Ov",metaData$Experiment.Title))] = "PO:0020003 plant ovule"
metaData$TissueOntology[(metaData$BioProject == "PRJNA691326" & grepl("Ov",metaData$Experiment.Title))] = "PO:0020003 plant ovule"

# flower bud
metaData$TissueOntology[(metaData$BioProject == "PRJNA693252")] = "PO:0000056 flower bud"
metaData$TissueOntology[(metaData$BioProject == "PRJNA691326" & grepl("rep",metaData$Experiment.Title))] = "PO:0000056 flower bud"
metaData$TissueOntology[(metaData$BioProject == "PRJNA231088" & grepl("floral bud",metaData$LibraryName))] = "PO:0000056 flower bud"
metaData$TissueOntology[(metaData$BioProject == "PRJNA363056" & grepl("LF|EF",metaData$LibraryName))] = "PO:0000056 flower bud"
metaData$TissueOntology[(metaData$BioProject == "PRJNA596803")] = "PO:0000056 flower bud"
metaData$TissueOntology[(metaData$BioProject == "PRJNA232621" & grepl("FB",metaData$LibraryName))] = "PO:0000056 flower bud"

# flowers
metaData$TissueOntology[(metaData$BioProject == "PRJNA314076" & grepl("Flower|flower",metaData$LibraryName))] = "PO:0009046 flower"
metaData$TissueOntology[(metaData$BioProject == "PRJEB32665" & grepl("flower|sepal|carpel|petal|stamen",metaData$Tissue))] = "PO:0009046 flower"
metaData$TissueOntology[(metaData$BioProject == "PRJNA311764" & grepl("flower",metaData$LibraryName))] = "PO:0009046 flower"
metaData$TissueOntology[(metaData$BioProject == "PRJNA411947" & grepl("Flower|flower",metaData$LibraryName))] = "PO:0009046 flower"
metaData$TissueOntology[(metaData$BioProject == "PRJNA406971")] = "PO:0009046 flower"
metaData$TissueOntology[(metaData$BioProject == "PRJNA294703" & grepl("flower",metaData$Tissue))] = "PO:0009046 flower"
metaData$TissueOntology[(metaData$BioProject == "PRJNA624145")] = "PO:0009046 flower"
metaData$TissueOntology[(metaData$BioProject == "PRJNA407271" & grepl("Flower",metaData$LibraryName))] = "PO:0009046 flower"
metaData$TissueOntology[(metaData$BioProject == "PRJNA512644" & grepl("Flower",metaData$Experiment.Title))] = "PO:0009046 flower"
metaData$TissueOntology[(metaData$BioProject == "PRJNA430092")] = "PO:0009046 flower"

# pollen
metaData$TissueOntology[(metaData$BioProject == "PRJEB32665" & grepl("pollen",metaData$Tissue))] = "PO:0025281 pollen"

# stigma
metaData$TissueOntology[(metaData$BioProject == "PRJNA314076" & grepl("Stigma|stigma",metaData$LibraryName))] = "PO:0009073 stigma"
metaData$TissueOntology[(metaData$BioProject == "PRJEB24752")] = "PO:0009073 stigma"

# inflorescence
metaData$TissueOntology[(metaData$BioProject == "PRJNA272491" & grepl("Inflorescence",metaData$Experiment.Title))] = "PO:0009049 inflorescence"
metaData$TissueOntology[(metaData$BioProject == "PRJNA494179" & grepl("inflorescence",metaData$Experiment.Title))] = "PO:0009049 inflorescence"
metaData$TissueOntology[(metaData$BioProject == "PRJNA261433" & grepl("inflorescences",metaData$Experiment.Title))] = "PO:0009049 inflorescence"
metaData$TissueOntology[(metaData$BioProject == "PRJNA325158" & grepl("inflorescence",metaData$LibraryName))] = "PO:0009049 inflorescence"

# inflorescence axis
metaData$TissueOntology[(metaData$BioProject == "PRJNA314076" & grepl("Axis of the inflorescence",metaData$LibraryName))] = "PO:0020122 inflorescence axis"

# pedicel
metaData$TissueOntology[(metaData$BioProject == "PRJNA314076" & grepl("Pedicel",metaData$LibraryName))] = "PO:0030112 pedicel"

# anther
metaData$TissueOntology[(metaData$BioProject == "PRJNA314076" & grepl("Anther",metaData$LibraryName))] = "PO:0009066 anther"

# internode
metaData$TissueOntology[(metaData$BioProject == "PRJNA314076" & grepl("Internode|internode",metaData$LibraryName))] = "PO:0020142 internode"

# seedling hypocotyl
metaData$TissueOntology[(metaData$BioProject == "PRJNA314076" & grepl("Seedling Hypocotyl",metaData$LibraryName))] = "PO:0025291 seedling hypocotyl"

### SHOOTS ####
metaData$TissueOntology[(metaData$BioProject == "PRJNA525303" & grepl("Sht",metaData$LibraryName))] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA232536")] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA382136" & grepl("S",metaData$LibraryName))] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA478998" & grepl("_S",metaData$LibraryName))] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA275437")] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA435505" & grepl("shoot",metaData$LibraryName))] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA662710" & grepl("S;",metaData$Experiment.Title))] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA513856")] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA574854" & grepl("_S",metaData$Experiment.Title))] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA574855" & grepl("_S",metaData$Experiment.Title))] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA592037")] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA685167")] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA261433" & grepl("shoots",metaData$LibraryName))] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA271131" & grepl("S",metaData$LibraryName))] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA639498" & grepl("\\-S\\-",metaData$Experiment.Title))] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA639499" & grepl("AS|BS|CS",metaData$Experiment.Title))] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA681183")] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA543995")] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA494693")] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA132271" & grepl("shoot",metaData$LibraryName))] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA470732" & grepl("shoot",metaData$LibraryName))] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA608657" & grepl("shoot",metaData$Experiment.Title))] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA609261" & grepl("hook",metaData$Experiment.Title))] = "PO:0009006 shoot system" # apical hook
metaData$TissueOntology[(metaData$BioProject == "PRJNA809747")] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA308080" & grepl("Shoot",metaData$LibraryName))] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA392864" & grepl("_S",metaData$LibraryName))] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA492842" & grepl("_S",metaData$LibraryName))] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA512107")] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA473032")] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA297270" & grepl("shoot",metaData$LibraryName))] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJEB6701" & grepl("St",metaData$LibraryName))] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA210438" & grepl("shoot",metaData$LibraryName))] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA286779" & grepl("Shoot",metaData$LibraryName))] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA297861")] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA328771")] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA415015" & grepl("Shoot",metaData$LibraryName))] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA436778")] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA663969" & grepl("S",metaData$LibraryName))] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA213635" & grepl("aerial seedling",metaData$Tissue))] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA259680")] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA487166")] = "PO:0009006 shoot system"
metaData$TissueOntology[(metaData$BioProject == "PRJNA639497" & grepl("-S-",metaData$Experiment.Title))] = "PO:0009006 shoot system"

# stem
metaData$TissueOntology[(metaData$BioProject == "PRJNA344545" & grepl("^T",metaData$LibraryName))] = "PO:0009047 stem" 
metaData$TissueOntology[(metaData$BioProject == "PRJEB32665" & grepl("stem",metaData$Tissue))] = "PO:0009047 stem"
metaData$TissueOntology[(metaData$BioProject == "PRJNA284739" & grepl("stem",metaData$LibraryName))] = "PO:0009047 stem"
metaData$TissueOntology[(metaData$BioProject == "PRJNA512644" & grepl("Stem",metaData$Experiment.Title))] = "PO:0009047 stem"
metaData$TissueOntology[(metaData$BioProject == "PRJNA489360")] = "PO:0009047 stem"

# root tip
metaData$TissueOntology[(metaData$BioProject == "PRJNA375085")] = "PO:0000025 root tip"
metaData$TissueOntology[(metaData$BioProject == "PRJNA666631")] = "PO:0000025 root tip"

# lateral root apical meristem
metaData$TissueOntology[(metaData$BioProject == "PRJNA680241")] = "PO:0006020 lateral root apical meristem" # lateral root primordium

# root meristem
metaData$TissueOntology[(metaData$BioProject == "PRJNA590939")] = "PO:0006085 root meristem"
metaData$TissueOntology[(metaData$BioProject == "PRJNA607298")] = "PO:0006085 root meristem"

# root stele
metaData$TissueOntology[(metaData$BioProject == "PRJNA300543" & grepl("root stele",metaData$Tissue))] = "PO:0020124 root stele"

# apical meristem
metaData$TissueOntology[(metaData$BioProject == "PRJNA590705")] = "PO:0020144 apical meristem"

# shoot apical meristem
metaData$TissueOntology[(metaData$BioProject == "PRJNA268115")] = "PO:0020148 shoot apical meristem"
metaData$TissueOntology[(metaData$BioProject == "PRJNA354615")] = "PO:0020148 shoot apical meristem"
metaData$TissueOntology[(metaData$BioProject == "PRJNA151357")] = "PO:0020148 shoot apical meristem"
metaData$TissueOntology[(metaData$BioProject == "PRJNA314076" & grepl("Seedling Meristem",metaData$LibraryName))] = "PO:0020148 shoot apical meristem"

### SEEDS ###
metaData$TissueOntology[(metaData$BioProject == "PRJNA660263")] = "PO:0009010 seed"
metaData$TissueOntology[(metaData$BioProject == "PRJNA314076" & grepl("seed |Seed | seeds",metaData$LibraryName))] = "PO:0009010 seed"
metaData$TissueOntology[(metaData$BioProject == "PRJNA508214")] = "PO:0009010 seed"
metaData$TissueOntology[(metaData$BioProject == "PRJEB32665" & grepl("seed",metaData$Tissue))] = "PO:0009010 seed"
metaData$TissueOntology[(metaData$BioProject == "PRJNA529796")] = "PO:0009010 seed"
metaData$TissueOntology[(metaData$BioProject == "PRJNA496426")] = "PO:0009010 seed" # embryos
metaData$TissueOntology[(metaData$BioProject == "PRJNA301162")] = "PO:0009010 seed" # embryos
metaData$TissueOntology[(metaData$BioProject == "PRJNA411947" & grepl("Seeds",metaData$LibraryName))] = "PO:0009010 seed"
metaData$TissueOntology[(metaData$BioProject == "PRJNA369750")] = "PO:0009010 seed"
metaData$TissueOntology[(metaData$BioProject == "PRJNA703036")] = "PO:0009010 seed"
metaData$TissueOntology[(metaData$BioProject == "PRJNA635147")] = "PO:0009010 seed"
metaData$TissueOntology[(metaData$BioProject == "PRJNA474441")] = "PO:0009010 seed"
metaData$TissueOntology[(metaData$BioProject == "PRJNA704067")] = "PO:0009010 seed"
metaData$TissueOntology[(metaData$BioProject == "PRJNA684572")] = "PO:0009010 seed"
metaData$TissueOntology[(metaData$BioProject == "PRJNA507243")] = "PO:0009010 seed" # embryos
metaData$TissueOntology[(metaData$BioProject == "PRJNA558807")] = "PO:0009010 seed" # embryos
metaData$TissueOntology[(metaData$BioProject == "PRJNA306014")] = "PO:0009010 seed"
metaData$TissueOntology[(metaData$BioProject == "PRJNA419694")] = "PO:0009010 seed"
metaData$TissueOntology[(metaData$BioProject == "PRJNA314076" & grepl("Seeds",metaData$LibraryName))] = "PO:0009010 seed"

# How many samples are annotated?
# table(metaData$TreatmentOntology)
# table(metaData$TissueOntology)
sum(table(metaData$TreatmentOntology))
sum(table(metaData$TreatmentOntology))/nrow(metaData) * 100

# Look at balance between categories
# table(metaData$TreatmentOntology, metaData$TissueOntology)

# How many studies are included in the dataset thus far?
length(unique(metaData[!is.na(metaData$TreatmentOntology),"BioProject"]))
length(unique(metaData[!is.na(metaData$TreatmentOntology),"PMID"]))
length(unique(metaData[!is.na(metaData$TreatmentOntology),"PMID"]))/length(unique(metaData[,"PMID"]))

```

# fix study ids
```{r}
metaData$PMID = gsub("^ ", "", metaData$PMID)
metaData$PMID = gsub("\t", ", ", metaData$PMID)

# count up final number of studies
length(unlist(strsplit(unique(metaData$PMID), ", ")))
```

# summarize sample annotations
```{r}
# write table of annotations
# write.csv(metaData, paste("arabidopsis_thaliana_rna_metadata_", date, ".csv", sep = ""), quote = F, row.names = F)

# Number of short day samples
length(metaData$TreatmentOntology[grepl("PECO:0007200", metaData$TreatmentOntology)])

# table of treatments for best sampled tissues
runCountByTreatmentAndTissue = table(metaData$TreatmentOntology, metaData$TissueOntology)
rownames(runCountByTreatmentAndTissue) = gsub(",", ";", rownames(runCountByTreatmentAndTissue))

write.csv(runCountByTreatmentAndTissue, paste("arabidopsis_thaliana_runCountByTreatmentAndTissue_", date, ".csv", sep = ""), quote = F)

# Experiment count by treatment and tissue
metaDataNoRunId = unique(subset(metaData, select = -c(Run)))
any(duplicated(metaDataNoRunId$Experiment)) # want FALSE
expCountByTreatmentAndTissue = table(metaDataNoRunId$TreatmentOntology, metaDataNoRunId$TissueOntology)

rownames(expCountByTreatmentAndTissue) = gsub(",", ";", rownames(expCountByTreatmentAndTissue))
write.csv(expCountByTreatmentAndTissue, paste("arabidopsis_thaliana_experimentCountByTreatmentAndTissue_", date, ".csv", sep = ""), quote = F)

# Study count by treatment and tissue
metaDataNoRunOrExpId = unique(subset(metaData, select = c(PMID, TissueOntology, TreatmentOntology)))
studyCountByTreatmentAndTissue = table(metaDataNoRunOrExpId$TreatmentOntology, metaDataNoRunOrExpId$TissueOntology)

rownames(studyCountByTreatmentAndTissue) = gsub(",", ";", rownames(studyCountByTreatmentAndTissue))
write.csv(studyCountByTreatmentAndTissue, paste("arabidopsis_thaliana_studyCountByTreatmentAndTissue_", date, ".csv", sep = ""), quote = F)
```

# collapse tissue labels
```{r}
metaData$TissueOntologyCollapsed = metaData$TissueOntology

# root
metaData$TissueOntologyCollapsed[(metaData$TissueOntologyCollapsed == "PO:0000025 root tip")] = "PO:0009005 root"
metaData$TissueOntologyCollapsed[(metaData$TissueOntologyCollapsed == "PO:0006085 root meristem")] = "PO:0009005 root"
metaData$TissueOntologyCollapsed[(metaData$TissueOntologyCollapsed == "PO:0006020 lateral root apical meristem")] = "PO:0009005 root"
metaData$TissueOntologyCollapsed[(metaData$TissueOntologyCollapsed == "PO:0020124 root stele")] = "PO:0009005 root"

# flower
metaData$TissueOntologyCollapsed[(metaData$TissueOntologyCollapsed == "PO:0000056 flower bud")] = "PO:0009046 flower"
metaData$TissueOntologyCollapsed[(metaData$TissueOntologyCollapsed == "PO:0009049 inflorescence")] = "PO:0009046 flower"
metaData$TissueOntologyCollapsed[(metaData$TissueOntologyCollapsed == "PO:0009073 stigma")] = "PO:0009046 flower"
metaData$TissueOntologyCollapsed[(metaData$TissueOntologyCollapsed == "PO:0020003 plant ovule")] = "PO:0009046 flower"
metaData$TissueOntologyCollapsed[(metaData$TissueOntologyCollapsed == "PO:0009066 anther")] = "PO:0009046 flower"
metaData$TissueOntologyCollapsed[(metaData$TissueOntologyCollapsed == "PO:0020122 inflorescence axis")] = "PO:0009046 flower"
metaData$TissueOntologyCollapsed[(metaData$TissueOntologyCollapsed == "PO:0025281 pollen")] = "PO:0009046 flower"
metaData$TissueOntologyCollapsed[(metaData$TissueOntologyCollapsed == "PO:0006204 sporocyte")] = "PO:0009046 flower"
metaData$TissueOntologyCollapsed[(metaData$TissueOntologyCollapsed == "PO:0030112 pedicel")] = "PO:0009046 flower"

# shoot
metaData$TissueOntologyCollapsed[(metaData$TissueOntologyCollapsed == "PO:0020148 shoot apical meristem")] = "PO:0009006 shoot system"
metaData$TissueOntologyCollapsed[(metaData$TissueOntologyCollapsed == "PO:0020142 internode")] = "PO:0009006 shoot system"
metaData$TissueOntologyCollapsed[(metaData$TissueOntologyCollapsed == "PO:0025059 portion of ground tissue")] = "PO:0009006 shoot system"
metaData$TissueOntologyCollapsed[(metaData$TissueOntologyCollapsed == "PO:0020144 apical meristem")] = "PO:0009006 shoot system"
metaData$TissueOntologyCollapsed[(metaData$TissueOntologyCollapsed == "PO:0020030 cotyledon, PO:0020148 shoot apical meristem")] = "PO:0009006 shoot system"
metaData$TissueOntologyCollapsed[(metaData$TissueOntologyCollapsed == "PO:0009047 stem")] = "PO:0009006 shoot system"
metaData$TissueOntologyCollapsed[(metaData$TissueOntologyCollapsed == "PO:0020100 hypocotyl")] = "PO:0009006 shoot system"
metaData$TissueOntologyCollapsed[(metaData$TissueOntologyCollapsed == "PO:0025291 seedling hypocotyl")] = "PO:0009006 shoot system"
metaData$TissueOntologyCollapsed[(metaData$TissueOntologyCollapsed == "PO:0005052 callus")] = "PO:0009006 shoot system"

# leaf
metaData$TissueOntologyCollapsed[(metaData$TissueOntologyCollapsed == "PO:0020030 cotyledon")] = "PO:0025034 leaf"

# reproductive parts: flower or fruit
metaData$TissueOntologyCollapsed[(metaData$TissueOntologyCollapsed == "PO:0009001 fruit")] = "PO:0009001 fruit OR PO:0009046 flower"
metaData$TissueOntologyCollapsed[(metaData$TissueOntologyCollapsed == "PO:0009046 flower")] = "PO:0009001 fruit OR PO:0009046 flower"

# summarize collapsed data
sort(table(metaData$TissueOntologyCollapsed), decreasing = T)

# table of treatments for best sampled tissues
runCountByTreatmentAndTissue = table(metaData$TreatmentOntology, metaData$TissueOntologyCollapsed)
rownames(runCountByTreatmentAndTissue) = gsub(",", ";", rownames(runCountByTreatmentAndTissue))

write.csv(runCountByTreatmentAndTissue, paste("arabidopsis_thaliana_runCountByTreatmentAndTissueCollapsed_", date, ".csv", sep = ""), quote = F)

# Study count by treatment and tissue
metaDataNoRunOrExpId = unique(subset(metaData, select = c(PMID, TissueOntologyCollapsed, TreatmentOntology)))
studyCountByTreatmentAndTissue = table(metaDataNoRunOrExpId$TreatmentOntology, metaDataNoRunOrExpId$TissueOntologyCollapsed)

rownames(studyCountByTreatmentAndTissue) = gsub(",", ";", rownames(studyCountByTreatmentAndTissue))
write.csv(studyCountByTreatmentAndTissue, paste("arabidopsis_thaliana_studyCountByTreatmentAndTissueCollapsed_", date, ".csv", sep = ""), quote = F)

```

# Print sample sizes for top categories
```{r}
sort(colSums(runCountByTreatmentAndTissue), decreasing = T)

sort(rowSums(runCountByTreatmentAndTissue), decreasing = T)[1:5]
```

# Write list of SRA runs
```{r}
# save list of remaining SRA runs
write.table(metaData$Run, "rnaSra.txt", quote = F, row.names = F, col.names = F)
```

# Save metadata
```{r}
write.csv(metaData, paste("rnaseq_metadata_arabidopsis_thaliana_", Sys.Date(), ".csv", sep = ""), row.names = F)
```

