---
title: "surrogate variable and differential expression analysis"
author: "Miles Roberts"
date: "2022-06-14"
output: html_document
---

#Load packages
```{r}
# install.packages("BiocManager")
# library(BiocManager)
# BiocManager::install("sva") # install sva package
# BiocManager::install("limma") # install limma package
# BiocManager::install("genefilter")
# BiocManager::install("M3C")
# install.packages("tidyverse")
# install.packages("data.table")

rm(list = ls())
gc()
library(sva)
library(limma)
#library(tidyverse)
library(data.table)
library(R.utils)
library(reshape)
library(ggplot2)
# library(genefilter)
library(M3C)
library(Rfast)
library(ppcor)
library(gridExtra)
library(ggpubr)
library(Polychrome) # to create large color palettes

# set working directory for easy access of data
notebookDirectory = getwd()

# using caterpie color palette
myColorPalette = c("#ffb439", "#ff7b41", "#7b394a", "#52624a", "#e6cd94", "#acf641")

```

# Misc Functions
```{r}
# harmonic series
# Used to calculate watterson's theta
harmonicSeries = function(x){
  sum(1/(1:(x-1)))
}

# squared harmonic series
squaredHarmonicSeries = function(x){
  sum(1/((1:(x-1))^2))
}

# squaredHarmonicSeries(5)

# tajima's D variance
tajimasDeltaVar = function(n,S){
  # split up terms of this formula so that it's easier to read
  a = ( ((n+1)/(3*(n-1)) - 1/harmonicSeries(n)) / harmonicSeries(n))
  b = 2*(n^2 + n + 3)/(9*n*(n-1))
  c = (n+2)/(n*harmonicSeries(n))
  d = squaredHarmonicSeries(n)/(harmonicSeries(n)^2)
  e = harmonicSeries(n)^2 + squaredHarmonicSeries(n)
  result = sqrt(S*a + ((b - c + d)/e)*S*(S-1))
  return(result)
}

# tajimasDeltaVar(50, 100)
  
```

# Load genes
```{r}
# load gene intervals
genes = read.table("arabidopsis_thaliana_genes_overlapping.bed", header = F)
```

# Load variant data, calculate watterson's theta
## Only need to run this once, then save the output
```{r eval=FALSE, include=FALSE}
# Load variant counts
variantCounts = list.files(pattern="*frq.count")
variantCounts = lapply(variantCounts, fread, skip = 1, fill = T)
variantCounts = do.call("rbind", variantCounts)
# variantCounts = fread("arabidopsisThalianaJointGenotypes_variantAndInvariant_filtered_ChrAll.frq.count", skip = 1, fill = T, colClasses = c("character", "numeric", "numeric", "numeric", "numeric", "numeric"), data.table = F)

# add column names
names(variantCounts) = c("CHROM", "POS", "N_ALLELES", "N_CHR", "COUNT_REF", "COUNT_ALT")

# convert counts to frequencies
# variantCounts$FREQ_REF = variantCounts$COUNT_REF/variantCounts$N_CHR
# variantCounts$FREQ_ALT = 1-variantCounts$FREQ_REF # minor allele frequency is 1 - major allele frequency

# convert frequencies to heterozygosities
# variantCounts$HET = (variantCounts$N_CHR/(variantCounts$N_CHR - 1))*(1 - (variantCounts$FREQ_REF^2 + variantCounts$FREQ_ALT^2))

# split gene frame into list, hopefully will speed up for loop
genesList = split(genes, seq(nrow(genes)))
genesList = lapply(genesList, unlist) # convert one row dataframes to vectors

# output for pi and watterson's theta loop
# piList = list()
# piList = vector(mode = "list", length = nrow(genes))
# thetaList = list()
thetaList = vector(mode = "list", length = nrow(genes)) # watterson's theta
sList = vector(mode = "list", length = nrow(genes)) # number of segregating sites
nList = vector(mode = "list", length = nrow(genes)) # average sample size (i.e. number of chromsomes sampled) across gene

# Loop over each gene
i = 1
for(gene in genesList){
	# gene coordinates
  # genesRow = genes[i,]
	chr = gene[1]
	start = as.numeric(gene[2])
	end = as.numeric(gene[3])
	name = gene[4]
	
	# Get list of variants in gene interval
	geneSites = variantCounts[((variantCounts$CHROM == chr) & (variantCounts$POS >= start) & (variantCounts$POS <= end)),]

	# Sum heterozygosities across sites to calculate nucleotide diversity
	# do NOT divide by number of sites to get pi-per-site, Tajima's D requires straight pi
	# piPerSite = sum(geneSites$V9)

	# Subset to just segregating sites for calculating tajima's D	
	# geneSites = geneSites[(geneSites$V3 >= 2),]

	# For each segregating site, calculate harmonic series term
	# sum up terms to get wattersonsTheta for locus
	sampleSizes = geneSites$N_CHR
	wattersonsTheta = sum(1/unlist(lapply(sampleSizes, harmonicSeries)))
	
	# save output to list
	# piList[[i]] = piPerSite
	thetaList[[i]] = wattersonsTheta
	sList[[i]] = length(sampleSizes)
	nList[[i]] = mean(sampleSizes)
	
	# iterate index
	i = i + 1
}

# Save results
genes$thetaW = unlist(thetaList)
genes$segSites = unlist(sList)
genes$numChrSampled = unlist(nList)

write.table(genes, "arabidopsis_thaliana_genes_thetaW.txt", row.names = F, quote = F)
```

# Load and calculate other gene features
```{r}
# load wattersons theta from above chunk
thetaW = read.table("arabidopsis_thaliana_genes_thetaW.txt", header = T)

# load GC content, length, and methylation site information
gcContent = list.files(pattern = "arabidopsis_thaliana_gcContent.*")
gcContent = lapply(gcContent, read.table, header = F)
gcContent = Reduce(function(dtf1, dtf2) merge(dtf1, dtf2, by = c("V1", "V2", "V3", "V4", "V5", "V6", "V7", "V8", "V9", "V10", "V11", "V12", "V13"), all.x = TRUE), gcContent)

# gcContent = read.table("arabidopsis_thaliana_gcContent.txt", header = F)
names(gcContent) = c("chr", "start", "end", "name", "AT", "GC", "numA", "numC", "numG", "numT", "numN", "numOther", "length", "CAA", "CAC", "CAG", "CAT", "CCA", "CCC", "CCG", "CCT", "CG", "CTA", "CTC", "CTG", "CTT")

# load pixy output
pin = list.files(pattern = "pixy_missenseAndInvariant_Chr*")
pis = list.files(pattern = "pixy_synonymousAndInvariant_Chr*")
pia = list.files(pattern = "pixy_allSites_Chr*")

pin = lapply(pin, read.table, header = T)
pin = do.call("rbind", pin)

pis = lapply(pis, read.table, header = T)
pis = do.call("rbind", pis)

pia = lapply(pia, read.table, header = T)
pia = do.call("rbind", pia)

# Convert pi-per-site to total pi, feeds into tajima's D calculation
pin$total_pi_n = pin$avg_pi*pin$no_sites
pis$total_pi_s = pis$avg_pi*pis$no_sites
pia$total_pi = pia$avg_pi*pia$no_sites

# merge each measure of pi to gene list
names(genes)[1:4] = c("chr", "start", "end", "name")
names(pin)[2:4] = c("chr", "start", "end")
names(pis)[2:4] = c("chr", "start", "end")
names(pia)[2:4] = c("chr", "start", "end")

genes = merge(genes, pin[,c("chr", "start", "end", "total_pi_n")], by = c("chr", "start", "end"), all.x = T)
genes = merge(genes, pis[,c("chr", "start", "end", "total_pi_s")], by = c("chr", "start", "end"), all.x = T)
genes = merge(genes, pia[,c("chr", "start", "end", "total_pi")], by = c("chr", "start", "end"), all.x = T)

# merge thetaW into gene list
names(thetaW)[1:4] = c("chr", "start", "end", "name")
genes = merge(genes, thetaW, by = c("chr", "start", "end", "name"), all.x = T)

# merge gc content into gene list
genes = merge(genes, gcContent[,c("name", "chr", "start", "end", "GC", "length", "CAA", "CAC", "CAG", "CAT", "CCA", "CCC", "CCG", "CCT", "CG", "CTA", "CTC", "CTG", "CTT")], by = c("chr", "start", "end", "name"), all.x = T)

# convert methlation sites into percentages
# count of sites/number of windows
# For trinucleotide sites, # windows = length - 2
# For dinucleotide sites, # windows = length - 1
genes$CAApct = genes$CAA/(genes$length - 2)
genes$CACpct = genes$CAC/(genes$length - 2)
genes$CAGpct = genes$CAG/(genes$length - 2)
genes$CATpct = genes$CAT/(genes$length - 2)
genes$CCApct = genes$CCA/(genes$length - 2)
genes$CCCpct = genes$CCC/(genes$length - 2)
genes$CCGpct = genes$CCG/(genes$length - 2)
genes$CCTpct = genes$CCT/(genes$length - 2)
genes$CGpct = genes$CG/(genes$length - 1)
genes$CTApct = genes$CTA/(genes$length - 2)
genes$CTCpct = genes$CTC/(genes$length - 2)
genes$CTGpct = genes$CTG/(genes$length - 2)
genes$CTTpct = genes$CTT/(genes$length - 2)

genes$CHHpct = rowSums(genes[,c("CAA", "CAC", "CAT", "CCA", "CCC", "CCT", "CTA", "CTC", "CTT")])/(genes$length - 2)
genes$CHGpct = rowSums(genes[,c("CAG", "CCG", "CTG")])/(genes$length - 2)

# tajima's D calculation
# I'll only calculate the numerator, because the denominator simply makes tajima's D comparable between studies
genes$tajimaDelta = genes$total_pi - genes$thetaW

# remove genes with sample size of zero
genes = genes[(genes$segSites > 0),]

# denominator for Tajima's D
genes$tajimaVar = mapply(tajimasDeltaVar, n = genes$numChrSampled, S = genes$segSites)

# delta/variance -> scaled tajima's D
genes$tajimasD = genes$tajimaDelta/genes$tajimaVar
summary(genes$tajimasD)

# load polymorphism counts
synCounts = list.files(pattern = "arabidopsis_thaliana_synonymousCountsByGene_Chr*")
synCounts = lapply(synCounts, read.table)
synCounts = do.call("rbind", synCounts)
names(synCounts) = c("chr", "start", "end", "name", "Ps")

nonsynCounts = list.files(pattern = "arabidopsis_thaliana_nonsynonymousCountsByGene_Chr*")
nonsynCounts = lapply(nonsynCounts, read.table)
nonsynCounts = do.call("rbind", nonsynCounts)
names(nonsynCounts) = c("chr", "start", "end", "name", "Pn")

genes = merge(genes, synCounts, by = c("chr", "start", "end", "name"), all.x = T)
genes = merge(genes, nonsynCounts, by = c("chr", "start", "end", "name"), all.x = T)

# load dn/ds values
# dndsMLcodeml = read.table("arabidopsis_thaliana_v_arabidopsis_lyrata_Codeml_ML_dnds_withGeneIDs.txt", header = T, fill = T)
dndsBiopython = list.files(path = "./biopythonDnDs", pattern = "*.dnds", full.names = T)
dndsBiopython = lapply(dndsBiopython, fread, header = T)
dndsBiopython = do.call("rbind", dndsBiopython)

# prepare to merge
# names(dndsMLcodeml)[1] = "name"
# dndsMLcodeml$name = gsub("\\.[0-9]*", "", dndsMLcodeml$name)
names(dndsBiopython)[2] = "name"
dndsBiopython$name = gsub("\\.[0-9]*", "", dndsBiopython$name)

# genes = merge(genes, dndsMLcodeml[,c("name", "N", "S", "dn", "ds", "dnds")], all.x = T, by = "name")
genes = merge(genes, dndsBiopython, all.x = T, by = "name")

# back calculate number of nonsynonymous differences
# genes$Dn = genes$N*genes$dn
# genes$Ds = genes$S*genes$ds

# calculate neutrality index
#genes$NI = (genes$Pn/genes$Dn)/(genes$Ps/genes$Ds)
genes$NI = (genes$Pn/genes$Nd)/(genes$Ps/genes$Sd)

# calculate alpha (1 - NI)
genes$alpha = 1 - genes$NI

# direction of selection
genes$DoS = genes$Nd/(genes$Nd + genes$Sd) - genes$Pn/(genes$Pn + genes$Ps)

# load counts of nonysnonymous sites for every gene (not just genes with 1:1 orthologs in A. lyrata)
Nsites = read.table("arabidopsis_thaliana_nonsynonymousSitesForEveryGene.txt", header = T)
names(Nsites) = c("name", "Nall", "Sall", "length")
Nsites$name = gsub("\\.[0-9]*", "", Nsites$name)

genes = merge(genes, Nsites[,c("name", "Nall", "Sall")], by = "name", all.x = T)

# sanity check, number of sites should be greater or equal to number of sites in ortholog alignments
all(genes$Nall + genes$Sall >= genes$N + genes$S, na.rm = T)

# calculate piN, piS, piN/piS
genes$piN = genes$total_pi_n/genes$Nall
genes$piS = genes$total_pi_s/genes$Sall
genes$piNpiS = genes$piN/genes$piS

# Load orthogroup data
ortho = read.table("Orthogroups.tsv", header = T, sep = "\t")
names(ortho) = c("Orthogroup", "Arabidopsis_lyrata", "Arabidopsis_thaliana")
ortho$Arabidopsis_thaliana = gsub("\\.[0-9]*", "", ortho$Arabidopsis_thaliana)

# For each A. thaliana gene, find it's orthogroup and count the number of A. thaliana genes in said orthogroup
familySizes = list()
i = 1
for(gene in genes$name){
  athalGeneList = ortho[which(grepl(pattern = gene, ortho$Arabidopsis_thaliana)),"Arabidopsis_thaliana"]
  
  # If gene is not in orthogroup list, it has a family size of one
  if(length(athalGeneList) > 0){
    familySizes[[i]] = length(unlist(strsplit(athalGeneList, ",")))
  } else{
    familySizes[[i]] = 1
  }
  
  i = i + 1
}

# Add gene family size to feature table
genes$familySize = unlist(familySizes)
```

# summarize features
```{r}
summary(genes[,c("length", "GC", "familySize", "dN", "dS", "dNdS", "piN", "piS","piNpiS", "total_pi", "thetaW", "tajimasD", "NI", "alpha", "DoS", "CGpct", "CHGpct", "CHHpct")])
```

# compare dn/ds between biopython and codeml
```{r}
# load dnds from Nei Gojobori 1986 implementation in codeml
dndsNGcodeml = read.table("arabidopsis_thaliana_v_arabidopsis_lyrata_Codeml_NeiAndGojobori_dnds.txt", header = T)

# load dnds from my Nei Gojobori 1986 implementation in biopython
dndsBiopython = list.files(path = "./biopythonDnDs", pattern = "*.dnds", full.names = T)
dndsBiopython = lapply(dndsBiopython, fread, header = T)
dndsBiopython = do.call("rbind", dndsBiopython)

# merge for comparisons
names(dndsBiopython)[2] = "gene"
dndsCompare = merge(dndsNGcodeml, dndsBiopython, by = "gene", all.x = T, all.y = T)

# remove infinities
dndsCompare = dndsCompare[(is.finite(dndsCompare$dNdS)),]

# remove -1's
dndsCompare = dndsCompare[(dndsCompare$dnds > 0),]

# Correlations
cor(dndsCompare$dnds, dndsCompare$dNdS, use = "pairwise.complete.obs")
cor(dndsCompare$dn, dndsCompare$dN, use = "pairwise.complete.obs")
cor(dndsCompare$ds, dndsCompare$dS, use = "pairwise.complete.obs")

plot(dndsCompare$dnds, dndsCompare$dNdS, xlab = "dN/dS codeml", ylab = "dN/dS Biopython")
abline(a = 0, b = 1)

plot(dndsCompare$dn, dndsCompare$dN, xlab = "dN codeml", ylab = "dN Biopython")
abline(a = 0, b = 1)

plot(dndsCompare$ds, dndsCompare$dS, xlab = "dS codeml", ylab = "dS Biopython")
abline(a = 0, b = 1)
```

# Load expression matrix and meta data
```{r}
# read in expression data
mdata <- data.frame(fread("expressionMatrixCbind_NumReads.csv.gz", header=T))
rownames(mdata) <- gsub("\\.[0-9]*", "", mdata$Name) # Reset row names to gene names, needed for SVA, also remove isoform id so that merging with pixy data is easy

mdata = mdata[,which(colnames(mdata) != "Name")]
#mdata[1:10,1:10]

# read in phenotype data
pheno <- read.csv("rnaseq_metadata_arabidopsis_thaliana_2022-07-18.csv", header=T) 

# Check for duplicates. want FALSE
any(duplicated((pheno$Run)))

# Check all phenotype data has expression data. Want TRUE
all(colnames(mdata) %in% pheno$Run)
mdata = mdata[,(colnames(mdata) %in% pheno$Run)]
all(colnames(mdata) %in% pheno$Run)

all(pheno$Run %in% colnames(mdata))
pheno = pheno[(pheno$Run %in% colnames(mdata)),]
all(pheno$Run %in% colnames(mdata))
```

# Outlier removal
## Apply simple filters
```{r}
### SAMPLES OR GENES WITH LOW OR NO EXPRESSION ###
tranCountPerGene = rowSums(mdata)
hist(tranCountPerGene)
summary(tranCountPerGene)
#quantile(tranCountPerGene, 0.01)

tranCountPerRun = colSums(mdata)
hist(tranCountPerRun)
summary(tranCountPerRun)
#quantile(tranCountPerRun, 0.02)

# Remove genes with no expression, treatment-specificity would be undefined - so they wouldn't be included in analysis anyway
# Remove runs with no mapped transcripts - probably generated via some kind of error
mdata = mdata[(tranCountPerGene > 0),] # dropped 27537 - 27519 = 18 genes
mdata = mdata[,(tranCountPerRun > 0)] # dropped 24011 - 23796 = 215 runs
pheno = pheno[which(pheno$Run %in% colnames(mdata)),]

# Remove runs with really low mapping rates
# Which samples have low mapping rates?
mapRates = read.table("salmonMappingRates.txt", header = F)
summary(mapRates$V2)
quantile(mapRates$V2, 0.001)
hist(mapRates$V2, xlab = "Percent mapping rate", main = "Histogram of mapping rates for RNA-seq runs")
lowMapSamples = mapRates$V1[mapRates$V2 < 1]

# Remove samples with really low mapping rates
mdata = mdata[,!(colnames(mdata) %in% lowMapSamples)] # 423 runs dropped
pheno = pheno[which(pheno$Run %in% colnames(mdata)),]
```

# Load TPM data for later, subset by all previous filters on count data
```{r}
# Load TPM data, controls for differences in coverage between samples
tpmdata <- data.frame(fread("expressionMatrixCbind_TPM.csv.gz", header=T))
rownames(tpmdata) <- gsub("\\.[0-9]*", "", tpmdata$Name)

# Subset TPM data based on filters in above chunks
tpmdata = tpmdata[(rownames(tpmdata) %in% rownames(mdata)),(colnames(tpmdata) %in% colnames(mdata))]
tpmdata = tpmdata[,which(colnames(tpmdata) != "Name")]

# Average across technical replicates
all(colnames(tpmdata) == pheno$Run)
pheno = pheno[order(match(pheno$Run, colnames(tpmdata))),] # re-order rows of metadata to match columns of gene expression matrix
all(colnames(tpmdata) == pheno$Run) # Want TRUE

colnames(tpmdata) = pheno$Experiment # Swap run ids for experiment ids
nms = unique(colnames(tpmdata)) # average across columns with matching experiment ids
tpmdata = sapply(nms, function(x) rowMeans(tpmdata[(colnames(tpmdata) == x)]))

# Make sure phenotype file matches tpmdata
pheno = unique(pheno[,c("PMID", "BioProject", "Experiment", "TreatmentOntology", "TissueOntologyCollapsed")])
nrow(pheno) == ncol(tpmdata) # want TRUE
```

# plot sample sizes for different treatment-tissue categories
```{r eval=FALSE, include=FALSE}
treatByTissueTable = table(pheno$TreatmentOntology, pheno$TissueOntologyCollapsed)
treatByTissueTable = melt(treatByTissueTable)

# subset to just the top six treatment cateogries
topTreats = names(sort(table(pheno$TreatmentOntology), decreasing = T)[1:6])
treatByTissueTable = treatByTissueTable[(treatByTissueTable$Var.1 %in% topTreats),]

names(treatByTissueTable) = c("Treatment", "Tissue", "Count")

ggplot(treatByTissueTable, aes(x = Treatment, y = Count, color = Tissue, fill = Tissue)) +
  geom_bar(position = "stack", stat = "identity") +
  theme_classic() + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1)) +
  scale_color_manual(values = myColorPalette)

```



# Dimension reduction, without batch correction
```{r}
# perform PCA
pcaPreSva <- prcomp(t(tpmdata), center = T, scale = T) # run PCA
pcaPreSva$x[1:5,1:5] # visualize matrix of principle components (PCs)
var_explained <- pcaPreSva$sdev^2/sum(pcaPreSva$sdev^2) # explained variance for each PC
var_explained[1:5]

pcaPreSva = as.data.frame(pcaPreSva$x)

# plot PCA

ggplot(pcaPreSva, aes(x=PC1,y=PC2, col=pheno$TissueOntologyCollapsed)) + 
  geom_point() +
  theme_classic() +
  scale_color_manual(values = myColorPalette) +
  #scale_color_manual(values = c("#ffff4d", "#e6f261", "","#8cbf00"))
  labs(x=paste("PC1: ",round(var_explained[1]*100,2),"%"),
       y=paste("PC2: ",round(var_explained[2]*100,2),"%"),
       color = "Tissue")

ggsave("pcaBySample_Tissue.jpeg")

# ggplot(pcaPreSva, aes(x=PC1,y=PC2, col=pheno$envV3)) + 
#   geom_point() +
#   theme_classic() +
#   stat_ellipse(type = "norm", level = 0.99) +
#   labs(x=paste("PC1: ",round(var_explained[1]*100,2),"%"),
#        y=paste("PC2: ",round(var_explained[2]*100,2),"%"),
#        color="Treatment")
# 
# ggsave("pcaBySample_Treatment.jpeg", width = 10, height = 4.5, units = "in")

# plot tSNE using M3C package
# tsne(tpmdata, labels = as.factor(pheno$tissue))

# plot MDS using limma package
```

# Break up data by tissue, Measure treatment specificity
```{r}
# break up data by tissue type
leafPheno = pheno[(pheno$TissueOntologyCollapsed == "PO:0025034 leaf"),]
leafTPMdata = tpmdata[,(colnames(tpmdata) %in% leafPheno$Experiment)]

rootPheno = pheno[(pheno$TissueOntologyCollapsed == "PO:0009005 root"),]
rootTPMdata = tpmdata[,(colnames(tpmdata) %in% rootPheno$Experiment)]
dim(rootPheno)
dim(rootTPMdata)

plantPheno = pheno[(pheno$TissueOntologyCollapsed == "PO:0000003 whole plant"),]
plantTPMdata = tpmdata[,(colnames(tpmdata) %in% plantPheno$Experiment)]
dim(plantPheno)
dim(plantTPMdata)

shootPheno = pheno[(pheno$TissueOntologyCollapsed == "PO:0009006 shoot system"),]
shootTPMdata = tpmdata[,(colnames(tpmdata) %in% shootPheno$Experiment)]
dim(shootPheno)
dim(shootTPMdata)

seedPheno = pheno[(pheno$TissueOntologyCollapsed == "PO:0009010 seed"),]
seedTPMdata = tpmdata[,(colnames(tpmdata) %in% seedPheno$Experiment)]
dim(seedPheno)
dim(seedTPMdata)

flowerPheno = pheno[(pheno$TissueOntologyCollapsed == "PO:0009001 fruit OR PO:0009046 flower"),]
flowerTPMdata = tpmdata[,(colnames(tpmdata) %in% flowerPheno$Experiment)]
dim(flowerPheno)
dim(flowerTPMdata)

# check that data partitioned perfectly, want TRUE
nrow(leafPheno) + nrow(rootPheno) + nrow(plantPheno) + nrow(shootPheno) + nrow(seedPheno) + nrow(flowerPheno) == nrow(pheno)

# Function to calculate expression-specificity index
# index = 0 -> non-environment/non-tissue specific expression
# index = 1 -> environment/tissue-specific expression
specIndex = function(x){
  sum(1-(x/max(x, na.rm = T)), na.rm = T)/(length(x) - 1)
}

# Function to loop over treatments represented for each tissue, and calculate treatment profile: average expression in each treatment for each gene
treatmentProfile = function(myPheno, myMdata){
  # Matrix to store outputs
  environments = unique(myPheno$TreatmentOntology)
  genes = rownames(myMdata)
  expEnv = matrix(NA, nrow = length(genes), ncol = length(environments), dimnames = list(genes, environments))

  ### ENVIRONMENT EXPRESSION PROFILE ###
  # generate expression profiles for every gene
  # Expression profile = median expression of each gene in each environment
  for(environment in environments){
    # print(environment)
    #subset data from one environment
	  exps = unique(myPheno[(myPheno$TreatmentOntology == environment), "Experiment"])
	  tpmdataSub = as.matrix(myMdata[,exps])
	  # calculate mean expression of every gene in this environment
	  # envMedians = apply(tpmdataSub, MARGIN = 1, FUN = median)
	  envMeans = rowMeans(tpmdataSub, na.rm = T)
	  expEnv[,(colnames(expEnv) == environment)] = envMeans
  }

  # remove genes with mean of zero in all environments
  # these genes will have an undefined specificity-index (1 - 0/0)
  expEnv = expEnv[(rowSums(expEnv) != 0),]
  
  return(expEnv)
  
}

# calculate treatment profiles for each tissue
leafProfile = treatmentProfile(leafPheno, leafTPMdata)
rootProfile = treatmentProfile(rootPheno, rootTPMdata)
plantProfile = treatmentProfile(plantPheno, plantTPMdata)
shootProfile = treatmentProfile(shootPheno, shootTPMdata)
seedProfile = treatmentProfile(seedPheno, seedTPMdata)
flowerProfile = treatmentProfile(flowerPheno, flowerTPMdata)

# Calculate specificity index from treatment profile
leafSpec = as.data.frame(apply(leafProfile, MARGIN = 1, FUN = specIndex))
leafSpec$Name = rownames(leafSpec)
names(leafSpec) = c("leafEnvSpecIndex", "name")

rootSpec = as.data.frame(apply(rootProfile, MARGIN = 1, FUN = specIndex))
rootSpec$Name = rownames(rootSpec)
names(rootSpec) = c("rootEnvSpecIndex", "name")

plantSpec = as.data.frame(apply(plantProfile, MARGIN = 1, FUN = specIndex))
plantSpec$Name = rownames(plantSpec)
names(plantSpec) = c("plantEnvSpecIndex", "name")

shootSpec = as.data.frame(apply(shootProfile, MARGIN = 1, FUN = specIndex))
shootSpec$Name = rownames(shootSpec)
names(shootSpec) = c("shootEnvSpecIndex", "name")

seedSpec = as.data.frame(apply(seedProfile, MARGIN = 1, FUN = specIndex))
seedSpec$Name = rownames(seedSpec)
names(seedSpec) = c("seedEnvSpecIndex", "name")

flowerSpec = as.data.frame(apply(flowerProfile, MARGIN = 1, FUN = specIndex))
flowerSpec$Name = rownames(flowerSpec)
names(flowerSpec) = c("flowerEnvSpecIndex", "name")

# visualize specificity with histograms
# hist(leafSpec$leafEnvSpecIndex)
# hist(rootSpec$rootEnvSpecIndex)
# hist(plantSpec$plantEnvSpecIndex)
# hist(shootSpec$shootEnvSpecIndex)
# hist(seedSpec$seedEnvSpecIndex)
# hist(flowerSpec$flowerEnvSpecIndex)

# Plot all distributions of treatment specificity at once
specData = data.frame(tissue = as.factor(c(rep("PO:0000003 whole plant", times = nrow(plantSpec)), rep("PO:0009001 fruit OR PO:0009046 flower", times = nrow(flowerSpec)), rep("PO:0009005 root", times = nrow(rootSpec)), rep("PO:0009006 shoot system", times = nrow(shootSpec)), rep("PO:0009010 seed", times = nrow(seedSpec)), rep("PO:0025034 leaf", times = nrow(leafSpec)))), specificity = c(plantSpec$plantEnvSpecIndex, flowerSpec$flowerEnvSpecIndex, rootSpec$rootEnvSpecIndex, shootSpec$shootEnvSpecIndex, seedSpec$seedEnvSpecIndex, leafSpec$leafEnvSpecIndex))

# ggplot(specData, aes(x = specificity, color = tissue)) + 
#   theme_classic() +
#   labs(x = "Treatment-specificity index (Tau)", y = "Density") +
#   scale_color_manual(values=myColorPalette) +
#   # stat_density(aes(x=specificity, color = tissue), geom = "line")
#   geom_density()

ggplot(specData, aes(x = tissue, y = specificity, color = tissue)) + 
  theme_classic() +
  theme(axis.ticks.x = element_blank(), axis.text.x = element_blank())+
  labs(x = "", y = "Treatment-specificity index (Tau)") +
  scale_color_manual(values=myColorPalette) +
  # stat_density(aes(x=specificity, color = tissue), geom = "line")
  geom_violin() +
  geom_boxplot(width = 0.1)

ggsave("densityPlotsOfTreatmentSpecificity.jpeg")

# merge specificity with gene feature table, don't drop any genes
genes = merge(genes, leafSpec, by = "name", all.x = T)
genes = merge(genes, rootSpec, by = "name", all.x = T)
genes = merge(genes, plantSpec, by = "name", all.x = T)
genes = merge(genes, shootSpec, by = "name", all.x = T)
genes = merge(genes, seedSpec, by = "name", all.x = T)
genes = merge(genes, flowerSpec, by = "name", all.x = T)
```


# calculate average expression, not ignoring zeros
```{r}
# Calculate average expression for each tissue
leafAvgExp = as.data.frame(apply(leafTPMdata, MARGIN = 1, FUN = mean))
leafAvgExp$name = rownames(leafAvgExp)
names(leafAvgExp) = c("leafAvgExpWithZeros", "name")

rootAvgExp = as.data.frame(apply(rootTPMdata, MARGIN = 1, FUN = mean))
rootAvgExp$name = rownames(rootAvgExp)
names(rootAvgExp) = c("rootAvgExpWithZeros", "name")

plantAvgExp = as.data.frame(apply(plantTPMdata, MARGIN = 1, FUN = mean))
plantAvgExp$name = rownames(plantAvgExp)
names(plantAvgExp) = c("plantAvgExpWithZeros", "name")

shootAvgExp = as.data.frame(apply(shootTPMdata, MARGIN = 1, FUN = mean))
shootAvgExp$name = rownames(shootAvgExp)
names(shootAvgExp) = c("shootAvgExpWithZeros", "name")

seedAvgExp = as.data.frame(apply(seedTPMdata, MARGIN = 1, FUN = mean))
seedAvgExp$name = rownames(seedAvgExp)
names(seedAvgExp) = c("seedAvgExpWithZeros", "name")

flowerAvgExp = as.data.frame(apply(flowerTPMdata, MARGIN = 1, FUN = mean))
flowerAvgExp$name = rownames(flowerAvgExp)
names(flowerAvgExp) = c("flowerAvgExpWithZeros", "name")

# merge average expression with gene feature table, don't drop any genes
genes = merge(genes, leafAvgExp, by = "name", all.x = T)
genes = merge(genes, rootAvgExp, by = "name", all.x = T)
genes = merge(genes, plantAvgExp, by = "name", all.x = T)
genes = merge(genes, shootAvgExp, by = "name", all.x = T)
genes = merge(genes, seedAvgExp, by = "name", all.x = T)
genes = merge(genes, flowerAvgExp, by = "name", all.x = T)

# plot correlations between expression level and treatment-specificity
plotLevelTauComp = function(tau, avg, tissue, myCor){
  ggplot(genes, aes_string(x = tau, y = avg)) +
    geom_point() +
    theme_classic() +
    labs(x = "treatment specificity of expression", y = "Expression level (TPM)", title = paste(tissue, " (", round(myCor, 2), ")", sep = ""))
}

leafTauLevelCor = cor(genes$leafEnvSpecIndex, genes$leafAvgExpWithZeros, use = "pairwise.complete.obs", method = "spearman")
plotLevelTauComp("leafEnvSpecIndex", "leafAvgExpWithZeros", "leaf", leafTauLevelCor)

rootTauLevelCor = cor(genes$rootEnvSpecIndex, genes$rootAvgExpWithZeros, use = "pairwise.complete.obs", method = "spearman")
plotLevelTauComp("rootEnvSpecIndex", "rootAvgExpWithZeros", "root", rootTauLevelCor)

plantTauLevelCor = cor(genes$plantEnvSpecIndex, genes$plantAvgExpWithZeros, use = "pairwise.complete.obs", method = "spearman")
plotLevelTauComp("plantEnvSpecIndex", "plantAvgExpWithZeros", "plant", plantTauLevelCor)

shootTauLevelCor = cor(genes$shootEnvSpecIndex, genes$shootAvgExpWithZeros, use = "pairwise.complete.obs", method = "spearman")
plotLevelTauComp("shootEnvSpecIndex", "shootAvgExpWithZeros", "shoot", shootTauLevelCor)

seedTauLevelCor = cor(genes$seedEnvSpecIndex, genes$seedAvgExpWithZeros, use = "pairwise.complete.obs", method = "spearman")
plotLevelTauComp("seedEnvSpecIndex", "seedAvgExpWithZeros", "seed", seedTauLevelCor)

flowerTauLevelCor = cor(genes$flowerEnvSpecIndex, genes$flowerAvgExpWithZeros, use = "pairwise.complete.obs", method = "spearman")
plotLevelTauComp("flowerEnvSpecIndex", "flowerAvgExpWithZeros", "flower", flowerTauLevelCor)

# plot(genes$leafEnvSpecIndex, genes$leafAvgExpWithZeros)
# cor(genes$leafEnvSpecIndex, genes$leafAvgExpWithZeros, use = "pairwise.complete.obs", method = "spearman")
```


# Calculate average expression, split by tissue, ignoring zeros
```{r}
# function to calculate mean, excluding zeros
meanExcludeZeros = function(x){
  mean(x[(x > 5)])
}

# subset to only control condition, hopefully will break-up correlation between m
# controlExperiments = pheno[(pheno$TreatmentOntology == "PECO:0001062 control exposure"),"Experiment"]

# Calculate average expression for each tissue
leafAvgExp = as.data.frame(apply(leafTPMdata, MARGIN = 1, FUN = meanExcludeZeros))
# leafAvgExp = as.data.frame(apply(leafTPMdata[,(colnames(leafTPMdata) %in% controlExperiments)], MARGIN = 1, FUN = meanExcludeZeros))
leafAvgExp$name = rownames(leafAvgExp)
names(leafAvgExp) = c("leafAvgExpNoZeros", "name")

rootAvgExp = as.data.frame(apply(rootTPMdata, MARGIN = 1, FUN = meanExcludeZeros))
rootAvgExp$name = rownames(rootAvgExp)
names(rootAvgExp) = c("rootAvgExpNoZeros", "name")

plantAvgExp = as.data.frame(apply(plantTPMdata, MARGIN = 1, FUN = meanExcludeZeros))
plantAvgExp$name = rownames(plantAvgExp)
names(plantAvgExp) = c("plantAvgExpNoZeros", "name")

shootAvgExp = as.data.frame(apply(shootTPMdata, MARGIN = 1, FUN = meanExcludeZeros))
shootAvgExp$name = rownames(shootAvgExp)
names(shootAvgExp) = c("shootAvgExpNoZeros", "name")

seedAvgExp = as.data.frame(apply(seedTPMdata, MARGIN = 1, FUN = meanExcludeZeros))
seedAvgExp$name = rownames(seedAvgExp)
names(seedAvgExp) = c("seedAvgExpNoZeros", "name")

flowerAvgExp = as.data.frame(apply(flowerTPMdata, MARGIN = 1, FUN = meanExcludeZeros))
flowerAvgExp$name = rownames(flowerAvgExp)
names(flowerAvgExp) = c("flowerAvgExpNoZeros", "name")

# merge average expression with gene feature table, don't drop any genes
genes = merge(genes, leafAvgExp, by = "name", all.x = T)
genes = merge(genes, rootAvgExp, by = "name", all.x = T)
genes = merge(genes, plantAvgExp, by = "name", all.x = T)
genes = merge(genes, shootAvgExp, by = "name", all.x = T)
genes = merge(genes, seedAvgExp, by = "name", all.x = T)
genes = merge(genes, flowerAvgExp, by = "name", all.x = T)

# plot correlations between expression level and treatment-specificity
leafTauLevelCor = cor(genes$leafEnvSpecIndex, genes$leafAvgExpNoZeros, use = "pairwise.complete.obs", method = "spearman")
plotLevelTauComp("leafEnvSpecIndex", "leafAvgExpNoZeros", "leaf", leafTauLevelCor)

rootTauLevelCor = cor(genes$rootEnvSpecIndex, genes$rootAvgExpNoZeros, use = "pairwise.complete.obs", method = "spearman")
plotLevelTauComp("rootEnvSpecIndex", "rootAvgExpNoZeros", "root", rootTauLevelCor)

plantTauLevelCor = cor(genes$plantEnvSpecIndex, genes$plantAvgExpNoZeros, use = "pairwise.complete.obs", method = "spearman")
plotLevelTauComp("plantEnvSpecIndex", "plantAvgExpNoZeros", "plant", plantTauLevelCor)

shootTauLevelCor = cor(genes$shootEnvSpecIndex, genes$shootAvgExpNoZeros, use = "pairwise.complete.obs", method = "spearman")
plotLevelTauComp("shootEnvSpecIndex", "shootAvgExpNoZeros", "shoot", shootTauLevelCor)

seedTauLevelCor = cor(genes$seedEnvSpecIndex, genes$seedAvgExpNoZeros, use = "pairwise.complete.obs", method = "spearman")
plotLevelTauComp("seedEnvSpecIndex", "seedAvgExpNoZeros", "seed", seedTauLevelCor)

flowerTauLevelCor = cor(genes$flowerEnvSpecIndex, genes$flowerAvgExpNoZeros, use = "pairwise.complete.obs", method = "spearman")
plotLevelTauComp("flowerEnvSpecIndex", "flowerAvgExpNoZeros", "flower", flowerTauLevelCor)
```

# partial correlations, before batch correction
```{r}
# which genes have dS > 1? Exclude these from partial correlations looking at dn/ds or DoS or alpha
includedGenesDn = genes$name[which(genes$dS < 1)]

# It doesn't make sense to include family size in analyses using dN/dS because I'm only looking at 1:1 orthologs
leafDn = genes[(genes$name %in% includedGenesDn),c("dN", "dS", "leafEnvSpecIndex", "leafAvgExpNoZeros","GC", "length")]
leafDn = leafDn[complete.cases(leafDn),]
leafDn = pcor(leafDn, method = "spearman")

rootDn = genes[(genes$name %in% includedGenesDn),c("dN", "dS", "rootEnvSpecIndex", "rootAvgExpNoZeros","GC", "length")]
rootDn = rootDn[complete.cases(rootDn),]
rootDn = pcor(rootDn, method = "spearman")

plantDn = genes[(genes$name %in% includedGenesDn),c("dN", "dS", "plantEnvSpecIndex", "plantAvgExpNoZeros","GC", "length")]
plantDn = plantDn[complete.cases(plantDn),]
plantDn = pcor(plantDn, method = "spearman")

shootDn = genes[(genes$name %in% includedGenesDn),c("dN", "dS", "shootEnvSpecIndex", "shootAvgExpNoZeros","GC", "length")]
shootDn = shootDn[complete.cases(shootDn),]
shootDn = pcor(shootDn, method = "spearman")

seedDn = genes[(genes$name %in% includedGenesDn),c("dN", "dS", "seedEnvSpecIndex", "seedAvgExpNoZeros","GC", "length")]
seedDn = seedDn[complete.cases(seedDn),]
seedDn = pcor(seedDn, method = "spearman")

flowerDn = genes[(genes$name %in% includedGenesDn),c("dN", "dS", "flowerEnvSpecIndex", "flowerAvgExpNoZeros","GC", "length")]
flowerDn = flowerDn[complete.cases(flowerDn),]
flowerDn = pcor(flowerDn, method = "spearman")

# piN/piS, now include family size
leafPin = genes[,c("piN", "piS", "leafEnvSpecIndex", "leafAvgExpNoZeros", "GC", "length", "familySize")]
leafPin = leafPin[complete.cases(leafPin),]
leafPin = pcor(leafPin, method = "spearman")

rootPin = genes[,c("piN", "piS", "rootEnvSpecIndex", "rootAvgExpNoZeros", "GC", "length", "familySize")]
rootPin = rootPin[complete.cases(rootPin),]
rootPin = pcor(rootPin, method = "spearman")

plantPin = genes[,c("piN", "piS", "plantEnvSpecIndex", "plantAvgExpNoZeros", "GC", "length", "familySize")]
plantPin = plantPin[complete.cases(plantPin),]
plantPin = pcor(plantPin, method = "spearman")

shootPin = genes[,c("piN", "piS", "shootEnvSpecIndex", "shootAvgExpNoZeros", "GC", "length", "familySize")]
shootPin = shootPin[complete.cases(shootPin),]
shootPin = pcor(shootPin, method = "spearman")

seedPin = genes[,c("piN", "piS", "seedEnvSpecIndex", "seedAvgExpNoZeros", "GC", "length", "familySize")]
seedPin = seedPin[complete.cases(seedPin),]
seedPin = pcor(seedPin, method = "spearman")

flowerPin = genes[,c("piN", "piS", "flowerEnvSpecIndex", "flowerAvgExpNoZeros", "GC", "length", "familySize")]
flowerPin = flowerPin[complete.cases(flowerPin),]
flowerPin = pcor(flowerPin, method = "spearman")

# Tajima's D
leafTd = genes[,c("tajimasD", "leafEnvSpecIndex", "leafAvgExpNoZeros", "GC", "length", "familySize")]
leafTd = leafTd[complete.cases(leafTd),]
leafTd = pcor(leafTd, method = "spearman")

rootTd = genes[,c("tajimasD", "rootEnvSpecIndex", "rootAvgExpNoZeros", "GC", "length", "familySize")]
rootTd = rootTd[complete.cases(rootTd),]
rootTd = pcor(rootTd, method = "spearman")

plantTd = genes[,c("tajimasD", "plantEnvSpecIndex", "plantAvgExpNoZeros", "GC", "length", "familySize")]
plantTd = plantTd[complete.cases(plantTd),]
plantTd = pcor(plantTd, method = "spearman")

shootTd = genes[,c("tajimasD", "shootEnvSpecIndex", "shootAvgExpNoZeros", "GC", "length", "familySize")]
shootTd = shootTd[complete.cases(shootTd),]
shootTd = pcor(shootTd, method = "spearman")

seedTd = genes[,c("tajimasD", "seedEnvSpecIndex", "seedAvgExpNoZeros", "GC", "length", "familySize")]
seedTd = seedTd[complete.cases(seedTd),]
seedTd = pcor(seedTd, method = "spearman")

flowerTd = genes[,c("tajimasD", "flowerEnvSpecIndex", "flowerAvgExpNoZeros", "GC", "length", "familySize")]
flowerTd = flowerTd[complete.cases(flowerTd),]
flowerTd = pcor(flowerTd, method = "spearman")

# alpha
# leafAlpha = genes[,c("alpha", "leafEnvSpecIndex", "leafAvgExpNoZeros", "GC", "length")]
# leafAlpha = leafAlpha[complete.cases(leafAlpha),]
# leafAlpha = pcor(leafAlpha, method = "spearman")
# 
# rootAlpha = genes[,c("alpha", "rootEnvSpecIndex", "rootAvgExpNoZeros", "GC", "length")]
# rootAlpha = rootAlpha[complete.cases(rootAlpha),]
# rootAlpha = pcor(rootAlpha, method = "spearman")
# 
# plantAlpha = genes[,c("alpha", "plantEnvSpecIndex", "plantAvgExpNoZeros", "GC", "length")]
# plantAlpha = plantAlpha[complete.cases(plantAlpha),]
# plantAlpha = pcor(plantAlpha, method = "spearman")
# 
# shootAlpha = genes[,c("alpha", "shootEnvSpecIndex", "shootAvgExpNoZeros", "GC", "length")]
# shootAlpha = shootAlpha[complete.cases(shootAlpha),]
# shootAlpha = pcor(shootAlpha, method = "spearman")
# 
# seedAlpha = genes[,c("alpha", "seedEnvSpecIndex", "seedAvgExpNoZeros", "GC", "length")]
# seedAlpha = seedAlpha[complete.cases(seedAlpha),]
# seedAlpha = pcor(seedAlpha, method = "spearman")
# 
# flowerAlpha = genes[,c("alpha", "flowerEnvSpecIndex", "flowerAvgExpNoZeros", "GC", "length")]
# flowerAlpha = flowerAlpha[complete.cases(flowerAlpha),]
# flowerAlpha = pcor(flowerAlpha, method = "spearman")

# DoS
leafDoS = genes[(genes$name %in% includedGenesDn),c("DoS", "leafEnvSpecIndex", "leafAvgExpNoZeros", "GC", "length")]
leafDoS = leafDoS[complete.cases(leafDoS),]
leafDoS = pcor(leafDoS, method = "spearman")

rootDoS = genes[(genes$name %in% includedGenesDn),c("DoS", "rootEnvSpecIndex", "rootAvgExpNoZeros", "GC", "length")]
rootDoS = rootDoS[complete.cases(rootDoS),]
rootDoS = pcor(rootDoS, method = "spearman")

plantDoS = genes[(genes$name %in% includedGenesDn),c("DoS", "plantEnvSpecIndex", "plantAvgExpNoZeros", "GC", "length")]
plantDoS = plantDoS[complete.cases(plantDoS),]
plantDoS = pcor(plantDoS, method = "spearman")

shootDoS = genes[(genes$name %in% includedGenesDn),c("DoS", "shootEnvSpecIndex", "shootAvgExpNoZeros", "GC", "length")]
shootDoS = shootDoS[complete.cases(shootDoS),]
shootDoS = pcor(shootDoS, method = "spearman")

seedDoS = genes[(genes$name %in% includedGenesDn),c("DoS", "seedEnvSpecIndex", "seedAvgExpNoZeros", "GC", "length")]
seedDoS = seedDoS[complete.cases(seedDoS),]
seedDoS = pcor(seedDoS, method = "spearman")

flowerDoS = genes[(genes$name %in% includedGenesDn),c("DoS", "flowerEnvSpecIndex", "flowerAvgExpNoZeros", "GC", "length")]
flowerDoS = flowerDoS[complete.cases(flowerDoS),]
flowerDoS = pcor(flowerDoS, method = "spearman")
```

# Make control environment baseline environment
```{r}
# Order factor levels so that control is baseline environment
leafPheno$TreatmentOntology = factor(leafPheno$TreatmentOntology, levels = c("PECO:0001062 control exposure", unique(leafPheno$TreatmentOntology)[unique(leafPheno$TreatmentOntology) != "PECO:0001062 control exposure"]))

rootPheno$TreatmentOntology = factor(rootPheno$TreatmentOntology, levels = c("PECO:0001062 control exposure", unique(rootPheno$TreatmentOntology)[unique(rootPheno$TreatmentOntology) != "PECO:0001062 control exposure"]))

plantPheno$TreatmentOntology = factor(plantPheno$TreatmentOntology, levels = c("PECO:0001062 control exposure", unique(plantPheno$TreatmentOntology)[unique(plantPheno$TreatmentOntology) != "PECO:0001062 control exposure"]))

shootPheno$TreatmentOntology = factor(shootPheno$TreatmentOntology, levels = c("PECO:0001062 control exposure", unique(shootPheno$TreatmentOntology)[unique(shootPheno$TreatmentOntology) != "PECO:0001062 control exposure"]))

seedPheno$TreatmentOntology = factor(seedPheno$TreatmentOntology, levels = c("PECO:0001062 control exposure", unique(seedPheno$TreatmentOntology)[unique(seedPheno$TreatmentOntology) != "PECO:0001062 control exposure"]))

flowerPheno$TreatmentOntology = factor(flowerPheno$TreatmentOntology, levels = c("PECO:0001062 control exposure", unique(flowerPheno$TreatmentOntology)[unique(flowerPheno$TreatmentOntology) != "PECO:0001062 control exposure"]))

# Subset datasets to only treatments with multiple studies associated with it
phenoPMID = unique(subset(pheno, select = c(PMID, TissueOntologyCollapsed, TreatmentOntology)))
studyCountByTreatmentAndTissue = table(phenoPMID$TreatmentOntology, phenoPMID$TissueOntologyCollapsed)

# Subset treatments with multiple studies
leafTreats = rownames(studyCountByTreatmentAndTissue)[(studyCountByTreatmentAndTissue[,"PO:0025034 leaf"] > 1)]
rootTreats = rownames(studyCountByTreatmentAndTissue)[(studyCountByTreatmentAndTissue[,"PO:0009005 root"] > 1)]
plantTreats = rownames(studyCountByTreatmentAndTissue)[(studyCountByTreatmentAndTissue[,"PO:0000003 whole plant"] > 1)]
shootTreats = rownames(studyCountByTreatmentAndTissue)[(studyCountByTreatmentAndTissue[,"PO:0009006 shoot system"] > 1)]
seedTreats = rownames(studyCountByTreatmentAndTissue)[(studyCountByTreatmentAndTissue[,"PO:0009010 seed"] > 1)]
flowerTreats = rownames(studyCountByTreatmentAndTissue)[(studyCountByTreatmentAndTissue[,"PO:0009001 fruit OR PO:0009046 flower"] > 1)]

# Subset metadata and expression data to treatments with multiple studies
leafPhenoSva = leafPheno[(leafPheno$TreatmentOntology %in% leafTreats),]
leafTPMdataSva = leafTPMdata[,(colnames(leafTPMdata) %in% leafPhenoSva$Experiment)]
leafTPMdataSva = leafTPMdataSva[(rowSums(leafTPMdataSva) != 0),] # remove nonexpressed genes
rownames(leafPhenoSva) = leafPhenoSva$Experiment
leafPhenoSva = droplevels(leafPhenoSva) # remove unused factor levels

rootPhenoSva = rootPheno[(rootPheno$TreatmentOntology %in% rootTreats),]
rootTPMdataSva = rootTPMdata[,(colnames(rootTPMdata) %in% rootPhenoSva$Experiment)]
rootTPMdataSva = rootTPMdataSva[(rowSums(rootTPMdataSva) != 0),] # remove nonexpressed genes
rownames(rootPhenoSva) = rootPhenoSva$Experiment
rootPhenoSva = droplevels(rootPhenoSva) # remove unused factor levels

plantPhenoSva = plantPheno[(plantPheno$TreatmentOntology %in% plantTreats),]
plantTPMdataSva = plantTPMdata[,(colnames(plantTPMdata) %in% plantPhenoSva$Experiment)]
plantTPMdataSva = plantTPMdataSva[(rowSums(plantTPMdataSva) != 0),] # remove nonexpressed genes
rownames(plantPhenoSva) = plantPhenoSva$Experiment
plantPhenoSva = droplevels(plantPhenoSva) # remove unused factor levels

shootPhenoSva = shootPheno[(shootPheno$TreatmentOntology %in% shootTreats),]
shootTPMdataSva = shootTPMdata[,(colnames(shootTPMdata) %in% shootPhenoSva$Experiment)]
shootTPMdataSva = shootTPMdataSva[(rowSums(shootTPMdataSva) != 0),] # remove nonexpressed genes
rownames(shootPhenoSva) = shootPhenoSva$Experiment
shootPhenoSva = droplevels(shootPhenoSva) # remove unused factor levels

seedPhenoSva = seedPheno[(seedPheno$TreatmentOntology %in% seedTreats),]
seedTPMdataSva = seedTPMdata[,(colnames(seedTPMdata) %in% seedPhenoSva$Experiment)]
seedTPMdataSva = seedTPMdataSva[(rowSums(seedTPMdataSva) != 0),] # remove nonexpressed genes
rownames(seedPhenoSva) = seedPhenoSva$Experiment
seedPhenoSva = droplevels(seedPhenoSva) # remove unused factor levels

flowerPhenoSva = flowerPheno[(flowerPheno$TreatmentOntology %in% flowerTreats),]
flowerTPMdataSva = flowerTPMdata[,(colnames(flowerTPMdata) %in% flowerPhenoSva$Experiment)]
flowerTPMdataSva = flowerTPMdataSva[(rowSums(flowerTPMdataSva) != 0),] # remove nonexpressed genes
rownames(flowerPhenoSva) = flowerPhenoSva$Experiment # add rownames to match model and data
flowerPhenoSva = droplevels(flowerPhenoSva) # remove unused factor levels

```

# Correct for batch effects with SVA
```{r}
# full model matrix - family, stress, tissue should be sig
leafMod <- model.matrix(~TreatmentOntology, data=leafPhenoSva)
rootMod <- model.matrix(~TreatmentOntology, data=rootPhenoSva)
plantMod <- model.matrix(~TreatmentOntology, data=plantPhenoSva)
shootMod <- model.matrix(~TreatmentOntology, data=shootPhenoSva)
seedMod <- model.matrix(~TreatmentOntology, data=seedPhenoSva)
flowerMod <- model.matrix(~TreatmentOntology, data=flowerPhenoSva)

# null model matrix (no adjustment variables are included)
leafModNull <- model.matrix(~1, data=leafPhenoSva)
rootModNull <- model.matrix(~1, data=rootPhenoSva)
plantModNull <- model.matrix(~1, data=plantPhenoSva)
shootModNull <- model.matrix(~1, data=shootPhenoSva)
seedModNull <- model.matrix(~1, data=seedPhenoSva)
flowerModNull <- model.matrix(~1, data=flowerPhenoSva)

# expression data must be a matrix
leafTPMdataSva <- as.matrix(leafTPMdataSva)
rootTPMdataSva <- as.matrix(rootTPMdataSva)
plantTPMdataSva <- as.matrix(plantTPMdataSva)
shootTPMdataSva <- as.matrix(shootTPMdataSva)
seedTPMdataSva <- as.matrix(seedTPMdataSva)
flowerTPMdataSva <- as.matrix(flowerTPMdataSva)

# Apply sva
leafSvaOut = svaseq(leafTPMdataSva, leafMod, leafModNull)
rootSvaOut = svaseq(rootTPMdataSva, rootMod, rootModNull)
plantSvaOut = svaseq(plantTPMdataSva, plantMod, plantModNull)
shootSvaOut = svaseq(shootTPMdataSva, shootMod, shootModNull)
seedSvaOut = svaseq(seedTPMdataSva, seedMod, seedModNull)
flowerSvaOut = svaseq(flowerTPMdataSva, flowerMod, flowerModNull)

# plot of first 2 surrogate variables
plot(leafSvaOut$sv, pch=20, col="blue")
plot(rootSvaOut$sv, pch=20, col="blue")
plot(plantSvaOut$sv, pch=20, col="blue")
plot(shootSvaOut$sv, pch=20, col="blue")
plot(seedSvaOut$sv, pch=20, col="blue")
plot(flowerSvaOut$sv, pch=20, col="blue")
```
# Remove SV effects
```{r}
## Generate a clean matrix using a function by Andrew Jaffe
# This function removes the effects of SVs from our expression data
#'y' as the gene expresion matrix
#'mod' as the model matrix you sent to sva (the full model)
#'svs' as svobj$sv where svobj is the output from the sva function
cleanY = function(y, mod, svs) {
    X = cbind(mod, svs) # same as modSv
    Hat = solve(t(X) %*% X) %*% t(X)
    beta = (Hat %*% t(y))
    rm(Hat)
    gc()
    P = ncol(mod)
    return(y - t(as.matrix(X[,-c(1:P)]) %*% beta[-c(1:P),]))
}

#Remove effects of all svs, need to log-transform data then back-transform because SVs were estimated with log-transformed data
# y = log(x + 1) -> exp(y) - 1 = x
# round up negative expression values to zero
leafTPMdataSvaClean <- exp(cleanY(log(leafTPMdataSva + 1), leafMod, leafSvaOut$sv)) - 1
leafTPMdataSvaClean[which(leafTPMdataSvaClean < 0, arr.ind = T)] = 0 

rootTPMdataSvaClean <- exp(cleanY(log(rootTPMdataSva + 1), rootMod, rootSvaOut$sv)) - 1
rootTPMdataSvaClean[which(rootTPMdataSvaClean < 0, arr.ind = T)] = 0 

plantTPMdataSvaClean <- exp(cleanY(log(plantTPMdataSva + 1), plantMod, plantSvaOut$sv)) - 1
plantTPMdataSvaClean[which(plantTPMdataSvaClean < 0, arr.ind = T)] = 0 

shootTPMdataSvaClean <- exp(cleanY(log(shootTPMdataSva + 1), shootMod, shootSvaOut$sv)) - 1
shootTPMdataSvaClean[which(shootTPMdataSvaClean < 0, arr.ind = T)] = 0 

seedTPMdataSvaClean <- exp(cleanY(log(seedTPMdataSva + 1), seedMod, seedSvaOut$sv)) - 1
seedTPMdataSvaClean[which(seedTPMdataSvaClean < 0, arr.ind = T)] = 0 

flowerTPMdataSvaClean <- exp(cleanY(log(flowerTPMdataSva + 1), flowerMod, flowerSvaOut$sv)) - 1
flowerTPMdataSvaClean[which(flowerTPMdataSvaClean < 0, arr.ind = T)] = 0 
```

# Recalculate average expression, specificity, based on sva-corrected data
```{r}
# calculate treatment profiles for each tissue
leafProfileSva = treatmentProfile(leafPhenoSva, leafTPMdataSvaClean)
rootProfileSva = treatmentProfile(rootPhenoSva, rootTPMdataSvaClean)
plantProfileSva = treatmentProfile(plantPhenoSva, plantTPMdataSvaClean)
shootProfileSva = treatmentProfile(shootPhenoSva, shootTPMdataSvaClean)
seedProfileSva = treatmentProfile(seedPhenoSva, seedTPMdataSvaClean)
flowerProfileSva = treatmentProfile(flowerPhenoSva, flowerTPMdataSvaClean)

# Calculate specificity index from treatment profile
leafSpecSva = as.data.frame(apply(leafProfileSva, MARGIN = 1, FUN = specIndex))
leafSpecSva$name = rownames(leafSpecSva)
names(leafSpecSva) = c("leafEnvSpecIndexSva", "name")

rootSpecSva = as.data.frame(apply(rootProfileSva, MARGIN = 1, FUN = specIndex))
rootSpecSva$name = rownames(rootSpecSva)
names(rootSpecSva) = c("rootEnvSpecIndexSva", "name")

plantSpecSva = as.data.frame(apply(plantProfileSva, MARGIN = 1, FUN = specIndex))
plantSpecSva$name = rownames(plantSpecSva)
names(plantSpecSva) = c("plantEnvSpecIndexSva", "name")

shootSpecSva = as.data.frame(apply(shootProfileSva, MARGIN = 1, FUN = specIndex))
shootSpecSva$name = rownames(shootSpecSva)
names(shootSpecSva) = c("shootEnvSpecIndexSva", "name")

seedSpecSva = as.data.frame(apply(seedProfileSva, MARGIN = 1, FUN = specIndex))
seedSpecSva$name = rownames(seedSpecSva)
names(seedSpecSva) = c("seedEnvSpecIndexSva", "name")

flowerSpecSva = as.data.frame(apply(flowerProfileSva, MARGIN = 1, FUN = specIndex))
flowerSpecSva$name = rownames(flowerSpecSva)
names(flowerSpecSva) = c("flowerEnvSpecIndexSva", "name")

# merge specificity with gene feature table, don't drop any genes
genes = merge(genes, leafSpecSva, by = "name", all.x = T)
genes = merge(genes, rootSpecSva, by = "name", all.x = T)
genes = merge(genes, plantSpecSva, by = "name", all.x = T)
genes = merge(genes, shootSpecSva, by = "name", all.x = T)
genes = merge(genes, seedSpecSva, by = "name", all.x = T)
genes = merge(genes, flowerSpecSva, by = "name", all.x = T)

# Calculate average expression, ignoring zeros
leafAvgExpSva = as.data.frame(apply(leafTPMdataSvaClean, MARGIN = 1, FUN = meanExcludeZeros))
leafAvgExpSva$name = rownames(leafAvgExpSva)
names(leafAvgExpSva) = c("leafAvgExpNoZerosSva", "name")

rootAvgExpSva = as.data.frame(apply(rootTPMdataSvaClean, MARGIN = 1, FUN = meanExcludeZeros))
rootAvgExpSva$name = rownames(rootAvgExpSva)
names(rootAvgExpSva) = c("rootAvgExpNoZerosSva", "name")

plantAvgExpSva = as.data.frame(apply(plantTPMdataSvaClean, MARGIN = 1, FUN = meanExcludeZeros))
plantAvgExpSva$name = rownames(plantAvgExpSva)
names(plantAvgExpSva) = c("plantAvgExpNoZerosSva", "name")

shootAvgExpSva = as.data.frame(apply(shootTPMdataSvaClean, MARGIN = 1, FUN = meanExcludeZeros))
shootAvgExpSva$name = rownames(shootAvgExpSva)
names(shootAvgExpSva) = c("shootAvgExpNoZerosSva", "name")

seedAvgExpSva = as.data.frame(apply(seedTPMdataSvaClean, MARGIN = 1, FUN = meanExcludeZeros))
seedAvgExpSva$name = rownames(seedAvgExpSva)
names(seedAvgExpSva) = c("seedAvgExpNoZerosSva", "name")

flowerAvgExpSva = as.data.frame(apply(flowerTPMdataSvaClean, MARGIN = 1, FUN = meanExcludeZeros))
flowerAvgExpSva$name = rownames(flowerAvgExpSva)
names(flowerAvgExpSva) = c("flowerAvgExpNoZerosSva", "name")

# merge average expression with gene feature table, don't drop any genes
genes = merge(genes, leafAvgExpSva, by = "name", all.x = T)
genes = merge(genes, rootAvgExpSva, by = "name", all.x = T)
genes = merge(genes, plantAvgExpSva, by = "name", all.x = T)
genes = merge(genes, shootAvgExpSva, by = "name", all.x = T)
genes = merge(genes, seedAvgExpSva, by = "name", all.x = T)
genes = merge(genes, flowerAvgExpSva, by = "name", all.x = T)
```

# Partial correlations batch-corrected data
```{r}
# dn/ds
leafDnSva = genes[,c("dN", "dS", "leafEnvSpecIndexSva", "leafAvgExpNoZerosSva", "GC", "length")]
leafDnSva = leafDnSva[complete.cases(leafDnSva),]
leafDnSva = pcor(leafDnSva, method = "spearman")

rootDnSva = genes[,c("dN", "dS", "rootEnvSpecIndexSva", "rootAvgExpNoZerosSva", "GC", "length")]
rootDnSva = rootDnSva[complete.cases(rootDnSva),]
rootDnSva = pcor(rootDnSva, method = "spearman")

plantDnSva = genes[,c("dN", "dS", "plantEnvSpecIndexSva", "plantAvgExpNoZerosSva", "GC", "length")]
plantDnSva = plantDnSva[complete.cases(plantDnSva),]
plantDnSva = pcor(plantDnSva, method = "spearman")

shootDnSva = genes[,c("dN", "dS", "shootEnvSpecIndexSva", "shootAvgExpNoZerosSva", "GC", "length")]
shootDnSva = shootDnSva[complete.cases(shootDnSva),]
shootDnSva = pcor(shootDnSva, method = "spearman")

seedDnSva = genes[,c("dN", "dS", "seedEnvSpecIndexSva", "seedAvgExpNoZerosSva", "GC", "length")]
seedDnSva = seedDnSva[complete.cases(seedDnSva),]
seedDnSva = pcor(seedDnSva, method = "spearman")

flowerDnSva = genes[,c("dN", "dS", "flowerEnvSpecIndexSva", "flowerAvgExpNoZerosSva", "GC", "length")]
flowerDnSva = flowerDnSva[complete.cases(flowerDnSva),]
flowerDnSva = pcor(flowerDnSva, method = "spearman")

# piN/piS
leafPinSva = genes[,c("piN", "piS", "leafEnvSpecIndexSva", "leafAvgExpNoZerosSva", "GC", "length", "familySize")]
leafPinSva = leafPinSva[complete.cases(leafPinSva),]
leafPinSva = pcor(leafPinSva, method = "spearman")

rootPinSva = genes[,c("piN", "piS", "rootEnvSpecIndexSva", "rootAvgExpNoZerosSva", "GC", "length", "familySize")]
rootPinSva = rootPinSva[complete.cases(rootPinSva),]
rootPinSva = pcor(rootPinSva, method = "spearman")

plantPinSva = genes[,c("piN", "piS", "plantEnvSpecIndexSva", "plantAvgExpNoZerosSva", "GC", "length", "familySize")]
plantPinSva = plantPinSva[complete.cases(plantPinSva),]
plantPinSva = pcor(plantPinSva, method = "spearman")

shootPinSva = genes[,c("piN", "piS", "shootEnvSpecIndexSva", "shootAvgExpNoZerosSva", "GC", "length", "familySize")]
shootPinSva = shootPinSva[complete.cases(shootPinSva),]
shootPinSva = pcor(shootPinSva, method = "spearman")

seedPinSva = genes[,c("piN", "piS", "seedEnvSpecIndexSva", "seedAvgExpNoZerosSva", "GC", "length", "familySize")]
seedPinSva = seedPinSva[complete.cases(seedPinSva),]
seedPinSva = pcor(seedPinSva, method = "spearman")

flowerPinSva = genes[,c("piN", "piS", "flowerEnvSpecIndexSva", "flowerAvgExpNoZerosSva", "GC", "length", "familySize")]
flowerPinSva = flowerPinSva[complete.cases(flowerPinSva),]
flowerPinSva = pcor(flowerPinSva, method = "spearman")

# tajima's D
leafTdSva = genes[,c("tajimasD", "leafEnvSpecIndexSva", "leafAvgExpNoZerosSva", "GC", "length", "familySize")]
leafTdSva = leafTdSva[complete.cases(leafTdSva),]
leafTdSva = pcor(leafTdSva, method = "spearman")

rootTdSva = genes[,c("tajimasD", "rootEnvSpecIndexSva", "rootAvgExpNoZerosSva", "GC", "length", "familySize")]
rootTdSva = rootTdSva[complete.cases(rootTdSva),]
rootTdSva = pcor(rootTdSva, method = "spearman")

plantTdSva = genes[,c("tajimasD", "plantEnvSpecIndexSva", "plantAvgExpNoZerosSva", "GC", "length", "familySize")]
plantTdSva = plantTdSva[complete.cases(plantTdSva),]
plantTdSva = pcor(plantTdSva, method = "spearman")

shootTdSva = genes[,c("tajimasD", "shootEnvSpecIndexSva", "shootAvgExpNoZerosSva", "GC", "length", "familySize")]
shootTdSva = shootTdSva[complete.cases(shootTdSva),]
shootTdSva = pcor(shootTdSva, method = "spearman")

seedTdSva = genes[,c("tajimasD", "seedEnvSpecIndexSva", "seedAvgExpNoZerosSva", "GC", "length", "familySize")]
seedTdSva = seedTdSva[complete.cases(seedTdSva),]
seedTdSva = pcor(seedTdSva, method = "spearman")

flowerTdSva = genes[,c("tajimasD", "flowerEnvSpecIndexSva", "flowerAvgExpNoZerosSva", "GC", "length", "familySize")]
flowerTdSva = flowerTdSva[complete.cases(flowerTdSva),]
flowerTdSva = pcor(flowerTdSva, method = "spearman")

# alpha
leafAlphaSva = genes[,c("alpha", "leafEnvSpecIndexSva", "leafAvgExpNoZerosSva", "GC", "length")]
leafAlphaSva = leafAlphaSva[complete.cases(leafAlphaSva),]
leafAlphaSva = pcor(leafAlphaSva, method = "spearman")

rootAlphaSva = genes[,c("alpha", "rootEnvSpecIndexSva", "rootAvgExpNoZerosSva", "GC", "length")]
rootAlphaSva = rootAlphaSva[complete.cases(rootAlphaSva),]
rootAlphaSva = pcor(rootAlphaSva, method = "spearman")

plantAlphaSva = genes[,c("alpha", "plantEnvSpecIndexSva", "plantAvgExpNoZerosSva", "GC", "length")]
plantAlphaSva = plantAlphaSva[complete.cases(plantAlphaSva),]
plantAlphaSva = pcor(plantAlphaSva, method = "spearman")

shootAlphaSva = genes[,c("alpha", "shootEnvSpecIndexSva", "shootAvgExpNoZerosSva", "GC", "length")]
shootAlphaSva = shootAlphaSva[complete.cases(shootAlphaSva),]
shootAlphaSva = pcor(shootAlphaSva, method = "spearman")

seedAlphaSva = genes[,c("alpha", "seedEnvSpecIndexSva", "seedAvgExpNoZerosSva", "GC", "length")]
seedAlphaSva = seedAlphaSva[complete.cases(seedAlphaSva),]
seedAlphaSva = pcor(seedAlphaSva, method = "spearman")

flowerAlphaSva = genes[,c("alpha", "flowerEnvSpecIndexSva", "flowerAvgExpNoZerosSva", "GC", "length")]
flowerAlphaSva = flowerAlphaSva[complete.cases(flowerAlphaSva),]
flowerAlphaSva = pcor(flowerAlphaSva, method = "spearman")

# DoS
leafDosSva = genes[,c("DoS", "leafEnvSpecIndexSva", "leafAvgExpNoZerosSva", "GC", "length")]
leafDosSva = leafDosSva[complete.cases(leafDosSva),]
leafDosSva = pcor(leafDosSva, method = "spearman")

rootDosSva = genes[,c("DoS", "rootEnvSpecIndexSva", "rootAvgExpNoZerosSva", "GC", "length")]
rootDosSva = rootDosSva[complete.cases(rootDosSva),]
rootDosSva = pcor(rootDosSva, method = "spearman")

plantDosSva = genes[,c("DoS", "plantEnvSpecIndexSva", "plantAvgExpNoZerosSva", "GC", "length")]
plantDosSva = plantDosSva[complete.cases(plantDosSva),]
plantDosSva = pcor(plantDosSva, method = "spearman")

shootDosSva = genes[,c("DoS", "shootEnvSpecIndexSva", "shootAvgExpNoZerosSva", "GC", "length")]
shootDosSva = shootDosSva[complete.cases(shootDosSva),]
shootDosSva = pcor(shootDosSva, method = "spearman")

seedDosSva = genes[,c("DoS", "seedEnvSpecIndexSva", "seedAvgExpNoZerosSva", "GC", "length")]
seedDosSva = seedDosSva[complete.cases(seedDosSva),]
seedDosSva = pcor(seedDosSva, method = "spearman")

flowerDosSva = genes[,c("DoS", "flowerEnvSpecIndexSva", "flowerAvgExpNoZerosSva", "GC", "length")]
flowerDosSva = flowerDosSva[complete.cases(flowerDosSva),]
flowerDosSva = pcor(flowerDosSva, method = "spearman")

# Put all correlations into a list
pcorBeforeAfterSvaList = list(leafDn, rootDn, plantDn, shootDn, seedDn, flowerDn, leafPin, rootPin, plantPin, shootPin, seedPin, flowerPin, leafTd, rootTd, plantTd, shootTd, seedTd, flowerTd, leafDoS, rootDoS, plantDoS, shootDoS, seedDoS, flowerDoS, leafDnSva, rootDnSva, plantDnSva, shootDnSva, seedDnSva, flowerDnSva, leafPinSva, rootPinSva, plantPinSva, shootPinSva, seedPinSva, flowerPinSva, leafTdSva, rootTdSva, plantTdSva, shootTdSva, seedTdSva, flowerTdSva, leafDosSva, rootDosSva, plantDosSva, shootDosSva, seedDosSva, flowerDosSva)
```

# Plot comparisons of partial correlations, batch corrected and non-batch corrected
```{r}
# change variable names
changePcorVariableNames = function(pCorOutput){
  colnames(pCorOutput$estimate)[(grepl("EnvSpecIndex", colnames(pCorOutput$estimate)))] = "Treatment specificity"
  colnames(pCorOutput$estimate)[(grepl("AvgExpNoZeros", colnames(pCorOutput$estimate)))] = "Average expression"
  colnames(pCorOutput$estimate)[(grepl("familySize", colnames(pCorOutput$estimate)))] = "Family size"
  colnames(pCorOutput$estimate)[(grepl("_tissue_specificity", colnames(pCorOutput$estimate)))] = "Tissue specificity"
  rownames(pCorOutput$estimate)[(grepl("EnvSpecIndex", rownames(pCorOutput$estimate)))] = "Treatment specificity"
  rownames(pCorOutput$estimate)[(grepl("AvgExpNoZeros", rownames(pCorOutput$estimate)))] = "Average expression"
  rownames(pCorOutput$estimate)[(grepl("familySize", rownames(pCorOutput$estimate)))] = "Family size"
  rownames(pCorOutput$estimate)[(grepl("_tissue_specificity", rownames(pCorOutput$estimate)))] = "Tissue specificity"
    
  return(pCorOutput)
}

# Get upper triangle of the correlation matrix
get_upper_tri <- function(cormat){
  cormat[lower.tri(cormat)]<- NA
  return(cormat)
}

# melt pcor output to matirx for plotting
meltPcorOutput = function(pCorOutput){
  reshape2::melt(get_upper_tri(pCorOutput$estimate), na.rm=TRUE,value.name="value",varnames=c("X1","X2"))
}

### Non-SVA-corrected data
pcorBeforeAfterSvaList = lapply(pcorBeforeAfterSvaList, changePcorVariableNames)
meltedCormats = lapply(pcorBeforeAfterSvaList, meltPcorOutput) 

## dn/ds
# leafDn = changePcorVariableNames(leafDn)
# rootDn = changePcorVariableNames(rootDn)
# plantDn = changePcorVariableNames(plantDn)
# shootDn = changePcorVariableNames(shootDn)
# seedDn = changePcorVariableNames(seedDn)
# flowerDn = changePcorVariableNames(flowerDn)
# 
# ## piN/piS
# leafPin = changePcorVariableNames(leafPin)
# rootPin = changePcorVariableNames(rootPin)
# plantPin = changePcorVariableNames(plantPin)
# shootPin = changePcorVariableNames(shootPin)
# seedPin = changePcorVariableNames(seedPin)
# flowerPin = changePcorVariableNames(flowerPin)
# 
# ## tajima's D
# leafTd = changePcorVariableNames(leafTd)
# rootTd = changePcorVariableNames(rootTd)
# plantTd = changePcorVariableNames(plantTd)
# shootTd = changePcorVariableNames(shootTd)
# seedTd = changePcorVariableNames(seedTd)
# flowerTd = changePcorVariableNames(flowerTd)
# 
# ## alpha
# leafAlpha = changePcorVariableNames(leafAlpha)
# rootAlpha = changePcorVariableNames(rootAlpha)
# plantAlpha = changePcorVariableNames(plantAlpha)
# shootAlpha = changePcorVariableNames(shootAlpha)
# seedAlpha = changePcorVariableNames(seedAlpha)
# flowerAlpha = changePcorVariableNames(flowerAlpha)
# 
# ## DoS
# leafDoS = changePcorVariableNames(leafDoS)
# rootDoS = changePcorVariableNames(rootDoS)
# plantDoS = changePcorVariableNames(plantDoS)
# shootDoS = changePcorVariableNames(shootDoS)
# seedDoS = changePcorVariableNames(seedDoS)
# flowerDoS = changePcorVariableNames(flowerDoS)
# 
# ### SVA-corrected data
# ## dn/ds
# leafDnSva = changePcorVariableNames(leafDnSva)
# rootDnSva = changePcorVariableNames(rootDnSva)
# plantDnSva = changePcorVariableNames(plantDnSva)
# shootDnSva = changePcorVariableNames(shootDnSva)
# seedDnSva = changePcorVariableNames(seedDnSva)
# flowerDnSva = changePcorVariableNames(flowerDnSva)
# 
# ## piN/piS
# leafPinSva = changePcorVariableNames(leafPinSva)
# rootPinSva = changePcorVariableNames(rootPinSva)
# plantPinSva = changePcorVariableNames(plantPinSva)
# shootPinSva = changePcorVariableNames(shootPinSva)
# seedPinSva = changePcorVariableNames(seedPinSva)
# flowerPinSva = changePcorVariableNames(flowerPinSva)
# 
# ## tajima's D
# leafTdSva = changePcorVariableNames(leafTdSva)
# rootTdSva = changePcorVariableNames(rootTdSva)
# plantTdSva = changePcorVariableNames(plantTdSva)
# shootTdSva = changePcorVariableNames(shootTdSva)
# seedTdSva = changePcorVariableNames(seedTdSva)
# flowerTdSva = changePcorVariableNames(flowerTdSva)
# 
# ## alpha
# # leafAlphaSva = changePcorVariableNames(leafAlphaSva)
# 
# ## DoS
# leafDosSva = changePcorVariableNames(leafDosSva)
# rootDosSva = changePcorVariableNames(rootDosSva)
# plantDosSva = changePcorVariableNames(plantDosSva)
# shootDosSva = changePcorVariableNames(shootDosSva)
# seedDosSva = changePcorVariableNames(seedDosSva)
# flowerDosSva = changePcorVariableNames(flowerDosSva)



## dn/ds
# meltedCormatLeafDn <- reshape2::melt(get_upper_tri(leafDn$estimate), na.rm=TRUE,value.name="value",varnames=c("X1","X2"))
# meltedCormatRootDn <- reshape2::melt(get_upper_tri(rootDn$estimate), na.rm=TRUE,value.name="value",varnames=c("X1","X2"))
# meltedCormatPlantDn <- reshape2::melt(get_upper_tri(plantDn$estimate), na.rm=TRUE,value.name="value",varnames=c("X1","X2"))
# meltedCormatShootDn <- reshape2::melt(get_upper_tri(shootDn$estimate), na.rm=TRUE,value.name="value",varnames=c("X1","X2"))
# meltedCormatSeedDn <- reshape2::melt(get_upper_tri(seedDn$estimate), na.rm=TRUE,value.name="value",varnames=c("X1","X2"))
# meltedCormatFlowerDn <- reshape2::melt(get_upper_tri(flowerDn$estimate), na.rm=TRUE,value.name="value",varnames=c("X1","X2"))
# 
# meltedCormatLeafPin <- reshape2::melt(get_upper_tri(leafPin$estimate), na.rm=TRUE,value.name="value",varnames=c("X1","X2"))
# meltedCormatLeafPin <- reshape2::melt(get_upper_tri(leafPin$estimate), na.rm=TRUE,value.name="value",varnames=c("X1","X2"))
# meltedCormatLeafPin <- reshape2::melt(get_upper_tri(leafPin$estimate), na.rm=TRUE,value.name="value",varnames=c("X1","X2"))
# meltedCormatLeafPin <- reshape2::melt(get_upper_tri(leafPin$estimate), na.rm=TRUE,value.name="value",varnames=c("X1","X2"))
# meltedCormatLeafPin <- reshape2::melt(get_upper_tri(leafPin$estimate), na.rm=TRUE,value.name="value",varnames=c("X1","X2"))
# meltedCormatLeafPin <- reshape2::melt(get_upper_tri(leafPin$estimate), na.rm=TRUE,value.name="value",varnames=c("X1","X2"))
# 
# meltedCormatLeafTd <- reshape2::melt(get_upper_tri(leafTd$estimate), na.rm=TRUE,value.name="value",varnames=c("X1","X2"))
# meltedCormatLeafDos <- reshape2::melt(get_upper_tri(leafDoS$estimate), na.rm=TRUE,value.name="value",varnames=c("X1","X2"))
# 
# meltedCormatLeafDnSva <- reshape2::melt(get_upper_tri(leafDnSva$estimate), na.rm=TRUE,value.name="value",varnames=c("X1","X2"))
# meltedCormatLeafPinSva <- reshape2::melt(get_upper_tri(leafPinSva$estimate), na.rm=TRUE,value.name="value",varnames=c("X1","X2"))
# meltedCormatLeafTdSva <- reshape2::melt(get_upper_tri(leafTdSva$estimate), na.rm=TRUE,value.name="value",varnames=c("X1","X2"))
# meltedCormatLeafDosSva <- reshape2::melt(get_upper_tri(leafDosSva$estimate), na.rm=TRUE,value.name="value",varnames=c("X1","X2"))

### HEATMAP FUNCTION
myHeatmapPlot = function(myCorMat, myTitle){
  ggplot(data = myCorMat, aes(X1, X2, fill = value))+
  geom_tile(color = "white") +
  geom_text(aes(label = round(value, 2))) +
  scale_fill_gradient2(low = "#7b394a", 
                      high = "#acf641", 
                      mid = "#f6ee8b", 
                      midpoint = 0, 
                      limit = c(-1,1), 
                      space = "Lab", 
                      name="Partial\nspearman\ncorrelation\ncoefficient") +
  theme_minimal() + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1), 
        axis.title.x = element_blank(), 
        axis.title.y = element_blank(), 
        plot.title = element_text(hjust = 0.5),
        plot.margin = margin(t = 2,  # Top margin
                             r = 2,  # Right margin
                             b = 2,  # Bottom margin
                             l = 2)) +
  coord_fixed() + 
  labs(title = myTitle, xlab = "", ylab = "")
}

### HEATMAPS ###
# leafDnPlot = myHeatmapPlot(meltedCormatLeafDn, "Before SVA")
# leafPinPlot = myHeatmapPlot(meltedCormatLeafPin, "Before SVA")
# leafTdPlot = myHeatmapPlot(meltedCormatLeafTd, "Before SVA")
# leafAlphaPlot = myHeatmapPlot(meltedCormatLeafAlpha, "Before SVA")
# leafDosPlot = myHeatmapPlot(meltedCormatLeafDos, "Before SVA")
# 
# leafDnSvaPlot = myHeatmapPlot(meltedCormatLeafDnSva, "After SVA")
# leafPinSvaPlot = myHeatmapPlot(meltedCormatLeafPinSva, "After SVA")
# leafTdSvaPlot = myHeatmapPlot(meltedCormatLeafTdSva, "After SVA")
# leafAlphaSvaPlot = myHeatmapPlot(meltedCormatLeafAlphaSva, "After SVA")
# leafDosSvaPlot = myHeatmapPlot(meltedCormatLeafDosSva, "After SVA")

myHeatmapsBeforeSva = lapply(meltedCormats[1:24], myHeatmapPlot, myTitle = "Before SVA")
myHeatmapsAfterSva = lapply(meltedCormats[25:48], myHeatmapPlot, myTitle = "After SVA")

### leaf
pcorSvaComparisonLeaf = ggarrange(myHeatmapsBeforeSva[[1]], myHeatmapsAfterSva[[1]], myHeatmapsBeforeSva[[7]], myHeatmapsAfterSva[[7]], myHeatmapsBeforeSva[[13]], myHeatmapsAfterSva[[13]], myHeatmapsBeforeSva[[19]], myHeatmapsAfterSva[[19]], ncol = 2, nrow = 4, common.legend = T, labels = c("A", "B", "C", "D", "E", "F", "G", "H"))
  
ggsave("paritalCorrelations_leafBeforeAndAfterSva_2022-07-15.jpeg", pcorSvaComparisonLeaf, scale = 1, width = 7, height = 14, units = "in")
  
### root
pcorSvaComparisonRoot = ggarrange(myHeatmapsBeforeSva[[2]], myHeatmapsAfterSva[[2]], myHeatmapsBeforeSva[[8]], myHeatmapsAfterSva[[8]], myHeatmapsBeforeSva[[14]], myHeatmapsAfterSva[[14]], myHeatmapsBeforeSva[[20]], myHeatmapsAfterSva[[20]], ncol = 2, nrow = 4, common.legend = T, labels = c("A", "B", "C", "D", "E", "F", "G", "H"))
  
ggsave("paritalCorrelations_rootBeforeAndAfterSva_2022-07-15.jpeg", pcorSvaComparisonRoot, scale = 1, width = 7, height = 14, units = "in")

### plant
pcorSvaComparisonPlant = ggarrange(myHeatmapsBeforeSva[[3]], myHeatmapsAfterSva[[3]], myHeatmapsBeforeSva[[9]], myHeatmapsAfterSva[[9]], myHeatmapsBeforeSva[[15]], myHeatmapsAfterSva[[15]], myHeatmapsBeforeSva[[21]], myHeatmapsAfterSva[[21]], ncol = 2, nrow = 4, common.legend = T, labels = c("A", "B", "C", "D", "E", "F", "G", "H"))
  
ggsave("paritalCorrelations_plantBeforeAndAfterSva_2022-07-15.jpeg", pcorSvaComparisonPlant, scale = 1, width = 7, height = 14, units = "in")

### shoot
pcorSvaComparisonShoot = ggarrange(myHeatmapsBeforeSva[[4]], myHeatmapsAfterSva[[4]], myHeatmapsBeforeSva[[10]], myHeatmapsAfterSva[[10]], myHeatmapsBeforeSva[[16]], myHeatmapsAfterSva[[16]], myHeatmapsBeforeSva[[22]], myHeatmapsAfterSva[[22]], ncol = 2, nrow = 4, common.legend = T, labels = c("A", "B", "C", "D", "E", "F", "G", "H"))
  
ggsave("paritalCorrelations_shootBeforeAndAfterSva_2022-07-15.jpeg", pcorSvaComparisonShoot, scale = 1, width = 7, height = 14, units = "in")

### seed
pcorSvaComparisonSeed = ggarrange(myHeatmapsBeforeSva[[5]], myHeatmapsAfterSva[[5]], myHeatmapsBeforeSva[[11]], myHeatmapsAfterSva[[11]], myHeatmapsBeforeSva[[17]], myHeatmapsAfterSva[[17]], myHeatmapsBeforeSva[[23]], myHeatmapsAfterSva[[23]], ncol = 2, nrow = 4, common.legend = T, labels = c("A", "B", "C", "D", "E", "F", "G", "H"))
  
ggsave("paritalCorrelations_seedBeforeAndAfterSva_2022-07-15.jpeg", pcorSvaComparisonSeed, scale = 1, width = 7, height = 14, units = "in")

### flower
pcorSvaComparisonFlower = ggarrange(myHeatmapsBeforeSva[[6]], myHeatmapsAfterSva[[6]], myHeatmapsBeforeSva[[12]], myHeatmapsAfterSva[[12]], myHeatmapsBeforeSva[[18]], myHeatmapsAfterSva[[18]], myHeatmapsBeforeSva[[24]], myHeatmapsAfterSva[[24]], ncol = 2, nrow = 4, common.legend = T, labels = c("A", "B", "C", "D", "E", "F", "G", "H"))
  
ggsave("paritalCorrelations_flowerBeforeAndAfterSva_2022-07-15.jpeg", pcorSvaComparisonFlower, scale = 1, width = 7, height = 14, units = "in")

#allHeatMaps2 = ggarrange( ncol = 2, nrow = 2, common.legend = T, labels = c("A", "B", "C", "D"))

# allHeatMaps = grid.arrange(gLeafPi, gRootPi, gPlantPi, nrow = 1, ncol = 3)
# need to play with scale arguement to get figure to size correctly (switch from 3 -> 1 or 1 -> 3)
# ggsave("paritalCorrelations1_leafBeforeAndAfterSva_2022-07-12.jpeg", allHeatMaps1, scale = 3)
# ggsave("paritalCorrelations2_leafBeforeAndAfterSva_2022-07-12.jpeg", allHeatMaps2, scale = 3)

```

# Break up data by treatment, calculate tissue specificity
```{r}
# Function to calculate tissue profile: average expression in each tissue for each gene
tissueProfile = function(myPheno, myMdata){
  # Matrix to store outputs
  tissues = unique(myPheno$TissueOntologyCollapsed)
  genes = rownames(myMdata)
  expTis = matrix(NA, nrow = length(genes), ncol = length(tissues), dimnames = list(genes, tissues))

  # generate expression profiles for every gene
  # Expression profile = median expression of each gene in each environment
  for(tissue in tissues){
    # print(tissue)
    #subset data from one tissue
	  exps = unique(myPheno[(myPheno$TissueOntologyCollapsed == tissue), "Experiment"])
	  tpmdataSub = as.matrix(myMdata[,exps])
	  # calculate mean expression of every gene in this tissue
	  tisMeans = rowMeans(tpmdataSub, na.rm = T)
	  expTis[,(colnames(expTis) == tissue)] = tisMeans
  }

  # remove genes with mean of zero in all tissues
  # these genes will have an undefined specificity-index (1 - 0/0)
  expTis = expTis[(rowSums(expTis) != 0),]
  
  return(expTis)
}

# Subset data to only treatments where all tissues are represented
allRowElementsNonzero = function(x){
  all(x > 0)
}

treatmentsWithAllTissues = rownames(studyCountByTreatmentAndTissue)[apply(studyCountByTreatmentAndTissue, MARGIN = 1, FUN = allRowElementsNonzero)]

# phenoAllTissues = pheno[(pheno$TreatmentOntology %in% treatmentsWithAllTissues),]
# tpmdataAllTissues = tpmdata[,(colnames(tpmdata) %in% phenoAllTissues$Experiment)]

# Loop over each treatment with all tissues represented
for(treatment in treatmentsWithAllTissues){
  # subset data to treatment with all tissues represented
  phenoSub = pheno[(pheno$TreatmentOntology == treatment),]
  tpmdataSub = tpmdata[,(colnames(tpmdata) %in% phenoSub$Experiment)]
  
  # calculate tissue profile
  tissueProfileForTreatment = tissueProfile(phenoSub, tpmdataSub)
  
  # calculate tissue specificity
  tissueSpec = as.data.frame(apply(tissueProfileForTreatment, MARGIN = 1, FUN = specIndex))
  tissueSpec$Name = rownames(tissueSpec)
  names(tissueSpec) = c(gsub(" ", "_", paste(treatment, "tissue specificity")), "name")
  
  # merge results with gene feature table
  genes = merge(genes, tissueSpec, by = "name", all.x = T)

}

### PLOT TISSUE SPECIFICITY
# Plot all distributions of treatment specificity at once
tissueSpecData = melt(genes[,c("name", "PECO:0001062_control_exposure_tissue_specificity", "PECO:0007105_abscisic_acid_exposure_tissue_specificity", "PECO:0007162_continuous_light_exposure_tissue_specificity", "PECO:0007319_warm/hot_air_temperature_exposure_tissue_specificity", "PECO:0007332_cold_air_temperature_exposure_tissue_specificity")], id = "name")

tissueSpecData$variable = gsub("_tissue_specificity", "", tissueSpecData$variable)
tissueSpecData$variable = gsub("_", " ", tissueSpecData$variable)

names(tissueSpecData) = c("name", "treatment", "specificity")

ggplot(tissueSpecData, aes(x = treatment, y = specificity, color = treatment)) + 
  theme_classic() +
  theme(axis.ticks.x = element_blank(), axis.text.x = element_blank())+
  labs(x = "", y = "Tissue-specificity index (Tau)") +
  scale_color_manual(values=myColorPalette) +
  geom_violin() +
  geom_boxplot(width = 0.1)

ggsave("densityPlotsOfTissueSpecificity.jpeg")
```

# Tissue specificity partial correlations, before SVA
```{r}
### dn/ds
leafDnTau = na.omit(genes[(genes$name %in% includedGenesDn),c("dN", "dS", "PECO:0001062_control_exposure_tissue_specificity", "leafEnvSpecIndex", "leafAvgExpNoZeros", "GC", "length")])
leafDnTau = pcor(leafDnTau, method = "spearman")

rootDnTau = na.omit(genes[(genes$name %in% includedGenesDn),c("dN", "dS", "PECO:0001062_control_exposure_tissue_specificity", "rootEnvSpecIndex", "rootAvgExpNoZeros", "GC", "length")])
rootDnTau = pcor(rootDnTau, method = "spearman")

plantDnTau = na.omit(genes[(genes$name %in% includedGenesDn),c("dN", "dS", "PECO:0001062_control_exposure_tissue_specificity", "plantEnvSpecIndex", "plantAvgExpNoZeros", "GC", "length")])
plantDnTau = pcor(plantDnTau, method = "spearman")

shootDnTau = na.omit(genes[(genes$name %in% includedGenesDn),c("dN", "dS", "PECO:0001062_control_exposure_tissue_specificity", "shootEnvSpecIndex", "shootAvgExpNoZeros", "GC", "length")])
shootDnTau = pcor(shootDnTau, method = "spearman")

seedDnTau = na.omit(genes[(genes$name %in% includedGenesDn),c("dN", "dS", "PECO:0001062_control_exposure_tissue_specificity", "seedEnvSpecIndex", "seedAvgExpNoZeros", "GC", "length")])
seedDnTau = pcor(seedDnTau, method = "spearman")

flowerDnTau = na.omit(genes[(genes$name %in% includedGenesDn),c("dN", "dS", "PECO:0001062_control_exposure_tissue_specificity", "flowerEnvSpecIndex", "flowerAvgExpNoZeros", "GC", "length")])
flowerDnTau = pcor(flowerDnTau, method = "spearman")

### piN/piS
leafPinTau = na.omit(genes[,c("piN", "piS", "PECO:0001062_control_exposure_tissue_specificity", "leafEnvSpecIndex", "leafAvgExpNoZeros", "GC", "length", "familySize")])
leafPinTau = pcor(leafPinTau, method = "spearman")

rootPinTau = na.omit(genes[,c("piN", "piS", "PECO:0001062_control_exposure_tissue_specificity", "rootEnvSpecIndex", "rootAvgExpNoZeros", "GC", "length", "familySize")])
rootPinTau = pcor(rootPinTau, method = "spearman")

plantPinTau = na.omit(genes[,c("piN", "piS", "PECO:0001062_control_exposure_tissue_specificity", "plantEnvSpecIndex", "plantAvgExpNoZeros", "GC", "length", "familySize")])
plantPinTau = pcor(plantPinTau, method = "spearman")

shootPinTau = na.omit(genes[,c("piN", "piS", "PECO:0001062_control_exposure_tissue_specificity", "shootEnvSpecIndex", "shootAvgExpNoZeros", "GC", "length", "familySize")])
shootPinTau = pcor(shootPinTau, method = "spearman")

seedPinTau = na.omit(genes[,c("piN", "piS", "PECO:0001062_control_exposure_tissue_specificity", "seedEnvSpecIndex", "seedAvgExpNoZeros", "GC", "length", "familySize")])
seedPinTau = pcor(seedPinTau, method = "spearman")

flowerPinTau = na.omit(genes[,c("piN", "piS", "PECO:0001062_control_exposure_tissue_specificity", "flowerEnvSpecIndex", "flowerAvgExpNoZeros", "GC", "length", "familySize")])
flowerPinTau = pcor(flowerPinTau, method = "spearman")

### tajima's D
leafTdTau = na.omit(genes[,c("tajimasD", "PECO:0001062_control_exposure_tissue_specificity", "leafEnvSpecIndex", "leafAvgExpNoZeros", "GC", "length", "familySize")])
leafTdTau = pcor(leafTdTau, method = "spearman")

rootTdTau = na.omit(genes[,c("tajimasD", "PECO:0001062_control_exposure_tissue_specificity", "rootEnvSpecIndex", "rootAvgExpNoZeros", "GC", "length", "familySize")])
rootTdTau = pcor(rootTdTau, method = "spearman")

plantTdTau = na.omit(genes[,c("tajimasD", "PECO:0001062_control_exposure_tissue_specificity", "plantEnvSpecIndex", "plantAvgExpNoZeros", "GC", "length", "familySize")])
plantTdTau = pcor(plantTdTau, method = "spearman")

shootTdTau = na.omit(genes[,c("tajimasD", "PECO:0001062_control_exposure_tissue_specificity", "shootEnvSpecIndex", "shootAvgExpNoZeros", "GC", "length", "familySize")])
shootTdTau = pcor(shootTdTau, method = "spearman")

seedTdTau = na.omit(genes[,c("tajimasD", "PECO:0001062_control_exposure_tissue_specificity", "seedEnvSpecIndex", "seedAvgExpNoZeros", "GC", "length", "familySize")])
seedTdTau = pcor(seedTdTau, method = "spearman")

flowerTdTau = na.omit(genes[,c("tajimasD", "PECO:0001062_control_exposure_tissue_specificity", "flowerEnvSpecIndex", "flowerAvgExpNoZeros", "GC", "length", "familySize")])
flowerTdTau = pcor(flowerTdTau, method = "spearman")

### DoS
leafDoSTau = na.omit(genes[(genes$name %in% includedGenesDn),c("DoS", "PECO:0001062_control_exposure_tissue_specificity", "leafEnvSpecIndex", "leafAvgExpNoZeros", "GC", "length")])
leafDoSTau = pcor(leafDoSTau, method = "spearman")

rootDoSTau = na.omit(genes[(genes$name %in% includedGenesDn),c("DoS", "PECO:0001062_control_exposure_tissue_specificity", "rootEnvSpecIndex", "rootAvgExpNoZeros", "GC", "length")])
rootDoSTau = pcor(rootDoSTau, method = "spearman")

plantDoSTau = na.omit(genes[(genes$name %in% includedGenesDn),c("DoS", "PECO:0001062_control_exposure_tissue_specificity", "plantEnvSpecIndex", "plantAvgExpNoZeros", "GC", "length")])
plantDoSTau = pcor(plantDoSTau, method = "spearman")

shootDoSTau = na.omit(genes[(genes$name %in% includedGenesDn),c("DoS", "PECO:0001062_control_exposure_tissue_specificity", "shootEnvSpecIndex", "shootAvgExpNoZeros", "GC", "length")])
shootDoSTau = pcor(shootDoSTau, method = "spearman")

seedDoSTau = na.omit(genes[(genes$name %in% includedGenesDn),c("DoS", "PECO:0001062_control_exposure_tissue_specificity", "seedEnvSpecIndex", "seedAvgExpNoZeros", "GC", "length")])
seedDoSTau = pcor(seedDoSTau, method = "spearman")

flowerDoSTau = na.omit(genes[(genes$name %in% includedGenesDn),c("DoS", "PECO:0001062_control_exposure_tissue_specificity", "flowerEnvSpecIndex", "flowerAvgExpNoZeros", "GC", "length")])
flowerDoSTau = pcor(flowerDoSTau, method = "spearman")

# store all partial correlation results in list
pcorTauList = list(leafDnTau, leafPinTau, leafTdTau, leafDoSTau, rootDnTau, rootPinTau, rootTdTau, rootDoSTau, plantDnTau, plantPinTau, plantTdTau, plantDoSTau, shootDnTau, shootPinTau, shootTdTau, shootDoSTau, seedDnTau, seedPinTau, seedTdTau, seedDoSTau, flowerDnTau, flowerPinTau, flowerTdTau, flowerDoSTau)

# apply functions to list, change variable names
pcorTauList = lapply(pcorTauList, changePcorVariableNames)
# leafDnTau = changePcorVariableNames(leafDnTau)
# leafPinTau = changePcorVariableNames(leafPinTau)

# Get upper triangle of the correlation matrix, then melt
meltedCormatsTau = lapply(pcorTauList, meltPcorOutput)
# meltedCormatLeafDnTau <- reshape2::melt(get_upper_tri(leafDnTau$estimate), na.rm=TRUE,value.name="value",varnames=c("X1","X2"))
# meltedCormatLeafPinTau <- reshape2::melt(get_upper_tri(leafPinTau$estimate), na.rm=TRUE,value.name="value",varnames=c("X1","X2"))

# plot heatmaps
myHeatmapsTau = lapply(meltedCormatsTau, myHeatmapPlot, "")
# leafDnTauPlot = myHeatmapPlot(meltedCormatLeafDnTau, "")
# leafPinTauPlot = myHeatmapPlot(meltedCormatLeafPinTau, "")

### leaf
allHeatMapsTau = ggarrange(myHeatmapsTau[[1]], myHeatmapsTau[[2]], myHeatmapsTau[[3]], myHeatmapsTau[[4]], ncol = 2, nrow = 2, common.legend = T, labels = c("A", "B", "C", "D"))
ggsave("paritalCorrelations_leaf_tissueSpecificity_2022-07-15.jpeg", allHeatMapsTau, scale = 1, width = 7, height = 7, units = "in")

### root
allHeatMapsTau = ggarrange(myHeatmapsTau[[5]], myHeatmapsTau[[6]], myHeatmapsTau[[7]], myHeatmapsTau[[8]], ncol = 2, nrow = 2, common.legend = T, labels = c("A", "B", "C", "D"))
ggsave("paritalCorrelations_root_tissueSpecificity_2022-07-15.jpeg", allHeatMapsTau, scale = 1, width = 7, height = 7, units = "in")

### plant
allHeatMapsTau = ggarrange(myHeatmapsTau[[9]], myHeatmapsTau[[10]], myHeatmapsTau[[11]], myHeatmapsTau[[12]], ncol = 2, nrow = 2, common.legend = T, labels = c("A", "B", "C", "D"))
ggsave("paritalCorrelations_plant_tissueSpecificity_2022-07-15.jpeg", allHeatMapsTau, scale = 1, width = 7, height = 7, units = "in")

### shoot
allHeatMapsTau = ggarrange(myHeatmapsTau[[13]], myHeatmapsTau[[14]], myHeatmapsTau[[15]], myHeatmapsTau[[16]], ncol = 2, nrow = 2, common.legend = T, labels = c("A", "B", "C", "D"))
ggsave("paritalCorrelations_shoot_tissueSpecificity_2022-07-15.jpeg", allHeatMapsTau, scale = 1, width = 7, height = 7, units = "in")

### seed
allHeatMapsTau = ggarrange(myHeatmapsTau[[17]], myHeatmapsTau[[18]], myHeatmapsTau[[19]], myHeatmapsTau[[20]], ncol = 2, nrow = 2, common.legend = T, labels = c("A", "B", "C", "D"))
ggsave("paritalCorrelations_seed_tissueSpecificity_2022-07-15.jpeg", allHeatMapsTau, scale = 1, width = 7, height = 7, units = "in")

### flower
allHeatMapsTau = ggarrange(myHeatmapsTau[[21]], myHeatmapsTau[[22]], myHeatmapsTau[[23]], myHeatmapsTau[[24]], ncol = 2, nrow = 2, common.legend = T, labels = c("A", "B", "C", "D"))
ggsave("paritalCorrelations_flowerAndFruit_tissueSpecificity_2022-07-15.jpeg", allHeatMapsTau, scale = 1, width = 7, height = 7, units = "in")

# allHeatMaps = grid.arrange(gLeafPi, gRootPi, gPlantPi, nrow = 1, ncol = 3)
# need to play with scale arguement to get figure to size correctly (switch from 3 -> 1 or 1 -> 3)


```

# How does SVA change average expression and specificity index values?
```{r}
# plot(genes$leafAvgExpNoZeros, genes$leafAvgExpNoZerosSva)
# abline(a = 0, b = 1)
# 
# plot(genes$leafEnvSpecIndex, genes$leafEnvSpecIndexSva)
# abline(a = 0, b = 1)

# Function to plot comparisons before and after Sva
avgAndSpecSvaComparison = function(data, avgExpBeforeSva, avgExpAfterSva, specBeforeSva, specAfterSva, tissueType, myColor){
  
  avgExpSvaComparison = ggplot(data, aes_string(x = avgExpBeforeSva, y = avgExpAfterSva)) +
    geom_point(color = myColor) +
    geom_abline(intercept = 0, slope = 1) + 
    theme_classic() +
    theme(plot.title = element_text(hjust = 0.5)) +
    labs(x = "Average expression (excluding TPM < 5) before SVA", y = "Average expression (excluding TPM < 5) after SVA", title = tissueType)

treatSpecSvaComparison = ggplot(data, aes_string(x = specBeforeSva, y = specAfterSva)) +
    geom_point(color = myColor) +
    geom_abline(intercept = 0, slope = 1) + 
    theme_classic() +
    theme(plot.title = element_text(hjust = 0.5)) +
    labs(x = "Treatment-specificity before SVA", y = "Treatment-specificity after SVA", title = tissueType) 

  # avgSpecSvaComparison = ggarrange(avgExpSvaComparison, treatSpecSvaComparison, ncol = 2, nrow = 1, common.legend = T, labels = c("A", "B"))
  
  #return(avgSpecSvaComparison)
  return(list(avgExpSvaComparison, treatSpecSvaComparison))
}


leafSvaComp = avgAndSpecSvaComparison(genes, "leafAvgExpNoZeros", "leafAvgExpNoZerosSva", "leafEnvSpecIndex", "leafEnvSpecIndexSva", "Leaf", myColorPalette[6])

rootSvaComp = avgAndSpecSvaComparison(genes, "rootAvgExpNoZeros", "rootAvgExpNoZerosSva", "rootEnvSpecIndex", "rootEnvSpecIndexSva", "Root", myColorPalette[3])

plantSvaComp = avgAndSpecSvaComparison(genes, "plantAvgExpNoZeros", "plantAvgExpNoZerosSva", "plantEnvSpecIndex", "plantEnvSpecIndexSva", "Whole plant", myColorPalette[1])

shootSvaComp = avgAndSpecSvaComparison(genes, "shootAvgExpNoZeros", "shootAvgExpNoZerosSva", "shootEnvSpecIndex", "shootEnvSpecIndexSva", "Shoot", myColorPalette[4])

seedSvaComp = avgAndSpecSvaComparison(genes, "seedAvgExpNoZeros", "seedAvgExpNoZerosSva", "seedEnvSpecIndex", "seedEnvSpecIndexSva", "Seed", myColorPalette[5])

flowerSvaComp = avgAndSpecSvaComparison(genes, "flowerAvgExpNoZeros", "flowerAvgExpNoZerosSva", "flowerEnvSpecIndex", "flowerEnvSpecIndexSva", "Flower or fruit", myColorPalette[2])

# ggsave("avgExp_envSpec_beforeAndAfterSVA_leaf_2022-07-13.jpeg", leafSvaComp, scale = 1)
# ggsave("avgExp_envSpec_beforeAndAfterSVA_root_2022-07-13.jpeg", rootSvaComp, scale = 1)
# ggsave("avgExp_envSpec_beforeAndAfterSVA_plant_2022-07-13.jpeg", plantSvaComp, scale = 1)
# ggsave("avgExp_envSpec_beforeAndAfterSVA_shoot_2022-07-13.jpeg", shootSvaComp, scale = 1)
# ggsave("avgExp_envSpec_beforeAndAfterSVA_seed_2022-07-13.jpeg", seedSvaComp, scale = 1)
# ggsave("avgExp_envSpec_beforeAndAfterSVA_flowerORfruit_2022-07-13.jpeg", flowerSvaComp, scale = 1)
```

# PCA before and after SVA
```{r}
pcaPreSvaLeaf = prcomp(t(leafTPMdataSva), center = T, scale = T)
pcaPostSvaLeaf = prcomp(t(leafTPMdataSvaClean), center = T, scale = T)

pcaPreSvaRoot = prcomp(t(rootTPMdataSva), center = T, scale = T)
pcaPostSvaRoot = prcomp(t(rootTPMdataSvaClean), center = T, scale = T)

pcaPreSvaPlant = prcomp(t(plantTPMdataSva), center = T, scale = T)
pcaPostSvaPlant = prcomp(t(plantTPMdataSvaClean), center = T, scale = T)

pcaPreSvaShoot = prcomp(t(shootTPMdataSva), center = T, scale = T)
pcaPostSvaShoot = prcomp(t(shootTPMdataSvaClean), center = T, scale = T)

pcaPreSvaSeed = prcomp(t(seedTPMdataSva), center = T, scale = T)
pcaPostSvaSeed = prcomp(t(seedTPMdataSvaClean), center = T, scale = T)

pcaPreSvaFlower = prcomp(t(flowerTPMdataSva), center = T, scale = T)
pcaPostSvaFlower = prcomp(t(flowerTPMdataSvaClean), center = T, scale = T)

# Function to plot PCA results
plotMyPCA = function(prcompOutput, metaData, myTitle, myColors){
  var_explained <- prcompOutput$sdev^2/sum(prcompOutput$sdev^2) # explained variance for each PC

  plotdata = as.data.frame(prcompOutput$x)
  
  # plot PCA
  ggplot(plotdata, aes(x=PC1,y=PC2, col=metaData)) + 
    geom_point() +
    theme_classic() +
    theme(plot.title = element_text(hjust = 0.5)) +
    #theme(plot.title = element_text(hjust = 0.5), legend.position = "bottom") +
    scale_color_manual(values = myColors) +
    #scale_color_manual(values = c("#ffff4d", "#e6f261","#8cbf00"))
    labs(x=paste("PC1: ",round(var_explained[1]*100,2),"%"),
      y=paste("PC2: ",round(var_explained[2]*100,2),"%"),
      color = "Treatment", title = myTitle)
}



# initial plots, replace commas with newlines so that legends are more compact
# create large color palette based on seed colors, share palette for both plots
myColors = unname(createPalette(length(unique(leafPhenoSva$TreatmentOntology)), myColorPalette))
leafPreSva = plotMyPCA(pcaPreSvaLeaf, gsub(", *", "\n", leafPhenoSva$TreatmentOntology), "Leaf - Before SVA", myColors)
leafPostSva = plotMyPCA(pcaPostSvaLeaf, gsub(", *", "\n", leafPhenoSva$TreatmentOntology), "Leaf - After SVA", myColors)

myColors = unname(createPalette(length(unique(rootPhenoSva$TreatmentOntology)), myColorPalette))
rootPreSva = plotMyPCA(pcaPreSvaRoot, gsub(", *", "\n", rootPhenoSva$TreatmentOntology), "Root - Before SVA", myColors)
rootPostSva = plotMyPCA(pcaPostSvaRoot, gsub(", *", "\n", rootPhenoSva$TreatmentOntology), "Root - After SVA", myColors)

myColors = unname(createPalette(length(unique(plantPhenoSva$TreatmentOntology)), myColorPalette))
plantPreSva = plotMyPCA(pcaPreSvaPlant, gsub(", *", "\n", plantPhenoSva$TreatmentOntology), "Whole plant - Before SVa", myColors)
plantPostSva = plotMyPCA(pcaPostSvaPlant, gsub(", *", "\n", plantPhenoSva$TreatmentOntology), "Whole plant - After SVA", myColors)

myColors = unname(createPalette(length(unique(shootPhenoSva$TreatmentOntology)), myColorPalette))
shootPreSva = plotMyPCA(pcaPreSvaShoot, gsub(", *", "\n", shootPhenoSva$TreatmentOntology), "Shoot - Before SVA", myColors)
shootPostSva = plotMyPCA(pcaPostSvaShoot, gsub(", *", "\n", shootPhenoSva$TreatmentOntology), "Shoot - After SVA", myColors)

myColors = unname(createPalette(length(unique(seedPhenoSva$TreatmentOntology)), myColorPalette))
seedPreSva = plotMyPCA(pcaPreSvaSeed, gsub(", *", "\n", seedPhenoSva$TreatmentOntology), "Seed - Before SVA", myColors)
seedPostSva = plotMyPCA(pcaPostSvaSeed, gsub(", *", "\n", seedPhenoSva$TreatmentOntology), "Seed - After SVA", myColors)

myColors = unname(createPalette(length(unique(flowerPhenoSva$TreatmentOntology)), myColorPalette))
flowerPreSva = plotMyPCA(pcaPreSvaFlower, gsub(", *", "\n", flowerPhenoSva$TreatmentOntology), "Flower - Before SVA", myColors)
flowerPostSva = plotMyPCA(pcaPostSvaFlower, gsub(", *", "\n", flowerPhenoSva$TreatmentOntology), "Flower - After SVA", myColors)

# grab legends to plot separately
leafPreSvaLegend = as_ggplot(get_legend(leafPreSva))
rootPreSvaLegend = as_ggplot(get_legend(rootPreSva))
plantPreSvaLegend = as_ggplot(get_legend(plantPreSva))
shootPreSvaLegend = as_ggplot(get_legend(shootPreSva))
seedPreSvaLegend = as_ggplot(get_legend(seedPreSva))
flowerPreSvaLegend = as_ggplot(get_legend(flowerPreSva))

# save plots and legends separately
### leaf
leafPreAndPostSva = ggarrange(leafSvaComp[[1]], leafSvaComp[[2]], leafPreSva + theme(legend.position = "none"), leafPostSva + theme(legend.position = "none"), ncol = 2, nrow = 2, labels = c("A", "B","C", "D"))
ggsave("pca_beforeAndAfterSva_leaf_2022-07-15.jpeg", leafPreAndPostSva, width = 7, height = 7, units = "in") # save plot
ggsave("pca_beforeAndAfterSva_leaf_legend_2022-07-15.jpeg", leafPreSvaLegend) # save legend

### root
rootPreAndPostSva = ggarrange(rootSvaComp[[1]], rootSvaComp[[2]], rootPreSva + theme(legend.position = "none"), rootPostSva + theme(legend.position = "none"), ncol = 2, nrow = 2, labels = c("A", "B","C", "D"))
ggsave("pca_beforeAndAfterSva_root_2022-07-15.jpeg", rootPreAndPostSva, width = 7, height = 7, units = "in") # save plot
ggsave("pca_beforeAndAfterSva_root_legend_2022-07-15.jpeg", rootPreSvaLegend) # save legend

### plant
plantPreAndPostSva = ggarrange(plantSvaComp[[1]], plantSvaComp[[2]], plantPreSva + theme(legend.position = "none"), plantPostSva + theme(legend.position = "none"), ncol = 2, nrow = 2, labels = c("A", "B","C", "D"))
ggsave("pca_beforeAndAfterSva_plant_2022-07-15.jpeg", plantPreAndPostSva, width = 7, height = 7, units = "in") # save plot
ggsave("pca_beforeAndAfterSva_plant_legend_2022-07-15.jpeg", plantPreSvaLegend) # save legend

### shoot
shootPreAndPostSva = ggarrange(shootSvaComp[[1]], shootSvaComp[[2]], shootPreSva + theme(legend.position = "none"), shootPostSva + theme(legend.position = "none"), ncol = 2, nrow = 2, labels = c("A", "B","C", "D"))
ggsave("pca_beforeAndAfterSva_shoot_2022-07-15.jpeg", shootPreAndPostSva, width = 7, height = 7, units = "in") # save plot
ggsave("pca_beforeAndAfterSva_shoot_legend_2022-07-15.jpeg", shootPreSvaLegend) # save legend

### seed
seedPreAndPostSva = ggarrange(seedSvaComp[[1]], seedSvaComp[[2]], seedPreSva + theme(legend.position = "none"), seedPostSva + theme(legend.position = "none"), ncol = 2, nrow = 2, labels = c("A", "B","C", "D"))
ggsave("pca_beforeAndAfterSva_seed_2022-07-15.jpeg", seedPreAndPostSva, width = 7, height = 7, units = "in") # save plot
ggsave("pca_beforeAndAfterSva_seed_legend_2022-07-15.jpeg", seedPreSvaLegend) # save legend

### flower
flowerPreAndPostSva = ggarrange(flowerSvaComp[[1]], flowerSvaComp[[2]], flowerPreSva + theme(legend.position = "none"), flowerPostSva + theme(legend.position = "none"), ncol = 2, nrow = 2, labels = c("A", "B","C", "D"))
ggsave("pca_beforeAndAfterSva_flower_2022-07-15.jpeg", flowerPreAndPostSva, width = 7, height = 7, units = "in") # save plot
ggsave("pca_beforeAndAfterSva_flower_legend_2022-07-15.jpeg", flowerPreSvaLegend) # save legend
```

# Save tables for supplement
```{r}

```
